<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <title>Planet Neuroscientists</title>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet"> 
    <link href="https://fonts.googleapis.com/css?family=Comfortaa" rel="stylesheet">
    <!-- Google analytics -->
    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-60261100-3', 'auto');
      ga('send', 'pageview');
    </script>
    <style>
      /* Sticky footer styles
      -------------------------------------------------- */
      html {
        position: relative;
        min-height: 100%;
      }
      body {
        /* Margin bottom by footer height */
        margin-bottom: 60px;
      }
      .footer {
        position: absolute;
        bottom: 0;
        width: 100%;
        /* Set the fixed height of the footer here */
        height: 60px;
        background-color: #f5f5f5;
      }
      .container .text-muted {
        margin: 20px 0;
      }
    </style>
  </head>
  <body>
    <nav class="navbar navbar-default navbar-static-top">
      <div class="container">
        <div class="navbar-header">
          <a class="navbar-brand" href="#">Planet Neuroscientists</a>
        </div>
          <ul class="nav navbar-nav navbar-right">
            <li class="dropdown">
              <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">More Neuroscience <span class="caret"></span></a>
              <ul class="dropdown-menu">
                <li><a href="http://sanjayankur31.github.io/planet-neuroscience/">Planet Neuroscience</a></li>
                <li><a href="http://home.earthlink.net/~perlewitz/index.html">Computational Neuroscience on the web</a></li>
              </ul>
            </li>
            <li class="dropdown">
              <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Options <span class="caret"></span></a>
              <ul class="dropdown-menu">
                <li><a href="https://github.com/sanjayankur31/planet-neuroscientists/issues/new">Suggest a new feed</a></li>
                <li><a href="https://github.com/sanjayankur31/planet-neuroscientists">View Planet source</a></li>
                <li role="separator" class="divider"></li>
                <li><a href="https://github.com/sanjayankur31/venus">View Venus source</a></li>
              </ul>
            </li>
            <li class="dropdown">
              <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Feeds <span class="caret"></span></a>
              <ul class="dropdown-menu">
                <li><a href="./atom.xml">Atom</a></li>
                <li><a href="./rss10.xml">RSS 1.0</a></li>
                <li><a href="./rss20.xml">RSS 2.0</a></li>
                <li role="separator" class="divider"></li>
                <li><a href="./foafroll.xml">FOAF blogroll</a></li>
                <li><a href="./opml.xml">OPML blogroll</a></li>
              </ul>
            </li>
          </ul>
        </div><!--/.nav-collapse -->
      </div>
    </nav>
    <div class="bodycontent">
      <div class="container">
        <div class="page-header">
          <h1>Planet Neuroscientists</h1>
          <p>
          last updated by <a href="http://intertwingly.net/code/venus/">Venus</a> 
          on March 26, 2017 05:00 PM on behalf of Ankur Sinha.
          </p>
        </div>
        <div class="col-md-9">
          <ul class="list-group">
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with_25.html" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with_25.html" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with_25.html&title=Nerd Food: Northwind, or Using Dogen with ODB - Part IV" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with_25.html">Nerd Food: Northwind, or Using Dogen with ODB - Part IV</a></h4>

Nerd Food: Northwind, or Using Dogen with ODB - Part IV<div id="content"><p>So, dear reader, we meet again for the fourth and final instalment of our series of posts on using Dogen with ODB! And if you missed an episode - unlikely as it may be - well, fear not for you can always catch up! Here are the links: <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with.html">Part I</a>, <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with_24.html">Part II</a> and <a href="http://mcraveiro.blogspot.co.uk/2017/03/nerd-food-northwind-or-using-dogen-with.html">Part III</a>. But, if you are too lazy and need a summary: all we've done thus far is to install and setup an Oracle Express database, populate it with a schema (and data) and finally code-generate an ORM model with Dogen and ODB. </p> <p>I guess it would not be entirely unfair to describe our adventure thus far as a prelude; if nothing else, it was a character building experience. But <i>now</i> we can finally enjoy the code. </p> <div class="outline-2" id="outline-container-sec-1"><h2 id="sec-1">Building Zango</h2><div class="outline-text-2" id="text-1"><p>Assuming you have checked out <code>zango</code> as described in <a href="http://mcraveiro.blogspot.co.uk/2017/03/nerd-food-northwind-or-using-dogen-with.html">Part III</a> and you are sitting on its containing directory, you can "configure" the project fairly simply: </p> <pre class="example"><br />$ . /u01/app/oracle/product/11.2.0/xe/bin/oracle_env.sh<br />$ cd zango<br />$ git pull origin master<br />$ cd build<br />$ mkdir output<br />$ cd output<br />$ CMAKE_INCLUDE_PATH=/full/path/to/local/include CMAKE_LIBRARY_PATH=/full/path/to/local/lib cmake ../.. -G Ninja<br />-- The C compiler identification is GNU 6.3.0<br />&lt;lots of CMake output&gt;<br />-- Generating done<br />-- Build files have been written to: /path/to/zango/build/output<br /></pre> <p>As always, do not forget to replace <code>/full/path/to/local</code> with your path to the directory containing the ODB libraries. If all has gone according to plan, CMake should have found ODB, Boost, Dogen and all other dependencies we have carefully and painstakingly setup in the previous three parts. </p> <p>Once the configuration is done, you can fire up Ninja to build: </p> <pre class="example"><br />$ ninja -j5<br />[1/100] Building CXX object projects/northwind/src/CMakeFiles/northwind.dir/io/category_id_io.cpp.o<br />&lt;lots of Ninja output&gt;<br />[98/100] Linking CXX static library projects/northwind/src/libzango.northwind.a<br />[99/100] Building CXX object CMakeFiles/application.dir/projects/application/main.cpp.o<br />[100/100] Linking CXX executable application<br /></pre> <p>That was easy! But what exactly have we just built? </p></div></div> <div class="outline-2" id="outline-container-sec-2"><h2 id="sec-2">The "Application"</h2><div class="outline-text-2" id="text-2"><p>We've created a really simple application to test drive the northwind model. Of course, this is really <i>not how your production code should look like</i>, but it'll do just fine for our purposes. We shall start by reading a password from the command line and then we use it to instantiate our Oracle database: </p> <div class="org-src-container"> <pre class="src src-c++">    const std::string password(argv[1]);<br />    using odb::oracle::database;<br />    std::unique_ptr&lt;database&gt;<br />        db(new database("northwind", password, "XE", "localhost", 1521));<br /></pre></div> <p>We then use this database to read all available customers: </p> <div class="org-src-container"> <pre class="src src-c++">std::list&lt;zango::northwind::customers&gt;<br />load_customers(odb::oracle::database&amp; db) {<br />    odb::oracle::transaction t(db.begin());<br /><br />    std::list&lt;zango::northwind::customers&gt; r;<br />    auto rs(db.query&lt;zango::northwind::customers&gt;());<br />    for (auto i(rs.begin ()); i != rs.end (); ++i)<br />        r.push_back(*i);<br />    return r;<br />}<br /></pre></div> <p>Please note that this is a straightforward use of the ODB API, but barely scratches the surface of what ODB can do. ODB supports all sorts of weird and wonderful things, including fairly complex queries and other great features. If you'd like more details on how to use ODB, you should read its manual: <a href="http://www.codesynthesis.com/products/odb/doc/manual.xhtml">C++ Object Persistence with ODB</a>. It's extremely comprehensive and very well written. </p> <p>Once we have the customers in memory, we can start to do things with them. We can for example serialise them to a Boost serialisation binary archive and read them back out: </p> <div class="org-src-container"> <pre class="src src-c++">    boost::filesystem::path file("a_file.bin");<br />    {<br />        boost::filesystem::ofstream os(file);<br />        boost::archive::binary_oarchive oa(os);<br />        oa &lt;&lt; customers;<br />    }<br /><br />    std::cout &lt;&lt; "Wrote customers to file: "<br />              &lt;&lt; file.generic_string() &lt;&lt; std::endl;<br /><br />    std::list&lt;zango::northwind::customers&gt; customers_from_file;<br />    {<br />        boost::filesystem::ifstream is(file);<br />        boost::archive::binary_iarchive ia(is);<br />        ia &gt;&gt; customers_from_file;<br />    }<br /></pre></div> <p>This is where hopefully you should start to see the advantages of Dogen: without writing any code, we have full serialisation support to all classes in the model - in addition to ODB support, of course. </p> <p>Another very useful feature is to dump objects into a stream: </p> <div class="org-src-container"> <pre class="src src-c++">    for (const auto&amp; c : customers_from_file)<br />        std::cout &lt;&lt; "Customer: " &lt;&lt; c &lt;&lt; std::endl;<br /></pre></div> <p>The objects are written in JSON, making it easy to post-process the output with JSON tools such as <a href="https://github.com/stedolan/jq">JQ</a>, resulting in a nicely formatted string: </p> <pre class="example"><br />{<br />  "__type__": "zango::northwind::customers",<br />  "customer_id": {<br />    "__type__": "zango::northwind::customer_id",<br />    "value": 90<br />  },<br />  "customer_code": "WILMK",<br />  "company_name": "Wilman Kala",<br />  "contact_name": "Matti Karttunen",<br />  "contact_title": "Owner/Marketing Assistant",<br />  "address": "Keskuskatu 45",<br />  "city": "Helsinki",<br />  "region": "",<br />  "postal_code": "21240",<br />  "country": "Finland",<br />  "phone": "90-224 8858",<br />  "fax": "90-224 8858"<br />}<br /></pre> <p>Dogen supports dumping arbitrarily-nested graphs, so it's great for logging program state as you go along. We make extensive use of this in Dogen, since - <i>of course</i> - we use Dogen to develop Dogen. Whilst this has proven invaluable, we have also hit some limits. For example, sometimes you may bump into really large and complex objects and JQ just won't cut it. But the great thing is that you can always dump the JSON into PostgreSQL - very easily indeed, given the ODB support - and then run queries on the object using the power of <a href="http://schinckel.net/2014/05/25/querying-json-in-postgres/">JSONB</a>. With a tiny bit more bother you can also dump the objects into MongoDB. </p> <p>However, with all of this said, it is also important to notice that we <b>do not support proper JSON serialisation</b> in Dogen at the moment. This will be added Real-Soon-Now, as we have a real need for it in production, but its not there yet. At present all you have is this debug-dumping of objects into streams which <i>happens</i> to be JSON. It is not real JSON serialisation. Real JSON support is very high on our priority list though, so expect it to land in the next few sprints. </p> <p>Another useful Dogen feature is test data generation. This can be handy for performance testing, for example. Let's say we want to generate ~10K customers and see how Oracle fares: </p> <div class="org-src-container"> <pre class="src src-c++">std::vector&lt;zango::northwind::customers&gt; generate_customers() {<br />    std::vector&lt;zango::northwind::customers&gt; r;<br />    const auto total(10 * 1000);<br />    r.reserve(total);<br /><br />    zango::northwind::customers_generator g;<br />    for (int i = 0; i &lt; total; ++i) {<br />        const auto c(g());<br />        if (i &gt; 100)<br />            r.push_back(g());<br />    }<br /><br />    return r;<br />}<br /></pre></div> <p>Note that we skipped the first hundred customers just to avoid clashes with the <code>customer_id</code> primary key. Now, thanks to the magic of ODB we can easily push this data into the database: </p> <div class="org-src-container"> <pre class="src src-c++">void save_customers(odb::oracle::database&amp; db,<br />    const std::vector&lt;zango::northwind::customers&gt;&amp; customers) {<br /><br />    odb::transaction t(db.begin());<br />    for (const auto c : customers)<br />        db.persist(c);<br />    t.commit();<br />}<br /></pre></div> <p><i>Et voilá</i>, we have lots of customers in the database now: </p>  <div class="org-src-container"> <pre class="src src-c++">SQL&gt; select count(1) from customers;<br /><br />  COUNT(1)<br />----------<br />      9990<br /></pre></div> <p>To be totally honest, this exercise revealed a shortcoming in Dogen: since it does not know of the size of fields on the database, the generated test data may in some cases be too big to fit the database fields: </p> <pre class="example"><br />Saving customers...<br />terminate called after throwing an instance of 'odb::oracle::database_exception'<br />  what():  12899: ORA-12899: value too large for column "NORTHWIND"."CUSTOMERS"."CUSTOMER_CODE" (actual: 6, maximum: 5)<br /></pre> <p>I solved this problem with a quick hack for this article (by removing the prefix used in the test data) but a proper fix is now sitting in Dogen's product backlog for implementation in the near future. </p> <p>Finally, just for giggles, I decided to push the data we read from Oracle into <a href="https://redis.io">Redis</a>, an in-memory cache that seems to be all the rage amongst the Cool-Kid community. To keep things simple, I used the C API provided by <a href="https://github.com/redis/hiredis">hiredis</a>. Of course, if this was the real world, I would have used one of the <a href="https://redis.io/clients#c--">many c++ clients</a> for Redis such as <a href="https://github.com/mrpi/redis-cplusplus-client">redis-cplusplus-client</a> or <a href="https://github.com/cylix/cpp_redis">cpp redis</a>. As it was, I could not find any Debian packages for them, so I'll just have to pretend I know C. Since I'm not much of a C programmer, I decided to do a very bad copy and paste job from <a href="http://stackoverflow.com/questions/26799074/can-we-set-c-int-array-as-a-keys-value-in-redis-by-hiredis">this Stack Overflow article</a>. The result was this beauty (forgive me in advance, C programmers): </p> <pre class="example"><br />    redisContext *c;<br />    redisReply *reply;<br />    const char *hostname = "localhost";<br />    int port = 6379;<br />    struct timeval timeout = { 1, 500000 }; // 1.5 seconds<br />    c = redisConnectWithTimeout(hostname, port, timeout);<br />    if (c == NULL || c-&gt;err) {<br />        if (c) {<br />            std::cerr &lt;&lt; "Connection error: " &lt;&lt; c-&gt;errstr &lt;&lt; std::endl;<br />            redisFree(c);<br />        } else {<br />            std::cerr &lt;&lt; "Connection error: can't allocate redis context"<br />                      &lt;&lt; std::endl;<br />        }<br />        return 1;<br />    }<br /><br />    std::ostringstream os;<br />    boost::archive::binary_oarchive oa(os);<br />    oa &lt;&lt; customers;<br />    const auto value(os.str());<br />    const std::string key("customers");<br />    reply = (redisReply*)redisCommand(c, "SET %b %b", key.c_str(),<br />        (size_t) key.size(), value.c_str(), (size_t) value.size());<br />    if (!reply)<br />        return REDIS_ERR;<br />    freeReplyObject(reply);<br /><br />    reply = (redisReply*)redisCommand(c, "GET %b", key.c_str(),<br />        (size_t) key.size());<br />    if (!reply)<br />        return REDIS_ERR;<br /><br />    if ( reply-&gt;type != REDIS_REPLY_STRING ) {<br />        std::cerr &lt;&lt; "ERROR: " &lt;&lt; reply-&gt;str &lt;&lt; std::endl;<br />        return 1;<br />    }<br /><br />    const std::string redis_value(reply-&gt;str, reply-&gt;len);<br />    std::istringstream is(redis_value);<br />    std::list&lt;zango::northwind::customers&gt; customers_from_redis;<br />    boost::archive::binary_iarchive ia(is);<br />    ia &gt;&gt; customers_from_redis;<br />    std::cout &lt;&lt; "Read from redis: " &lt;&lt; customers_from_redis.size()<br />              &lt;&lt; std::endl;<br />    std::cout &lt;&lt; "Front customer (redis): "<br />              &lt;&lt; customers_from_redis.front() &lt;&lt; std::endl;<br />    freeReplyObject(reply);<br /></pre> <p>And it <i>actually works</i>. Here's the output, with manual formatting of JSON: </p> <pre class="example"><br />Read from redis: 91<br />Front customer (redis):  {<br />  "__type__": "zango::northwind::customers",<br />  "customer_id": {<br />    "__type__": "zango::northwind::customer_id",<br />    "value": 1<br />  },<br />  "customer_code": "ALFKI",<br />  "company_name": "Alfreds Futterkiste",<br />  "contact_name": "Maria Anders",<br />  "contact_title": "Sales Representative",<br />  "address": "Obere Str. 57",<br />  "city": "Berlin",<br />  "region": "",<br />  "postal_code": "12209",<br />  "country": "Germany",<br />  "phone": "030-0074321",<br />  "fax": "030-0076545"<br />}<br /></pre> <p>As you can hopefully see, in very few lines of code we managed to connect to a RDBMS, read some data, push it into a stream, read it and write into Boost Serialization archives and push it into and out of Redis. All this in fairly efficient C++ code (and some very dodgy C code, but we'll keep that one quiet). </p> <p>A final note on the CMake targets. Zango comes with a couple of targets for Dogen and ODB: </p> <ul class="org-ul"><li><code>knit_northwind</code> generates the Dogen code from the model. </li><li><code>odb_northwind</code> runs ODB against the Dogen model, generating the ODB sources. </li></ul> <p>The ODB target is added automatically by Dogen. The Dogen target was added manually by yours truly, and it is considered good practice to have one such target when you use Dogen so that other Dogen users know how to generate your models. You can, of course, name it what you like, but in the interest of making everyone's life easier its best if you follow the convention. </p></div></div> <div class="outline-2" id="outline-container-sec-3"><h2 id="sec-3">Oracle and Bulk Fetching</h2><div class="outline-text-2" id="text-3"><p>Whilst I was playing around with ODB and Oracle, I noticed a slight problem: there is no bulk fetch support in the ODB Oracle wrappers at present; it works for other scenarios, but not for selects. I reported this to the main ODB mailing list <a href="http://www.codesynthesis.com/pipermail/odb-users/2017-February/003641.html">here</a>. By the by, the ODB community is very friendly and their mailing list is a very responsive place to chat about ODB issues. </p> <p>Anyway, so you can have an idea of this problem, here's a fetch of our generated customers without prefetch support: </p> <pre class="example"><br />&lt;snip&gt;<br />Generating customers...<br />Generated customers. Size: 9899<br />Saving customers...<br />Saved customers.<br />Read generated customers. Size: 9990 time (ms): 263.449<br />&lt;snip&gt;<br /></pre> <p>Remember the <code>263.449</code> for a moment. Now say you delete all rows we generated: </p> <pre class="example"><br />delete from  customers where customer_id &gt; 100;<br /></pre> <p>Then, say you apply to <code>libodb-oracle</code> the hastily-hacked patch I mentioned <a href="http://www.codesynthesis.com/pipermail/odb-users/2017-February/003648.html">in that mailing list thread</a>. Of course, I am hand-waving here greatly, as you need to rebuild the library, install the binaries, rebuild zango, etc, but you get the gist. At any rate, here's the patch, hard-coding an unscientifically-obtained-prefetch of 5K rows: </p> <pre class="example"><br />--- original_statement.txt 2017-02-09 15:45:56.585765500 +0000<br />+++ statement.cxx        2017-02-13 10:18:28.447916100 +0000<br />@@ -1574,18 +1574,29 @@<br />       OCIError* err (conn_.error_handle ());<br />+      const int prefetchSize(5000);<br />+      sword r = OCIAttrSet (stmt_,<br />+          OCI_HTYPE_STMT,<br />+          (void*)&amp;prefetchSize,<br />+          sizeof(int),<br />+          OCI_ATTR_PREFETCH_ROWS,<br />+          err);<br />+<br />+      if (r == OCI_ERROR || r == OCI_INVALID_HANDLE)<br />+          translate_error (err, r);<br />+<br />       // @@ Retrieve a single row into the already bound output buffers as an<br />       // optimization? This will avoid multiple server round-trips in the case<br />       // of a single object load.<br />       //<br />-      sword r (OCIStmtExecute (conn_.handle (),<br />+      r = OCIStmtExecute (conn_.handle (),<br />                                stmt_,<br />                                err,<br />                                0,<br />                                0,<br />                                0,<br />                                0,<br />-                               OCI_DEFAULT));<br />+                               OCI_DEFAULT);<br />       if (r == OCI_ERROR || r == OCI_INVALID_HANDLE)<br />         translate_error (conn_, r);<br /></pre> <p>And now re-run the command: </p> <pre class="example"><br />Generated customers. Size: 9899<br />Saving customers...<br />Saved customers.<br />Read generated customers. Size: 9990 time (ms): 40.85<br /></pre> <p>Magic! We're down to <code>40.85</code>. Now that I have a proper setup, I am going to start working on upstreaming this patch, so that ODB can expose the fetch configuration for fetching in a similar manner it already does for other purposes. If you are interested in the gory technical details, have a look at <a href="http://www.codesynthesis.com/pipermail/odb-users/2017-February/003651.html">Boris' reply</a>. </p></div></div> <div class="outline-2" id="outline-container-sec-4"><h2 id="sec-4">Conclusion</h2><div class="outline-text-2" id="text-4"><p>Hopefully this concluding part gave you an idea of why you might want to use Dogen with ODB for your modeling needs. Sadly, its not easy to frame the discussion adequately, so that you have all the required context in order to place these two tools in the continuum of tooling; but I'm hoping this series of articles was useful to at least help you setup Oracle Express in Debian and get an idea of what you can do with these two tools. </p></div></div></div><div class="status" id="postamble"><p class="date">Created: 2017-03-25 Sat 20:26</p><p class="creator"><a href="http://www.gnu.org/software/emacs/">Emacs</a> 25.1.1 (<a href="http://orgmode.org">Org</a> mode 8.2.10)</p><p class="validation"><a href="http://validator.w3.org/check?uri=referer">Validate</a></p></div>
                <p class="text-muted">in <a href="http://mcraveiro.blogspot.com/">Marco Craveiro</a> on March 25, 2017 08:27 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://mcraveiro.blogspot.com/2017/02/nerd-food-northwind-or-using-dogen-with.html" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://mcraveiro.blogspot.com/2017/02/nerd-food-northwind-or-using-dogen-with.html" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://mcraveiro.blogspot.com/2017/02/nerd-food-northwind-or-using-dogen-with.html&title=Nerd Food: Northwind, or Using Dogen with ODB - Part I" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://mcraveiro.blogspot.com/2017/02/nerd-food-northwind-or-using-dogen-with.html">Nerd Food: Northwind, or Using Dogen with ODB - Part I</a></h4>

Nerd Food: Northwind, or Using Dogen with ODB - Part I<div id="content"><p>Thanks to my first <a href="https://github.com/DomainDrivenConsulting/dogen">Dogen</a> paying customer, I finally got a chance to work with <a href="http://www.codesynthesis.com/products/odb/">ODB</a> - <a href="http://www.codesynthesis.com">Code Synthesis'</a> amazingly interesting C++ Object-Relational Mapping tool, built on the back of the GCC plugin system. I've personally always been in awe of what Boris Kolpackov has achieved, and, of course, it being a South African company made me all the more keen to use their wares. More importantly: the product just seems to rock in terms of polish, features and documentation. </p> <p>Astute readers of this blog will point out that Dogen has been supporting ODB for quite some time. That is indeed true, but since I haven't used this feature in anger, I wasn't sure how good the support really was; our fairly trivial database model (<a href="https://raw.githubusercontent.com/DomainDrivenConsulting/dogen/fe6a0c7b48b49554d3211e1a2c2b07fb0f09d502/projects/input_models/database.dia">Dia</a>) explored only a small fraction of what is possible. Now that I finally had a chance to use it in production, I needed to expand the tests and try to replicate the customer's scenario as close as possible. As always in these situations, there was a snag: instead of using <a href="https://www.postgresql.org/">PostgreSQL</a> - the RDBMS I had originally used in my Dogen tests - they were using Oracle. So my first task was to setup Oracle locally on my beloved Debian Linux. </p> <p>Never one to miss an opportunity, I decided this adventure was worthy of a quick blog post; it soon turned out to be a series of posts, if I was to do any justice to this strange and wild experiment, through all of its twists and turns. But hopefully it is worth the effort, as it also demonstrates what you can do with Dogen and ODB. And so, here we are. </p> <p>The first part of the series deals with just trying to convince Oracle to run on Debian Testing - something that <i>obviously</i> Oracle does not support out of the box. </p> <p>Before we proceed, a word to the wise: this is a <b>highly experimental</b>Oracle setup, which I embarked upon just because I could; follow it at your own peril, and <b>do not use it anywhere near production</b>. More generally, if you want to stick to the beaten track, use Oracle on RHEL, CentOS or - god forbid - even Windows. All of that said, if like me, you are a Debian person, well, there's not much for it other than to fire off a VM and start looking for those damned faint tracks in the jungle. </p> <div class="outline-2" id="outline-container-sec-1"><h2 id="sec-1">Alien Worlds</h2><div class="outline-text-2" id="text-1"><p>The very first stumbling block was Oracle itself. After all, for all the many years of using this RBDMS at work - more than I care to admit in polite company - I suddenly realised I actually never used it at home. Or course, Oracle has supported Linux for a little while now; and the good news is they have a "free" version available: Oracle Database Express Edition (XE). A quick glance at the Oracle website revealed RPM's for 64-bits (Intel only, of course). So before anything else, I decided to brush up my knowledge of Alien. </p> <p>Alien is a debian package that converts RPMs into DEBs. I've used it in the past for another (lovely) Oracle technology: Java. It had worked wonderfully well then so I thought I'd give it a try. The <a href="https://help.ubuntu.com/community/RPM/AlienHowto">Ubuntu Alien HowTo</a> is pretty straightforward, and so is <a href="https://wiki.debian.org/Alien">Debian's</a>. Basically, obtain Alien: </p> <pre class="example"><br />sudo apt-get install alien<br /></pre> <p>And then apply it to the RPM in question. So the next quest was obtaining those darn RPMs. </p> <p>Of course, once you move away from the easy world of Free and Open Source Software, things start to get a bit more complicated. Those lovely links you can easy Google for don't actually work until you sign up for an Oracle developer account, asking <i>all sorts</i> of personal information. Sadly, even listening to Tesla earnings conferences requires registering these days. Undaunted, I filled all required fields, obtained my developer account and returned to download my loot. For Oracle Express it's rather straightforward: there is a grand-total of one package for Linux 64-bit, so you can't really go wrong. Here's the link, just in case: </p> <ul class="org-ul"><li><a href="http://www.oracle.com/technetwork/database/database-technologies/express-edition/downloads/index.html">Oracle Express</a>: download the 64-bit Linux RPM <code>oracle-xe-11.2.0-1.0.x86_64.rpm.zip</code>. </li></ul> <p>It is interesting that they decided to zip the RPM but you can easily unzip it with the <code>unzip</code> tool. The contents are the RPM Alien expects, plus a few oracle specific files which I decided to ignore for now: </p> <pre class="example"><br />$ unzip oracle-xe-11.2.0-1.0.x86_64.rpm.zip<br />Archive:  oracle-xe-11.2.0-1.0.x86_64.rpm.zip<br />   creating: Disk1/<br />   creating: Disk1/upgrade/<br />  inflating: Disk1/upgrade/gen_inst.sql<br />   creating: Disk1/response/<br />  inflating: Disk1/response/xe.rsp<br />  inflating: Disk1/oracle-xe-11.2.0-1.0.x86_64.rpm<br /></pre> <p>From a quick glance at the instructions, it appeared the Oracle Express package contained just the database server - that meant it did not include a command line client, or the APIs to build applications that talk to the database. To be fair, this is not an entirely uncommon approach; Debian also packages the PostgreSQL server separately from the development libraries. But behind <code>apt-get</code> and <code>synaptic</code>, installation of packages is all so trivial. Not so when you have to go through lots of detailed explanations of different packages and variations. But; onwards! In the <a href="http://www.oracle.com/technetwork/topics/linuxx86-64soft-092277.html">Instant client</a> page, I settled on the downloading the following: </p> <ul class="org-ul"><li><i>Basic</i>: client shared libraries. Package: <code>oracle-instantclient12.1-basic-12.1.0.2.0-1.x86_64.rpm</code></li><li><i>SQL Plus</i>: command-line client. Package: <code>oracle-instantclient12.1-sqlplus-12.1.0.2.0-1.x86_64.rpm</code></li><li><i>SDK</i>: header files to compile code. Package: <code>oracle-instantclient12.1-devel-12.1.0.2.0-1.x86_64.rpm</code></li></ul> <p><b>Update</b>: As it turns out, I was wrong on my original expectations, and you don't really need the SQL Plus package - its already included with Oracle Express. But I only figured it out much later, so I'll leave the steps as I originally followed them. </p> <p>With all of these packages in hand, I swiftly got busy with Alien, only to also rather swiftly hit an issue: </p> <pre class="example"><br />$ cd Disk1<br />$ alien --scripts oracle-xe-11.2.0-1.0.x86_64.rpm<br />Must run as root to convert to deb format (or you may use fakeroot).<br /></pre> <p>Yes, sadly you cannot run alien directly as an unprivileged user. I did not wish to start reading up on <a href="https://wiki.debian.org/FakeRoot">FakeRoot</a> - seems straightforward enough, to be fair, but hey - so I took the easy way out and ran all the Alien commands as root. Note also the <code>--scripts</code> to ensure the scripts will also get converted across. This will bring us some other… interesting issues, shall we say, but seems worthwhile doing. </p> <p>Quite a few seconds later (hey, it was a 300 MB RPM!), a nice looking DEB was generated: </p> <pre class="example"><br /># alien --scripts oracle-xe-11.2.0-1.0.x86_64.rpm<br />oracle-xe_11.2.0-2_amd64.deb generated<br /></pre> <p>A rather promising start. For good measure, I repeated the process with all RPMs, all with similar results: </p> <pre class="example"><br /># alien oracle-instantclient12.1-basic_12.1.0.2.0-2_amd64.deb<br />oracle-instantclient12.1-basic_12.1.0.2.0-2_amd64.deb generated<br /><br /># alien oracle-instantclient12.1-sqlplus-12.1.0.2.0-1.x86_64.rpm<br />oracle-instantclient12.1-sqlplus_12.1.0.2.0-2_amd64.deb generated<br /><br /># alien  oracle-instantclient12.1-devel-12.1.0.2.0-1.x86_64.rpm<br />oracle-instantclient12.1-devel_12.1.0.2.0-2_amd64.deb generated<br /></pre> <p>Voila, all DEBs generated. Of course, as the English love to say, the proof is in the pudding - whatever that means, exactly. So before one can celebrate, you should try to install the generated packages. That can be easily done with our old trusty <code>dpkg</code>: </p> <pre class="example"><br /># dpkg -i oracle-xe_11.2.0-2_amd64.deb<br />Selecting previously unselected package oracle-xe.<br />(Reading database ... 564824 files and directories currently installed.)<br />Preparing to unpack oracle-xe_11.2.0-2_amd64.deb ...<br />Unpacking oracle-xe (11.2.0-2) ...<br />Setting up oracle-xe (11.2.0-2) ...<br />Executing post-install steps...<br />/var/lib/dpkg/info/oracle-xe.postinst: line 114: /sbin/chkconfig: No such file or directory<br />You must run '/etc/init.d/oracle-xe configure' as the root user to configure the database.<br /><br />Processing triggers for libc-bin (2.24-8) ...<br />Processing triggers for systemd (232-8) ...<br />Processing triggers for desktop-file-utils (0.23-1) ...<br />Processing triggers for gnome-menus (3.13.3-8) ...<br />Processing triggers for mime-support (3.60) ...<br />Processing triggers for mime-support (3.60) ...<br /></pre> <p>As it turns out, it seems the error for chkconfig is related to setting up the service to  autostart. Since this was not a key requirement for my purposes, I decided to ignore it. The remaining RPMs - or should I say DEBs - installed beautifully: </p> <pre class="example"><br /># dpkg -i oracle-instantclient12.1-basic_12.1.0.2.0-2_amd64.deb<br />Selecting previously unselected package oracle-instantclient12.1-basic.<br />(Reading database ... 564801 files and directories currently installed.)<br />Preparing to unpack oracle-instantclient12.1-basic_12.1.0.2.0-2_amd64.deb ...<br />Unpacking oracle-instantclient12.1-basic (12.1.0.2.0-2) ...<br />Setting up oracle-instantclient12.1-basic (12.1.0.2.0-2) ...<br />Processing triggers for libc-bin (2.24-8) ...<br /></pre> <pre class="example"><br /># dpkg -i oracle-instantclient12.1-sqlplus_12.1.0.2.0-2_amd64.deb<br />Selecting previously unselected package oracle-instantclient12.1-sqlplus.<br />(Reading database ... 567895 files and directories currently installed.)<br />Preparing to unpack oracle-instantclient12.1-sqlplus_12.1.0.2.0-2_amd64.deb ...<br />Unpacking oracle-instantclient12.1-sqlplus (12.1.0.2.0-2) ...<br />Setting up oracle-instantclient12.1-sqlplus (12.1.0.2.0-2) ...<br /></pre> <pre class="example"><br /># dpkg -i oracle-instantclient12.1-devel_12.1.0.2.0-2_amd64.deb<br />Selecting previously unselected package oracle-instantclient12.1-devel.<br />(Reading database ... 567903 files and directories currently installed.)<br />Preparing to unpack oracle-instantclient12.1-devel_12.1.0.2.0-2_amd64.deb ...<br />Unpacking oracle-instantclient12.1-devel (12.1.0.2.0-2) ...<br />Setting up oracle-instantclient12.1-devel (12.1.0.2.0-2) ...<br /></pre></div></div> <div class="outline-2" id="outline-container-sec-2"><h2 id="sec-2">Talking to the Oracle</h2><div class="outline-text-2" id="text-2"><p>So, at this point in time we have a bunch of stuff installed in all sorts of random (read: Oracle-like) locations. The database itself is under <code>/u01/app/oracle/product/11.2.0/</code>, and all the other packages seemed to have gone into <code>/usr/lib/oracle/12.1/client64/</code> and <code>/usr/include/oracle/12.1/client64/</code>. The first task is now to start the database server. For this we can rely on the scripts we installed earlier on. However, before we proceed, one little spoiler: we need to ensure the scripts can find <code>awk</code> at <code>/bin/awk</code> (these days it lives in <code>/usr/bin/awk</code>). For this we can do a swift (and brutal) hack: </p> <pre class="example"><br /># ln -s /usr/bin/awk /bin/awk<br /></pre> <p>Now we can configure it. I accepted all of the defaults, and setup a suitably sensible password: </p> <pre class="example"><br /># cd /etc/init.d/<br /># /etc/init.d/oracle-xe configure<br /><br />Oracle Database 11g Express Edition Configuration<br />-------------------------------------------------<br />This will configure on-boot properties of Oracle Database 11g Express<br />Edition.  The following questions will determine whether the database should<br />be starting upon system boot, the ports it will use, and the passwords that<br />will be used for database accounts.  Press &lt;Enter&gt; to accept the defaults.<br />Ctrl-C will abort.<br /><br />Specify the HTTP port that will be used for Oracle Application Express [8080]:<br /><br />Specify a port that will be used for the database listener [1521]:<br /><br />Specify a password to be used for database accounts.  Note that the same<br />password will be used for SYS and SYSTEM.  Oracle recommends the use of<br />different passwords for each database account.  This can be done after<br />initial configuration:<br /><br />Confirm the password:<br /><br /><br />Do you want Oracle Database 11g Express Edition to be started on boot (y/n) [y]:y<br />y<br /><br />Starting Oracle Net Listener...Done<br />Configuring database...<br />Starting Oracle Database 11g Express Edition instance...Done<br />Installation completed successfully.<br /></pre> <p>Notice how your port 8080 has been hogged. If you are using it for other work, you may need to move the Oracle Application Express server to some other port. At any rate, after this I could indeed see a whole load of Oracle processes running: </p> <pre class="example"><br />$ ps -ef | grep oracle<br />oracle   20228     1  0 22:35 ?        00:00:00 /u01/app/oracle/product/11.2.0/xe/bin/tnslsnr LISTENER -inhe<br />oracle   21251     1  0 22:36 ?        00:00:00 xe_pmon_XE<br />oracle   21253     1  0 22:36 ?        00:00:00 xe_psp0_XE<br />oracle   21257     1  0 22:36 ?        00:00:00 xe_vktm_XE<br />oracle   21261     1  0 22:36 ?        00:00:00 xe_gen0_XE<br />oracle   21263     1  0 22:36 ?        00:00:00 xe_diag_XE<br />oracle   21265     1  0 22:36 ?        00:00:00 xe_dbrm_XE<br />oracle   21267     1  0 22:36 ?        00:00:00 xe_dia0_XE<br />oracle   21269     1  0 22:36 ?        00:00:00 xe_mman_XE<br />oracle   21271     1  0 22:36 ?        00:00:00 xe_dbw0_XE<br />oracle   21273     1  0 22:36 ?        00:00:00 xe_lgwr_XE<br />...<br /></pre> <p>To the untrained eye, this seems like a healthy start; but for more details, there are also a bunch of useful logs under the Oracle directories: </p> <pre class="example"><br /># ls -l /u01/app/oracle/product/11.2.0/xe/config/log<br />ls -l /u01/app/oracle/product/11.2.0/xe/config/log<br />total 20<br />-rw-r--r-- 1 oracle dba 1369 Feb 23 22:36 CloneRmanRestore.log<br />-rw-r--r-- 1 oracle dba 7377 Feb 23 22:36 cloneDBCreation.log<br />-rw-r--r-- 1 oracle dba 1278 Feb 23 22:36 postDBCreation.log<br />-rw-r--r-- 1 oracle dba  227 Feb 23 22:36 postScripts.log<br /></pre> <p>Now, at this point in time, if all had gone according to plan we should be able to connect to our new instance. A typical trick in Oracle is to use <code>tnsping</code> to validate the setup. For this we need to know what to ping, and that is where <a href="https://docs.oracle.com/database/121/NETRF/tnsnames.htm#NETRF260">TNS Names</a> comes in handy: </p> <pre class="example"><br />$ cat /u01/app/oracle/product/11.2.0/xe/network/admin/tnsnames.ora<br /># tnsnames.ora Network Configuration File:<br /><br />XE =<br />  (DESCRIPTION =<br />    (ADDRESS = (PROTOCOL = TCP)(HOST = lorenz)(PORT = 1521))<br />    (CONNECT_DATA =<br />      (SERVER = DEDICATED)<br />      (SERVICE_NAME = XE)<br />    )<br />  )<br />...<br /></pre> <p>The magic word is <code>XE</code> (the net service name, i.e. what we will be connecting against). Now we can simply do: </p> <pre class="example"><br />$ . /u01/app/oracle/product/11.2.0/xe/bin/oracle_env.sh<br />$ tnsping XE<br /><br />TNS Ping Utility for Linux: Version 11.2.0.2.0 - Production on 23-FEB-2017 22:52:04<br /><br />Copyright (c) 1997, 2011, Oracle.  All rights reserved.<br /><br />Used parameter files:<br /><br /><br />Used TNSNAMES adapter to resolve the alias<br />Attempting to contact (DESCRIPTION = (ADDRESS = (PROTOCOL = TCP)(HOST = lorenz)(PORT = 1521)) (CONNECT_DATA = (SERVER = DEDICATED) (SERVICE_NAME = XE)))<br />OK (0 msec)<br /></pre> <p>Success! Worth noticing that the first step was to call <code>oracle_env.sh</code> to bring in all the required environment variables of our Oracle setup. </p> <p>The final test at this stage is to ensure we can connect with SQL Plus. For this we will just rely on the <code>SYSTEM</code> user. </p> <pre class="example"><br />$ sqlplus SYSTEM@XE<br /><br />SQL*Plus: Release 11.2.0.2.0 Production on Thu Feb 23 22:56:31 2017<br /><br />Copyright (c) 1982, 2011, Oracle.  All rights reserved.<br /><br />Enter password:<br /><br />Connected to:<br />Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production<br /><br />SQL&gt; select table_name from all_tables where rownum &lt; 4;<br />select table_name from all_tables where rownum &lt; 4;<br /><br />TABLE_NAME<br />------------------------------<br />ICOL$<br />CON$<br />UNDO$<br /></pre> <p>And there you go. We have an absolutely minimal, bare-bones setup of Oracle Express running on Debian Linux. Worth bearing in mind that if you want to make use of SQL Plus from within emacs you must make sure you start emacs on a shell that has all the variables defined in <code>oracle_env.sh</code>. </p></div></div> <div class="outline-2" id="outline-container-sec-3"><h2 id="sec-3">Conclusions</h2><div class="outline-text-2" id="text-3"><p>In this first part we simply setup Oracle Express, and the client libraries. We also managed to prove that the setup is vaguely working by connecting to it first at a low-level via TNS ping and then at a proper client level using SQL Plus. The next part will wrap things up with the Oracle setup and then move on to ODB. </p></div></div></div><div class="status" id="postamble"><p class="date">Created: 2017-03-25 Sat 19:54</p><p class="creator"><a href="http://www.gnu.org/software/emacs/">Emacs</a> 25.1.1 (<a href="http://orgmode.org">Org</a> mode 8.2.10)</p><p class="validation"><a href="http://validator.w3.org/check?uri=referer">Validate</a></p></div>
                <p class="text-muted">in <a href="http://mcraveiro.blogspot.com/">Marco Craveiro</a> on March 25, 2017 07:54 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/25/weekend-reads-risks-spotlighting-reproducibility-harassment-scientific-misconduct-trouble-funnel-plots/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/25/weekend-reads-risks-spotlighting-reproducibility-harassment-scientific-misconduct-trouble-funnel-plots/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/25/weekend-reads-risks-spotlighting-reproducibility-harassment-scientific-misconduct-trouble-funnel-plots/&title=Weekend reads: The risks of spotlighting reproducibility; harassment = scientific misconduct?; trouble with funnel plots" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/25/weekend-reads-risks-spotlighting-reproducibility-harassment-scientific-misconduct-trouble-funnel-plots/">Weekend reads: The risks of spotlighting reproducibility; harassment = scientific misconduct?; trouble with funnel plots</a></h4>

<p>The week at Retraction Watch featured the case of a peer review nightmare, and a story about harassment by a would-be scientific critic. Here’s what was happening elsewhere: “Mr. Shiffrin […] said he recognizes the now-very-clear risk of spotlighting reproducibility while a U.S. president and a like-minded Congress appear eager to question the fundamental tenets of […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/25/weekend-reads-risks-spotlighting-reproducibility-harassment-scientific-misconduct-trouble-funnel-plots/" rel="nofollow">Weekend reads: The risks of spotlighting reproducibility; harassment = scientific misconduct?; trouble with funnel plots</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 25, 2017 02:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.elsevier.com/connect/elsevier-regrets-hrk-unwillingness-to-progress-german-national-agreement-talks-urges-return-to-negotiations" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.elsevier.com/connect/elsevier-regrets-hrk-unwillingness-to-progress-german-national-agreement-talks-urges-return-to-negotiations" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.elsevier.com/connect/elsevier-regrets-hrk-unwillingness-to-progress-german-national-agreement-talks-urges-return-to-negotiations&title=Elsevier regrets HRK unwillingness to progress German national agreement talks, urges return to negotiations" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.elsevier.com/connect/elsevier-regrets-hrk-unwillingness-to-progress-german-national-agreement-talks-urges-return-to-negotiations">Elsevier regrets HRK unwillingness to progress German national agreement talks, urges return to negotiations</a></h4>

Elsevier and the Hochschulrektorenkonferenz (HRK) met on Thursday to continue discussions to develop a nationwide agreement for access to Elsevier’s academic literature database ScienceDirect
                <p class="text-muted">in <a href="">Elsevier Connect</a> on March 24, 2017 11:52 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://blogs.discovermagazine.com/neuroskeptic/?p=8646" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://blogs.discovermagazine.com/neuroskeptic/?p=8646" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://blogs.discovermagazine.com/neuroskeptic/?p=8646&title=The Misuse of Meta-Analysis?" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://blogs.discovermagazine.com/neuroskeptic/?p=8646">The Misuse of Meta-Analysis?</a></h4>

Over at Data Colada, Uri Simonsohn argues that The Funnel Plot is Invalid.

Funnel plots are a form of scatter diagram which are widely used to visualize possible publication bias in the context of a meta-analysis. In a funnel plot, each data point represents one of the studies in the meta-analysis. The x-axis shows the effect size reported by the study, while the y-axis represents the standard error of the effect size, which is usually inversely related to the sample size.



In theory, t
                <p class="text-muted">in <a href="http://discovermagazine.com/rss/blog-feeds/neuroskeptic">Discovery magazine - Neuroskeptic</a> on March 24, 2017 10:33 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/blog/scicurious/most-americans-science-and-are-willing-pay-it" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/blog/scicurious/most-americans-science-and-are-willing-pay-it" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/blog/scicurious/most-americans-science-and-are-willing-pay-it&title=Most Americans like science — and are willing to pay for it" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/blog/scicurious/most-americans-science-and-are-willing-pay-it">Most Americans like science — and are willing to pay for it</a></h4>

Americans drastically overestimate how much the government spends on science. But when correctly informed, they want the government to spend more.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 24, 2017 05:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/ancient-romans-may-have-been-cozier-huns-they-let" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/ancient-romans-may-have-been-cozier-huns-they-let" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/ancient-romans-may-have-been-cozier-huns-they-let&title=Ancient Romans may have been cozier with Huns than they let on" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/ancient-romans-may-have-been-cozier-huns-they-let">Ancient Romans may have been cozier with Huns than they let on</a></h4>

Nomadic Huns and Roman farmers shared ways of life on the Roman Empire’s fifth century frontier.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 24, 2017 03:38 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/24/three-tough-years-macchiarini-whistleblower-cleared-previous-charges/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/24/three-tough-years-macchiarini-whistleblower-cleared-previous-charges/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/24/three-tough-years-macchiarini-whistleblower-cleared-previous-charges/&title=“It’s been three tough years:” Macchiarini whistleblower cleared of previous charges" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/24/three-tough-years-macchiarini-whistleblower-cleared-previous-charges/">“It’s been three tough years:” Macchiarini whistleblower cleared of previous charges</a></h4>

<p>Karl-Henrik Grinnemo was worried. The doctor and clinical researcher at the Karolinska Institute was working with a high-profile surgeon who was performing a potentially life-saving procedure on patients, but Grinnemo saw that the patients weren’t doing very well. So in 2013, Grinnemo and three other doctors raised concerns about the work of Paolo Macchiarini. The […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/24/three-tough-years-macchiarini-whistleblower-cleared-previous-charges/" rel="nofollow">“It’s been three tough years:” Macchiarini whistleblower cleared of previous charges</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 24, 2017 03:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/looking-needle-haystack-find-mycobacterial-dna-animal-tissue-samples/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/looking-needle-haystack-find-mycobacterial-dna-animal-tissue-samples/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/looking-needle-haystack-find-mycobacterial-dna-animal-tissue-samples/&title=Looking for the needle in a haystack – or how to find mycobacterial DNA in animal tissue samples" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/looking-needle-haystack-find-mycobacterial-dna-animal-tissue-samples/">Looking for the needle in a haystack – or how to find mycobacterial DNA in animal tissue samples</a></h4>

<img alt="TB-Cover-2" class="attachment-thumbnail size-thumbnail wp-post-image" height="110" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/TB-Cover-2-140x110.jpg" style="display: block; margin-bottom: 5px; clear: both;" width="140" /><p>Tuberculosis (TB) is a zoonotic infectious disease and is still of significance in both humans and animals worldwide. It remains an epidemic in much of the world, causing the death of nearly one-and-a-half-million people each year, mostly in developing countries.</p>
<p>Human tuberculosis is primarily caused by <em>Mycobacterium tuberculosis</em>, while the bacteria <em>M. bovis</em> and <em>M. caprae</em> are known to be the causative agents of bovine tuberculosis (bTB). The three pathogens belong to the <em>Mycobacterium-tuberculosis</em>-complex (MTC) and can cause tuberculosis not only in humans and cattle, but also in other mammalian species.</p>
<p>Bovine tuberculosis in cattle was widespread in Germany until the second half of the 20<sup>th</sup> century. Due to the effective control and eradication campaigns Germany was declared “officially free of tuberculosis” (OTF) by the European Commission in 1996. Long-term eradication programs have shown that pasteurization of milk in developed countries also reduced the human <em>M.-bovis-</em> and <em>M.-caprae-</em> infections.</p>
<p>Since 2008 cases of bovine tuberculosis in cattle and red deer have occurred consistently in the Bavarian Alps. The Bavarian alpine region is an area with plenty of cattle summer pastures. Cattle on these summer pastures can acquire the infection from red deer and this ruminant is considered to be a reservoir host for bTB.</p>
<p>Bovine tuberculosis can manifest as an acute or chronic infectious disease with a long subclinical phase, in which infected animals do not show clinical signs and with the potential of reactivation after years. After infection, the pathogens hide and persist inside macrophages and usually the host immune system forms a tuberculoid granuloma to prevent further spread of these pathogens (Figure 1). Bacilli within the center of the lesion do not multiply but may remain dormant and the resulting latent infection may persist for years. Immunosuppression can disturb the balance between host control and agent dissemination, allowing replication of the silenced pathogen with reactivation and spread of the infection. However, in animals generalized manifestation of the infection is less common than in humans.</p>
<figure class="wp-caption alignleft" id="attachment_8762" style="width: 1220px;"><img alt="Figure 1: Granuloma in a lymph node from a M.-caprae-infected cattle; January 2014" class="size-full wp-image-8762" height="806" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/TB-blog-1a.jpg" width="1210" /><div class="attachment-image--caption">Figure 1: Granuloma in a lymph node from a M.-caprae-infected cattle; January 2014</div><div class="attachment-image--source"></div></figure>
<p> </p>
<p>The diagnosis of bTB is challenging. Non-visible lesions may develop in individuals after infection with bTB, which is in contrast to the typical granuloma formation found mostly in lymphoid tissue. Mycobacterial infections are sometimes only very small granulomas or tiny lesions, especially in early stages of infection, which are difficult to detect</p>
<p>As a consequence, diagnostic procedures for the detection of bTB in cattle are challenging, expensive and currently are based on an authority regulated prescribed protocol. Indicating the infectious status of a living animal, cellular immune reactions are measured by using the tuberculin skin test or an interferon gamma release assay. The immune response should then be confirmed after the animals´ death by direct detection of the pathogen via bacteriological culture or its DNA using PCR.</p>
<p>For the extraction of DNA that serves as a template for subsequent PCR, only very small amounts of the tissue of the size of a rice grain are commonly used. This may lead to false negative results due to the inhomogeneous distribution of mycobacteria in tissue material. In addition, animals infected in an early pre-granuloma forming status do not show granulomas at all (non-visible lesions).</p>
<p>This situation makes it impossible to recognize mycobacterial presence and localization in the organ to allow targeted DNA extraction – choosing adequate, bacteria containing tissue sample is like looking for the needle in a haystack.</p>
<p>In our study we therefore used a magnet, throwing it into the haystack to find needles (= MTC-DNA). We processed very large sample volumes to increase the probability to include mycobacterial DNA in the investigated sample and used the principle of magnetic capture to concentrate MTC-DNA out of the large sample pool (Figure 2). In a second approach, remaining sediments from the magnetic capture protocol were further processed with a less complex DNA extraction protocol that can be used in daily routine diagnostics.</p>
<figure class="wp-caption alignleft" id="attachment_8760" style="width: 1436px;"><img alt="Figure 2: Principle of magnetic capture " class="size-full wp-image-8760" height="768" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/TB-blog-2.png" width="1426" /><div class="attachment-image--caption">Figure 2: Principle of magnetic capture</div><div class="attachment-image--source"></div></figure>
<p>The use of large sample volumes (up to 15 grams) produced an increase of sensitivity, when our newly developed MTC-DNA detection was applied to animal tissue samples. Both new methods yielded increased detection rates when we analyzed 100 tissue samples from 34 cattle and 18 red deer and turned out to be superior compared to the protocol prescribed by German authorities.</p>
<p>References:</p>
<ul>
<li>Fell et al., 2016; <a href="http://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0816-2">Two alternative DNA extraction methods to improve the detection of Mycobacterium-tuberculosis-complex members in cattle and red deer tissue samples</a> (in <em>BMC Microbiology</em>)</li>
<li>Rettinger et al., 2015: <a href="http://onlinelibrary.wiley.com/doi/10.1111/tbed.12438/abstract">The Region of Difference Four is a Robust Genetic Marker for Subtyping Mycobacterium caprae Isolates and is Linked to Spatial Distribution of Three Subtypes</a> (in <em>Transboundary and Emerging Diseases</em>)</li>
<li>Broeckl et al., 2017: <a href="http://link.springer.com/article/10.1007/s11259-017-9679-8">Investigation of intra-herd spread of Mycobacterium caprae in cattle by generation and use of a whole-genome sequence</a> (in <em>Veterinary Research Communications)</em></li>
</ul>
<p>The post <a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/looking-needle-haystack-find-mycobacterial-dna-animal-tissue-samples/" rel="nofollow">Looking for the needle in a haystack – or how to find mycobacterial DNA in animal tissue samples</a> appeared first on <a href="http://blogs.biomedcentral.com/bmcseriesblog" rel="nofollow">BMC Series blog</a>.</p>
                <p class="text-muted">in <a href="http://blogs.biomedcentral.com/bmcseriesblog">BMC Series blog</a> on March 24, 2017 02:10 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/24/ousted-editor-speaks-not-manipulate-citations/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/24/ousted-editor-speaks-not-manipulate-citations/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/24/ousted-editor-speaks-not-manipulate-citations/&title=Ousted editor speaks: I did not manipulate citations" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/24/ousted-editor-speaks-not-manipulate-citations/">Ousted editor speaks: I did not manipulate citations</a></h4>

<p>Last month, a publisher announced that one of its editors had resigned, following accusations he’d asked authors to cite particular papers, boosting his profile and that of journals where he worked. The publisher declined to name the editor. But when an anonymous report began circulating about the incident, the publisher named the researcher: Artemi Cerdà, […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/24/ousted-editor-speaks-not-manipulate-citations/" rel="nofollow">Ousted editor speaks: I did not manipulate citations</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 24, 2017 01:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/mothers-day-maternal-health/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/mothers-day-maternal-health/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/mothers-day-maternal-health/&title=Mother’s Day and Maternal Health" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/mothers-day-maternal-health/">Mother’s Day and Maternal Health</a></h4>

<img alt="© takasu / Fotolia" class="attachment-thumbnail size-thumbnail wp-post-image" height="110" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Fotolia-ハートの贈り物-from-takasu-1000x-pixel-140x110.jpg" style="display: block; margin-bottom: 5px; clear: both;" width="140" /><p>Mothers form a significant part in the lives of their children, families and society. Improvement of maternal health was one of the eight important aims of the <a href="http://www.un.org/millenniumgoals/">Millennium Development goals (MDGs)</a> (a set of targets established to meet the needs of the world’s poorest by 2015). To build upon and carry on the momentum generated by the MDGs, a post-2015 agenda has been produced which is referred to as the <a href="http://www.un.org/sustainabledevelopment/">Sustainable Development Goals (SDGs)</a>. The SDGs consists of 17 goals to ‘Transform Our World’ by 2030, one of which is related to maternal health.</p>
<p>Maternal mortality has fallen by <a href="http://www.un.org/sustainabledevelopment/health/">almost 50% since 1990</a>, with Eastern Asia, Northern Africa and Southern Asia seeing a decline of around two thirds. Despite the declines reported, maternal mortality is still unacceptably high with 99% of maternal deaths occurring in low and middle income countries.</p>
<figure class="wp-caption alignright" id="attachment_8747" style="width: 310px;"><img alt="Image by Flickr user Terence Nance (CC BY 2.0)" class="wp-image-8747 size-medium" height="199" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Image-by-Flickr-user-Terence-Nance-CC-BY-2.0-300x199.jpg" width="300" /><div class="attachment-image--caption">Image by Flickr user Terence Nance (CC BY 2.0)</div><div class="attachment-image--source"></div></figure>
<p>It is reported that <a href="http://www.who.int/gho/maternal_health/mortality/maternal_mortality_text/en/">around 830 women</a> die from preventable causes related to pregnancy and childbirth each day. The major complications that account for <a href="http://www.who.int/mediacentre/factsheets/fs348/en/">nearly 75%</a> of all maternal deaths include high blood pressure during pregnancy (pre-eclampsia and eclampsia), severe bleeding, infections, complications from delivery and unsafe abortion. Pre-eclampsia occurs in <a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-016-1169-4">2% to 8%</a> of all pregnancies and has been found to be a <a href="http://www.sciencedirect.com/science/article/pii/S0140673610602796">leading cause</a> of maternal and perinatal morbidity and mortality and yet the <a href="http://apps.who.int/iris/bitstream/10665/119627/1/WHO_RHR_14.17_eng.pdf?ua=1">majority of cases</a> could be avoided if women received timely and effective care, delivered according to evidence-based standards.</p>
<p>From 1990-2015 maternal mortality declined from <a href="http://data.unicef.org/topic/maternal-health/maternal-mortality/">385 to 216 deaths per 100,000</a> live births. As part of the SDGs, the target is to further reduce the global maternal mortality ratio to less than 70 per 100,000 live births. Adolescent pregnancy remains a <a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1246-3#CR6">major contributor</a> to maternal and child mortality. Globally, it has been found that complications linked to pregnancy and childbirth are the <a href="http://www.who.int/mediacentre/factsheets/fs345/en/">second leading</a> cause of death, suicide being the first, amongst girls aged 15-19 years old. This risk is highest for those under the age of 15. One of the specific <a href="http://www.un.org/sustainabledevelopment/health/">health SDG targets</a> is to ensure the provision of and universal access to sexual and reproductive health-care services including family planning, education, and the integration of reproductive health in national strategies and programmes. Although it is known that adolescents have an increased risk for poor maternal and infant outcomes, there are a limited number of studies published in the area of adolescent utilisation of critical maternal health services. The reasons for this are not clear although it could potentially be due to challenges with collecting data from this cohort. <a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1246-3">Banke-Thomas and colleagues</a> report the need for more robust evidence on use of maternal health services by adolescents and the need to improving education and removing financial barriers to their access to care.</p>
<figure class="wp-caption alignleft" id="attachment_8748" style="width: 310px;"><img alt="© Zffoto / Fotolia" class="size-medium wp-image-8748" height="222" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Fotolia-sleeping-newborn-baby-on-male-hands-Zffoto-300x222.jpg" width="300" /><div class="attachment-image--caption">© Zffoto / Fotolia</div><div class="attachment-image--source"></div></figure>
<p>In 2015, it was <a href="http://www.who.int/mediacentre/factsheets/fs348/en/">reported</a> that the maternal mortality ratio was 239 per 100,000 live births in low and middle income countries, almost 20 times higher than that of high income countries where the figure is 12 per 100,000 live births. The high number of maternal deaths seen in certain locations reflects the inequities present with regards to access to health services, and highlights the gap between rich and poor. Disparities exist between and within countries as well as between those on high versus low incomes and those living in rural versus urban regions. It is also <a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1239-2">documented</a> that there are disparities in maternal health outcomes for women who are <a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/1471-2393-14-152">immigrants</a>.</p>
<p>Increasing institutional births rates and improvement in the access to comprehensive emergency obstetric care and medical assistance are central approaches to reducing maternal and neonatal deaths globally. Findings by <a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1267-y">Maru and colleagues</a>, showed that individuals opting for home births instead of an institutional delivery service in Nepal, may not necessarily be doing so due to patient preference, but instead it could be as a result of socioeconomic vulnerability to structural barriers.</p>
<p>With the advancements in science, technology and medicine, healthcare solutions to prevent or manage pregnancy related complications are well known thus making most maternal deaths preventable through the provision of timely and effective care.</p>
<figure class="wp-caption alignright" id="attachment_8753" style="width: 310px;"><img alt="© Luis Louro / Fotolia" class="size-medium wp-image-8753" height="207" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Fotolia-New-birth-Luis-Louro-cropped-300x207.jpg" width="300" /><div class="attachment-image--caption">© Luis Louro / Fotolia</div><div class="attachment-image--source"></div></figure>
<p>It is clear that more needs to be done to reduce maternal morbidity and mortality. This includes addressing issues related to barrier to healthcare and medical provision, equal treatment of people regardless of ethnicity, backgrounds and age, as well as ensuring equitable access for the most vulnerable girls and women to health services. Violence and discrimination against women are factors which must also be tackled to ensure maternal mortality is reduced.</p>
<p>So this mother’s day, celebrate motherhood and support improvements to maternal health, for the world would not be the same without the strength of these special women.</p>
<p>If you are interested in reading more on this topic, please visit the <a href="http://bmcpregnancychildbirth.biomedcentral.com/"><em>BMC Pregnancy and Childbirth</em></a> website. The following are a few of the interesting picks and recent publications in the journal:</p>
<p> </p>
<p><strong>Maternal health services</strong></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1282-z">Assessing emergency obstetric and newborn care: can performance indicators capture health system weaknesses?</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1267-y">Impact of the roll out of comprehensive emergency obstetric care on institutional birth rate in rural Nepal</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1239-2">Afghan migrants face more suboptimal care than natives: a maternal near-miss audit study at university hospitals in Tehran, Iran</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1268-x">Healthcare workers’ behaviors and personal determinants associated with providing adequate sexual and reproductive healthcare services in sub-Saharan Africa: a systematic review</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/1471-2393-14-152">Immigrant and non-immigrant women’s experiences of maternity care: a systematic and comparative review of studies in five countries</a></p>
<p> </p>
<p><strong>Maternal health, morbidity and mortality</strong></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-016-1084-8">Measuring maternal mortality using a Reproductive Age Mortality Study (RAMOS)</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1245-4">A narrative synthesis of factors that affect women speaking up about early warning signs and symptoms of pre-eclampsia and responses of healthcare staff</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1264-1">Antepartum complications and perinatal mortality in rural Bangladesh</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-015-0449-8">A qualitative study exploring the determinants of maternal health service uptake in post-conflict Burundi and Northern Uganda</a></p>
<p> </p>
<p><strong>Adolescents</strong></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-017-1246-3">Factors influencing utilisation of maternal health services by adolescent mothers in Low-and middle-income countries: a systematic review</a></p>
<p><a href="http://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-016-1120-8">The causes of maternal mortality in adolescents in low and middle income countries: a systematic review of the literature</a></p>
<p> </p>
<p>The post <a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/24/mothers-day-maternal-health/" rel="nofollow">Mother’s Day and Maternal Health</a> appeared first on <a href="http://blogs.biomedcentral.com/bmcseriesblog" rel="nofollow">BMC Series blog</a>.</p>
                <p class="text-muted">in <a href="http://blogs.biomedcentral.com/bmcseriesblog">BMC Series blog</a> on March 24, 2017 12:12 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feedproxy.google.com/~r/TheNeurocritic/~3/SjN8CJWb_V0/whats-popular-at-cns2017.html" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feedproxy.google.com/~r/TheNeurocritic/~3/SjN8CJWb_V0/whats-popular-at-cns2017.html" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feedproxy.google.com/~r/TheNeurocritic/~3/SjN8CJWb_V0/whats-popular-at-cns2017.html&title=What's Popular at #CNS2017?" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feedproxy.google.com/~r/TheNeurocritic/~3/SjN8CJWb_V0/whats-popular-at-cns2017.html">What's Popular at #CNS2017?</a></h4>

<span style="font-size: large;">Memory <a href="http://neurocritic.blogspot.com/2015/04/trends-in-cognitive-computational-and.html" target="_blank" title="Trends in Cognitive, Computational, and Systems Neuroscience 2015">wins again</a>!</span><br /><br /><br /><div class="separator" style="clear: both; text-align: center;"><a href="https://3.bp.blogspot.com/-GAUboS2TQGc/WNTrMTfNmDI/AAAAAAAAMFc/4xX7uNZq-2QWPMHhTQf-YKLNcq38cHndwCLcB/s1600/wordle%2BCNS%2B2017%2Babstract%2Btitles.png" style="margin-left: 0em; margin-right: 1em;"><img border="0" height="192" src="https://3.bp.blogspot.com/-GAUboS2TQGc/WNTrMTfNmDI/AAAAAAAAMFc/4xX7uNZq-2QWPMHhTQf-YKLNcq38cHndwCLcB/s400/wordle%2BCNS%2B2017%2Babstract%2Btitles.png" width="420" /></a></div><br /><div style="text-align: center;"><b><span><a href="http://www.wordle.net/" target="_blank" title="Wordle">Word cloud</a> for 835 <a href="http://www.cogneurosociety.org/mycns/?page=poster_sessions" target="_blank" title="Poster Sessions">poster titles</a> at CNS 2017.</span></b></div><br /><br />The 2017 <a href="https://www.cogneurosociety.org/annual-meeting/" target="_blank">Cognitive Neuroscience Society annual meeting</a> will start tomorrow, March 25. To no one's surprise, memory is the most popular topic in the bottom-up abstract submission sweepstakes.<br /><br />In contrast, the top-down selections of the <a href="https://www.cogneurosociety.org/people/" target="_blank">Cognosenti</a> are light on memory, with a greater emphasis on attention, speech, mind-wandering, and reward.<br /><br /><br /><div class="separator" style="clear: both; text-align: center;"><a href="https://3.bp.blogspot.com/-NBgug0sDPXo/WNTuWbemjSI/AAAAAAAAMFo/K-fikN__iPcdalGkb00oyf2GuGrs4D8EQCLcB/s1600/wordle%2BCNS%2B2017%2Binvited%2Bsymposia.png" style="margin-left: 0em; margin-right: 1em;"><img border="0" height="210" src="https://3.bp.blogspot.com/-NBgug0sDPXo/WNTuWbemjSI/AAAAAAAAMFo/K-fikN__iPcdalGkb00oyf2GuGrs4D8EQCLcB/s400/wordle%2BCNS%2B2017%2Binvited%2Bsymposia.png" width="420" /></a></div><br /><div style="text-align: center;"><b><span><a href="http://www.wordle.net/" target="_blank" title="Wordle">Word cloud</a> for 16 titles/abstracts in four <a href="https://www.cogneurosociety.org/mycns/?page=invited_symposia" target="_blank">Invited Symposia</a>.</span></b></div><br /><br />The member-generated <a href="https://www.cogneurosociety.org/mycns/?page=symposium_sessions" target="_blank">Symposium Sessions</a> are once again memory-centric, but with the key additions of speech, learning, information, and oscillations.<br /><br /><br /><div class="separator" style="clear: both; text-align: center;"><a href="https://3.bp.blogspot.com/-XiEMZ9tSuaw/WNTz4Eqp6AI/AAAAAAAAMF4/s2gS81FZdmY89AxRqaeuyylGyjvW09_AQCLcB/s1600/wordle%2BCNS%2B2017%2Bsymposia.png" style="margin-left: 0em; margin-right: 1em;"><img border="0" height="197" src="https://3.bp.blogspot.com/-XiEMZ9tSuaw/WNTz4Eqp6AI/AAAAAAAAMF4/s2gS81FZdmY89AxRqaeuyylGyjvW09_AQCLcB/s400/wordle%2BCNS%2B2017%2Bsymposia.png" width="420" /></a></div><br /><div style="text-align: center;"><b><span><a href="http://www.wordle.net/" target="_blank" title="Wordle">Word cloud</a> for 43 titles/abstracts in nine <a href="https://www.cogneurosociety.org/mycns/?page=symposium_sessions" target="_blank">Uninvited Symposia</a>.</span></b></div><br /><br />The hot area of the brain this year is <a href="https://www.cogneurosociety.org/mycns/?page=symposium_sessions#5" target="_blank" title="Cognitive maps in the orbitofrontal cortex for goal-directed behavior">OFC</a>, the <a href="https://www.cogneurosociety.org/mycns/?page=invited_symposia#1" target="_blank" title="From knowledge to action: the role of the primate orbitofrontal cortex ">orbitofrontal cortex</a>.<br /><br />Kicking off the meeting is a new addition to the program, a symposium on <a href="https://www.cogneurosociety.org/big-ideas/" target="_blank">Big Ideas in Cognitive Neuroscience</a>, which will focus on language, motor control/action, and (you guessed it) memory:<br /><br /><blockquote class="tr_bq"><i>Six speakers, in three pairs, will consider some major challenges and  cutting-edge advances, from molecular mechanisms to decoding approaches  to network computations. The presentations and debate aim to provide a  tentative outline of what might be a productive and ambitious agenda for  our fields.</i><br /><br /><b>Speakers:</b> <br /><ul><li>Charles R. Gallistel (Rutgers University) and Tomás Ryan (Trinity College Dublin &amp; MIT) on memory.</li><li>Angela Friederici (Max-Planck-Institute) and Jean-Rémi King (NYU) on language.</li><li>John Krakauer (Johns Hopkins University) and Danielle Bassett (University of Pennsylvania) on action/motor.</li></ul></blockquote><br /><div class="feedflare">
<a href="http://feeds.feedburner.com/~ff/TheNeurocritic?a=SjN8CJWb_V0:ZDS566gnAig:yIl2AUoC8zA"><img border="0" src="http://feeds.feedburner.com/~ff/TheNeurocritic?d=yIl2AUoC8zA" /></a> <a href="http://feeds.feedburner.com/~ff/TheNeurocritic?a=SjN8CJWb_V0:ZDS566gnAig:JUhcmGiK9AQ"><img border="0" src="http://feeds.feedburner.com/~ff/TheNeurocritic?d=JUhcmGiK9AQ" /></a> <a href="http://feeds.feedburner.com/~ff/TheNeurocritic?a=SjN8CJWb_V0:ZDS566gnAig:qj6IDK7rITs"><img border="0" src="http://feeds.feedburner.com/~ff/TheNeurocritic?d=qj6IDK7rITs" /></a> <a href="http://feeds.feedburner.com/~ff/TheNeurocritic?a=SjN8CJWb_V0:ZDS566gnAig:cGdyc7Q-1BI"><img border="0" src="http://feeds.feedburner.com/~ff/TheNeurocritic?d=cGdyc7Q-1BI" /></a> <a href="http://feeds.feedburner.com/~ff/TheNeurocritic?a=SjN8CJWb_V0:ZDS566gnAig:gIN9vFwOqvQ"><img border="0" src="http://feeds.feedburner.com/~ff/TheNeurocritic?i=SjN8CJWb_V0:ZDS566gnAig:gIN9vFwOqvQ" /></a> <a href="http://feeds.feedburner.com/~ff/TheNeurocritic?a=SjN8CJWb_V0:ZDS566gnAig:F7zBnMyn0Lo"><img border="0" src="http://feeds.feedburner.com/~ff/TheNeurocritic?i=SjN8CJWb_V0:ZDS566gnAig:F7zBnMyn0Lo" /></a>
</div><img alt="" height="1" src="http://feeds.feedburner.com/~r/TheNeurocritic/~4/SjN8CJWb_V0" width="1" />
                <p class="text-muted">in <a href="http://neurocritic.blogspot.com/">The Neurocritic</a> on March 24, 2017 11:11 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://biocomputation.herts.ac.uk/2017/03/24/mining-hubble-space-telescope-images.html" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://biocomputation.herts.ac.uk/2017/03/24/mining-hubble-space-telescope-images.html" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://biocomputation.herts.ac.uk/2017/03/24/mining-hubble-space-telescope-images.html&title=Mining Hubble Space Telescope images" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://biocomputation.herts.ac.uk/2017/03/24/mining-hubble-space-telescope-images.html">Mining Hubble Space Telescope images</a></h4>

<p>We present a unsupervised machine learning technique to explore large sky surveys produced by imaging telescopes. Distinct from previous approaches this technique requires no pre-selection of target galaxy type; instead it automatically identifies objects that are similar. We apply the technique to the five Hubble Space Telescope CANDELS fields and compare the machine-based classifications to human-classifications from the Galaxy Zoo: CANDELS project. We find that although there is not a direct mapping between Galaxy Zoo and our hierarchical labelling, there is a good level of concordance between human and machine classifications. A catalogue and galaxy similarity search is provided for the CANDELS fields at <a class="reference external" href="http://www.galaxyml.uk/">http://www.galaxyml.uk/</a></p>
<p><strong>Date:</strong> 24/03/2017 <br />
<strong>Time:</strong> 16:00 <br />
<strong>Location</strong>: LB252</p>
                <p class="text-muted">in <a href="http://biocomputation.herts.ac.uk/">UH Biocomputation group</a> on March 24, 2017 08:03 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.oist.jp/news-center/news/2017/3/24/parallel-computation-provides-deeper-insight-brain-function" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.oist.jp/news-center/news/2017/3/24/parallel-computation-provides-deeper-insight-brain-function" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.oist.jp/news-center/news/2017/3/24/parallel-computation-provides-deeper-insight-brain-function&title=Parallel Computation Provides Deeper Insight into Brain Function" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.oist.jp/news-center/news/2017/3/24/parallel-computation-provides-deeper-insight-brain-function">Parallel Computation Provides Deeper Insight into Brain Function</a></h4>

<div class="field field-name-field-image field-type-image field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><a href="https://www.oist.jp/news-center/news/2017/3/24/parallel-computation-provides-deeper-insight-brain-function"><img alt="" class="img-responsive" height="200" src="https://oist-prod-www.s3-ap-northeast-1.amazonaws.com/s3fs-public/styles/news_teaser_image__600x200_/public/20170216-deschutter-unit-STEPS-purkinje-HPI.jpg?itok=SacpXomm" width="600" /></a></div>
      </div>
</div>
<div class="field field-name-title-field field-type-text field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><h3><a href="https://www.oist.jp/news-center/news/2017/3/24/parallel-computation-provides-deeper-insight-brain-function">Parallel Computation Provides Deeper Insight into Brain Function</a></h3>
</div>
      </div>
</div>
<div class="field field-name-body field-type-text-with-summary field-label-hidden">
      <div class="field-items">
                <div class="field-item even">New computational software developed by OIST researchers is hundreds of times faster than conventional tools, opening up new opportunities to understand how individual neurons and networks of neurons function.  </div>
          </div>
  </div>
                <p class="text-muted">in <a href="https://www.oist.jp/groups/computational-neuroscience-unit-erik-de-schutter">OIST Japan - CNU - Eric De Schutter</a> on March 24, 2017 01:10 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.oist.jp/news-center/photos/left-dr-weiliang-chen-dr-iain-hepburn-and-professor-erik-de-schutter-front-sango" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.oist.jp/news-center/photos/left-dr-weiliang-chen-dr-iain-hepburn-and-professor-erik-de-schutter-front-sango" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.oist.jp/news-center/photos/left-dr-weiliang-chen-dr-iain-hepburn-and-professor-erik-de-schutter-front-sango&title=(From left): Dr. Weiliang Chen, Dr. Iain Hepburn and Professor Erik De Schutter in front of the Sango supercomputer at OIST" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.oist.jp/news-center/photos/left-dr-weiliang-chen-dr-iain-hepburn-and-professor-erik-de-schutter-front-sango">(From left): Dr. Weiliang Chen, Dr. Iain Hepburn and Professor Erik De Schutter in front of the Sango supercomputer at OIST</a></h4>

<div class="field field-name-field-image field-type-image field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><a href="https://www.oist.jp/news-center/photos/left-dr-weiliang-chen-dr-iain-hepburn-and-professor-erik-de-schutter-front-sango"><img alt="" class="img-responsive" height="200" src="https://oist-prod-www.s3-ap-northeast-1.amazonaws.com/s3fs-public/styles/news_teaser_image__600x200_/public/photos/20170221-deschutter-sango-parallelSTEPS2.JPG?itok=7jtywpEn" width="600" /></a></div>
      </div>
</div>
<div class="field field-name-title-field field-type-text field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><h3><a href="https://www.oist.jp/news-center/photos/left-dr-weiliang-chen-dr-iain-hepburn-and-professor-erik-de-schutter-front-sango">(From left): Dr. Weiliang Chen, Dr. Iain Hepburn and Professor Erik De Schutter in front of the Sango supercomputer at OIST </a></h3>
</div>
      </div>
</div>
<div class="field field-name-field-date-created field-type-date field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><span class="date-display-single">24 Mar 2017</span></div>
      </div>
</div>
                <p class="text-muted">in <a href="https://www.oist.jp/groups/computational-neuroscience-unit-erik-de-schutter">OIST Japan - CNU - Eric De Schutter</a> on March 24, 2017 01:02 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.oist.jp/news-center/photos/purkinje-cell-model" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.oist.jp/news-center/photos/purkinje-cell-model" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.oist.jp/news-center/photos/purkinje-cell-model&title=Purkinje Cell Model" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.oist.jp/news-center/photos/purkinje-cell-model">Purkinje Cell Model</a></h4>

<div class="field field-name-field-image field-type-image field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><a href="https://www.oist.jp/news-center/photos/purkinje-cell-model"><img alt="" class="img-responsive" height="200" src="https://oist-prod-www.s3-ap-northeast-1.amazonaws.com/s3fs-public/styles/news_teaser_image__600x200_/public/photos/20170324-purkinje-cell-de-schutter-oist.jpg?itok=RUkkITrO" width="600" /></a></div>
      </div>
</div>
<div class="field field-name-title-field field-type-text field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><h3><a href="https://www.oist.jp/news-center/photos/purkinje-cell-model">Purkinje Cell Model</a></h3>
</div>
      </div>
</div>
<div class="field field-name-field-date-created field-type-date field-label-hidden">
    <div class="field-items">
          <div class="field-item even"><span class="date-display-single">24 Mar 2017</span></div>
      </div>
</div>
                <p class="text-muted">in <a href="https://www.oist.jp/groups/computational-neuroscience-unit-erik-de-schutter">OIST Japan - CNU - Eric De Schutter</a> on March 24, 2017 01:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/4_kHEfdxN4o/nature.2017.21703" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/4_kHEfdxN4o/nature.2017.21703" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/4_kHEfdxN4o/nature.2017.21703&title=Big data renews fight over animal origins" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/4_kHEfdxN4o/nature.2017.21703">Big data renews fight over animal origins</a></h4>

<p>Study is the latest in the long-running dispute over which lineage — sponges or comb jellies — is the oldest branch in the animal tree.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21703">doi: 10.1038/nature.2017.21703</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/4_kHEfdxN4o" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 24, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/prtzWPiNHWM/nature.2017.21702" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/prtzWPiNHWM/nature.2017.21702" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/prtzWPiNHWM/nature.2017.21702&title=Citizen scientists to rescue 150 years of cosmic images" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/prtzWPiNHWM/nature.2017.21702">Citizen scientists to rescue 150 years of cosmic images</a></h4>

<p>Long-lost images could offer insight into rare and moving stars.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21702">doi: 10.1038/nature.2017.21702</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/prtzWPiNHWM" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 24, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/deadly-new-zealand-quake-hopscotched-across-faults" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/deadly-new-zealand-quake-hopscotched-across-faults" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/deadly-new-zealand-quake-hopscotched-across-faults&title=Deadly New Zealand quake hopscotched across faults" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/deadly-new-zealand-quake-hopscotched-across-faults">Deadly New Zealand quake hopscotched across faults</a></h4>

The Nov. 14, 2016, earthquake in New Zealand was much larger than thought possible at the time, prompting a rethink of hazard assessments.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 23, 2017 06:38 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/23/agreed-listen-complaint-paper-harassment-began/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/23/agreed-listen-complaint-paper-harassment-began/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/23/agreed-listen-complaint-paper-harassment-began/&title=They agreed to listen to a complaint about a paper. Then the harassment began." target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/23/agreed-listen-complaint-paper-harassment-began/">They agreed to listen to a complaint about a paper. Then the harassment began.</a></h4>

<p>We receive our fair share of tips, and most are well-intentioned attempts to clean up the scientific literature. However, sometimes would-be critics can veer into personal attacks. As chair of the Committee on Publication Ethics, Virginia Barbour has seen a lot. But nothing quite prepared her for being cyberbullied by someone the organisation had agreed to […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/23/agreed-listen-complaint-paper-harassment-began/" rel="nofollow">They agreed to listen to a complaint about a paper. Then the harassment began.</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 23, 2017 06:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/random-mutations-play-large-role-cancer-study-finds" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/random-mutations-play-large-role-cancer-study-finds" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/random-mutations-play-large-role-cancer-study-finds&title=Random mutations play large role in cancer, study finds" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/random-mutations-play-large-role-cancer-study-finds">Random mutations play large role in cancer, study finds</a></h4>

Mistakes made while copying DNA account for more mutations in cancer cells than environment or inheritance do.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 23, 2017 06:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/dengue-fever-spreads-neighborly-way" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/dengue-fever-spreads-neighborly-way" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/dengue-fever-spreads-neighborly-way&title=Dengue fever spreads in a neighborly way" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/dengue-fever-spreads-neighborly-way">Dengue fever spreads in a neighborly way</a></h4>

Individual strains of dengue spread locally, and new infections cluster near the home of the first person affected.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 23, 2017 06:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/blog/science-ticker/arctic-sea-ice-hits-record-wintertime-low" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/blog/science-ticker/arctic-sea-ice-hits-record-wintertime-low" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/blog/science-ticker/arctic-sea-ice-hits-record-wintertime-low&title=Arctic sea ice hits record wintertime low" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/blog/science-ticker/arctic-sea-ice-hits-record-wintertime-low">Arctic sea ice hits record wintertime low</a></h4>

Warm temperatures and heat waves reduced sea ice extent in the Arctic to its smallest maximum extent ever seen.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 23, 2017 04:31 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/23/plant-scientist-logs-4-notices-figure-errors/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/23/plant-scientist-logs-4-notices-figure-errors/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/23/plant-scientist-logs-4-notices-figure-errors/&title=Plant scientist logs 4 more notices over figure errors" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/23/plant-scientist-logs-4-notices-figure-errors/">Plant scientist logs 4 more notices over figure errors</a></h4>

<p>Researchers in China have retracted a paper and corrected three others in a plant journal, citing problems with multiple figures. Lixin Zhang, from the Chinese Academy of Sciences in Beijing, is corresponding author on all four papers, published by Plant Physiology. We’ve reported on three retractions and two corrections from Lixin Zhang and some of the same […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/23/plant-scientist-logs-4-notices-figure-errors/" rel="nofollow">Plant scientist logs 4 more notices over figure errors</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 23, 2017 03:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/23/flippers-fingers-mammalian-limbs-display-diversity/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/23/flippers-fingers-mammalian-limbs-display-diversity/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/23/flippers-fingers-mammalian-limbs-display-diversity/&title=From flippers to fingers: how do mammalian limbs display such diversity?" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/23/flippers-fingers-mammalian-limbs-display-diversity/">From flippers to fingers: how do mammalian limbs display such diversity?</a></h4>

<img alt="Erophylla stages" class="attachment-thumbnail size-thumbnail wp-post-image" height="110" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/FIgure1-Maier-crop-140x110.jpg" style="display: block; margin-bottom: 5px; clear: both;" width="140" /><p>The mammalian limb displays a breadth of morphological diversity: from the single toe of the horse, to the wings of the bat, to the flippers of the dolphin, to our own five digits. Yet, decades of studies on classical vertebrate models (e.g. chickens, mice) have shown that a similar suite of genes drives limb patterning across species. With this, how is such variety of form achieved?</p>
<p>It is now becoming more widely accepted that regulatory changes are a main driver of evolution and diversity. Many excellent studies on this topic have been done with invertebrates and, more recently, mammals.</p>
<p>The limb first begins as a ridge; a small outgrowth from the flank of the body. Next, the limb becomes roughly as wide as it is long (bud stage). The limb then flattens into a paddle, and between these stages the digit rays begin to condense. As limb development proceeds, digits become more prominent and tissue between the digits dies, leaving free digits. Some mammals (bats) will retain this tissue, which forms the wing (see figure below). Other mammals (<a href="http://www.nature.com/nature/journal/v511/n7507/full/nature13496.html">pigs, camels</a>, <a href="http://www.nature.com/nature/journal/v511/n7507/full/nature13289.html">cows</a>, and many others) undergo a variety of mechanisms to reduce the number of digits.</p>
<figure class="wp-caption alignleft" id="attachment_8714" style="width: 1528px;"><img alt="Erophylla sezekorni (Buffy flower bat) embryos from ridge, bud, paddle stages (top row). As development proceeds (bottom row), the digits condense in the developing fore- and hindlimbs. The interdigital tissue of the hindlimb will regress, leaving free toes, while the wing will retain this tissue. " class="size-full wp-image-8714" height="1333" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/FIgure1-Maier-2.jpg" width="1518" /><div class="attachment-image--caption"><em>Erophylla sezekorni</em> (Buffy flower bat) embryos from ridge, bud, paddle stages (top row). As development proceeds (bottom row), the digits condense in the developing fore- and hindlimbs. The interdigital tissue of the hindlimb will regress, leaving free toes, while the wing will retain this tissue.</div><div class="attachment-image--source">Maier et al.</div></figure>
<p>Studies of <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1525-142X.2005.05015.x/abstract">candidate genes</a> and <a href="http://rspb.royalsocietypublishing.org/content/281/1783/20133133">transcriptomes</a> of individual species have broadened our understanding of morphological diversity, however to our knowledge no wide-scale analysis of several species at once has been done on the limb. Such a study would permit us to address questions such as: when does gene expression diverge among the limbs of different species? Which pathways and genes differ most in expression in the fore- and hindlimbs of the same and different species?</p>
<p>To test this, we performed an RNA-seq analysis at the ridge, bud, and paddle stages of several representative species. Mouse (<em>Mus musculus</em>) for mammals with a typical 5-digit pattern, pig (<em>Sus scrofa</em>), which have evolved digit reduction, a species of microbat (<em>Carollia perspicillata</em>) – the only mammal capable of powered flight, and the opossum (<em>Monodelphis domestica</em>), a marsupial for which there is a growth disparity at birth in fore- and hindlimb formation.</p>
<p>We found that several genes known in limb development were differentially expressed in the fore- and hindlimbs of all species, including <em>Tbx5</em>, <em>Pax1</em>, and <em>Hand1</em>. In addition, some other known limb genes (<em>Wnt5a</em>, <em>Tbx4</em>, <em>Hoxa13</em>, others) were greatly divergently expressed in the fore- and hindlimbs of most species and among limbs of all species. We confirmed a few select genes (of the <em>Hoxd</em> and <em>Hoxa</em> families) by whole-mount <em>in situ</em> hybridization (WISH).</p>
<p>In general, we found that the fore- and hindlimbs of all species are most similar at the ridge stage of formation, and they begin to diverge at the onset of cartilage condensation (between the bud and paddle stages). We also calculated the evolutionary age of genes expressed at these stages across species and found that in general, oldest genes were from the ridge stages, while younger genes were at the paddle stage. Thus, our results suggest there is more diversity of gene expression as limbs become more morphologically diverse between species.</p>
<p>This work is a first step to examining regulatory changes that lead to differences in gene expression between diverse species. We were able to determine across stages and species a large number of divergently expressed genes. We are currently examining other genes by WISH to confirm their divergent expression, and investigating the enhancers surrounding our genes of interest.</p>
<p>The post <a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/23/flippers-fingers-mammalian-limbs-display-diversity/" rel="nofollow">From flippers to fingers: how do mammalian limbs display such diversity?</a> appeared first on <a href="http://blogs.biomedcentral.com/bmcseriesblog" rel="nofollow">BMC Series blog</a>.</p>
                <p class="text-muted">in <a href="http://blogs.biomedcentral.com/bmcseriesblog">BMC Series blog</a> on March 23, 2017 02:33 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.elsevier.com/connect/who-are-the-reviewers-of-journal-articles-and-how-often-do-they-review" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.elsevier.com/connect/who-are-the-reviewers-of-journal-articles-and-how-often-do-they-review" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.elsevier.com/connect/who-are-the-reviewers-of-journal-articles-and-how-often-do-they-review&title=Who are the reviewers of journal articles, and how often do they review?" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.elsevier.com/connect/who-are-the-reviewers-of-journal-articles-and-how-often-do-they-review">Who are the reviewers of journal articles, and how often do they review?</a></h4>

Editors of <em>Biological Conservation</em> investigate the characteristics of people invited to review manuscripts
                <p class="text-muted">in <a href="">Elsevier Connect</a> on March 23, 2017 01:54 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/23/oops-patient-misdiagnosed-rare-neurological-side-effect-retracted-case-study/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/23/oops-patient-misdiagnosed-rare-neurological-side-effect-retracted-case-study/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/23/oops-patient-misdiagnosed-rare-neurological-side-effect-retracted-case-study/&title=Patient misdiagnosed with rare neurological side effect in retracted case study" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/23/oops-patient-misdiagnosed-rare-neurological-side-effect-retracted-case-study/">Patient misdiagnosed with rare neurological side effect in retracted case study</a></h4>

<p>When two surgeons in Greece learned that a patient had developed a rare side effect following weight loss surgery, they were eager to publish the case. After extensive testing, the patient was diagnosed with Wernicke’s encephalopathy—a neurological disorder caused by thiamine deficiency—following a sleeve gastrectomy procedure. As the authors note in the paper, they had seen only eight […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/23/oops-patient-misdiagnosed-rare-neurological-side-effect-retracted-case-study/" rel="nofollow">Patient misdiagnosed with rare neurological side effect in retracted case study</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 23, 2017 01:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.elsevier.com/connect/why-gender-is-a-crucial-but-overlooked-factor-in-heart-disease" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.elsevier.com/connect/why-gender-is-a-crucial-but-overlooked-factor-in-heart-disease" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.elsevier.com/connect/why-gender-is-a-crucial-but-overlooked-factor-in-heart-disease&title=Why gender is a crucial — but overlooked — factor in heart disease" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.elsevier.com/connect/why-gender-is-a-crucial-but-overlooked-factor-in-heart-disease">Why gender is a crucial — but overlooked — factor in heart disease</a></h4>

6 problems that plague cardiovascular care for women – and how to apply a gender lens to all healthcare
                <p class="text-muted">in <a href="">Elsevier Connect</a> on March 23, 2017 01:19 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/its-time-redefine-what-qualifies-planet" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/its-time-redefine-what-qualifies-planet" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/its-time-redefine-what-qualifies-planet&title=It’s time to redefine what qualifies as a planet, scientists propose" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/its-time-redefine-what-qualifies-planet">It’s time to redefine what qualifies as a planet, scientists propose</a></h4>

Astronomers can have their definition of a planet, but some planetary scientists plan to stick to the long-held meaning of the word.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 23, 2017 01:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://digest.bps.org.uk/2017/03/23/workplace-venting-makes-it-harder-to-bounce-back-from-bad-experiences/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://digest.bps.org.uk/2017/03/23/workplace-venting-makes-it-harder-to-bounce-back-from-bad-experiences/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://digest.bps.org.uk/2017/03/23/workplace-venting-makes-it-harder-to-bounce-back-from-bad-experiences/&title=Workplace venting makes it harder to bounce back from bad experiences" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://digest.bps.org.uk/2017/03/23/workplace-venting-makes-it-harder-to-bounce-back-from-bad-experiences/">Workplace venting makes it harder to bounce back from bad experiences</a></h4>

<p><em><img data-attachment-id="21391" data-comments-opened="1" data-image-meta="{" data-orig-file="https://researchdigest.files.wordpress.com/2017/03/gettyimages-528707611.jpg?w=845" data-orig-size="2121,1414" data-permalink="https://digest.bps.org.uk/2017/03/23/workplace-venting-makes-it-harder-to-bounce-back-from-bad-experiences/i-dont-have-the-patience-for-this/" />By</em> <a href="http://www.alexfradera.net/">Alex Fradera</a></p>
<p>When you experience frustrations at work – spats with colleagues, or last-minute demands – it’s natural to want to voice your feelings. And surely it’s healthier. After all, better out than in! Not according to <a href="http://www.tandfonline.com/doi/full/10.1080/1359432X.2016.1257610">new evidence</a> in the <em>European Journal of Work and Organizational Psychology</em> that shows complaining about negative events actually cements their impact. The researchers Evangelia Demeroutia and Russell Cropanzano recommend trying instead to meet the slings and arrows of everyday indignity with all the “sportsmanship” you can muster.</p>
<p><span id="more-21378"></span></p>
<p>Demeroutia and Cropanzano, at Eindhoven University of Technology and the University of Colorado at Boulder, asked 112 employed people – from areas like finance, industry and healthcare – to complete diaries for three days in a row. At the end of each day, participants reported how much they had engaged in complaining during that day, how much they’d been focused on what was wrong with the situations they’d been in, and finally, how much they’d tended to make mountains out of molehills. Low scores across these items meant the participant had practiced “good sportsmanship”.</p>
<p>The researchers also asked participants to record in their dairies a single negative event they’d experienced during each day, and to rate its severity. Additionally, participants rated how much they felt a variety of moods during the day, like energetic or inspired, and their engagement with work through items like “Today, I felt proud of the work I did.”</p>
<p>When sportsmanship was low, worse negative events took a greater toll on participants – they not only reported lower momentary mood and less satisfaction and pride with the work they’d been doing that same day, but they also tended to experience lower mood the next morning, measured in a separate diary entry, and lower pride in next-day accomplishments.</p>
<p>But when sportsmanship was high – meaning that participants hadn’t complained, escalated minor issues, or stewed over things too much – bad events, even if rated as severe, didn’t impact mood or work engagement, that day or the next. Demeroutia and Cropanzano think there may be two reasons for this. Firstly, revisiting the event gives it a second wind, further reinforcing the association between it and the normally transient negative emotions that were initially provoked, turning a bad experience into That Bad Experience. Secondly, if complaints are poorly expressed or directed at the wrong person, they can exacerbate the situation, and that’s all too possible when you are still caught up in a drama.</p>
<p>The researchers stress that they are not asking people to refrain from talking about bad things. When a problem keeps manifesting in an organisation or relationship you need to resolve it, and that begins by putting it into words. But purposeless complaining can just as easily be a way to avoid moving on, the out-loud version of <a href="http://bps-occupational-digest.blogspot.de/2012/10/ruminative-thoughts-deepen-long-term.html">mental rumination</a> keeping us in its undertow. Demeroutia and Cropanzano point to more constructive methods like <a href="http://https://www.psychologytoday.com/blog/write-yourself-well/201208/expressive-writing">expressive writing</a>, which have an evidence base showing success in making sense of negative experience. This form of reflection, or attentive conversation focused on straightening out a knotty problem, are vastly preferred to unconstructive venting.</p>
<p>—<a href="http://www.tandfonline.com/doi/full/10.1080/1359432X.2016.1257610">The buffering role of sportsmanship on the effects of daily negative events</a></p>
<p>Image via <a href="http://www.gettyimages.co.uk/">gettyimages.co.uk</a></p>
<p><strong><a href="http://www.alexfradera.net/">Alex Fradera</a> (<a href="https://twitter.com/alexfradera">@alexfradera</a>) is Contributing Writer at <a href="https://digest.bps.org.uk/">BPS Research Digest</a></strong></p><br />  <a href="http://feeds.wordpress.com/1.0/gocomments/researchdigest.wordpress.com/21378/" rel="nofollow"><img alt="" border="0" src="http://feeds.wordpress.com/1.0/comments/researchdigest.wordpress.com/21378/" /></a> <img alt="" border="0" height="1" src="https://pixel.wp.com/b.gif?host=digest.bps.org.uk&amp;blog=114693443&amp;post=21378&amp;subd=researchdigest&amp;ref=&amp;feed=1" width="1" />
                <p class="text-muted">in <a href="https://digest.bps.org.uk">The British Psychological Society - Research Digest</a> on March 23, 2017 11:20 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/1967-lsd-was-briefly-labeled-breaker-chromosomes" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/1967-lsd-was-briefly-labeled-breaker-chromosomes" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/1967-lsd-was-briefly-labeled-breaker-chromosomes&title=In 1967, LSD was briefly labeled a breaker of chromosomes" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/1967-lsd-was-briefly-labeled-breaker-chromosomes">In 1967, LSD was briefly labeled a breaker of chromosomes</a></h4>

Claims that the hallucinogenic drug damaged DNA were quickly rejected. But questions remain about how LSD works.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 23, 2017 11:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/ofBD-cieN2I/nature.2017.21700" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/ofBD-cieN2I/nature.2017.21700" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/ofBD-cieN2I/nature.2017.21700&title=Gates Foundation announces open-access publishing venture" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/ofBD-cieN2I/nature.2017.21700">Gates Foundation announces open-access publishing venture</a></h4>

<p>Global health charity is latest funder to start its own publishing ‘channel’ — and the European Commission is considering its own service.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21700">doi: 10.1038/nature.2017.21700</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/ofBD-cieN2I" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 23, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/QRwvh6IOfoM/nature.2017.21699" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/QRwvh6IOfoM/nature.2017.21699" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/QRwvh6IOfoM/nature.2017.21699&title=Canada budget falls flat with scientists" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/QRwvh6IOfoM/nature.2017.21699">Canada budget falls flat with scientists</a></h4>

<p>Emphasis on innovation overshadowed by funding freeze for key research councils.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21699">doi: 10.1038/nature.2017.21699</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/QRwvh6IOfoM" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 23, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/zN3MrKL8f0U/nature.2017.21697" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/zN3MrKL8f0U/nature.2017.21697" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/zN3MrKL8f0U/nature.2017.21697&title=Zika mosquito genome mapped – at last" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/zN3MrKL8f0U/nature.2017.21697">Zika mosquito genome mapped – at last</a></h4>

<p>‘Breakthrough’ technique used to piece together genome sequence of Zika vector,  Aedes aegypti,  10 years after publication of draft sequence.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21697">doi: 10.1038/nature.2017.21697</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/zN3MrKL8f0U" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 23, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/S_eXAU80xRY/nature.2017.21696" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/S_eXAU80xRY/nature.2017.21696" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/S_eXAU80xRY/nature.2017.21696&title=DNA typos to blame for most cancer mutations" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/S_eXAU80xRY/nature.2017.21696">DNA typos to blame for most cancer mutations</a></h4>

<p>Environment and heredity might not contribute as much to cancer risk as researchers thought.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21696">doi: 10.1038/nature.2017.21696</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/S_eXAU80xRY" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 23, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/female-guppies-bigger-brains-pick-more-attractive-guys" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/female-guppies-bigger-brains-pick-more-attractive-guys" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/female-guppies-bigger-brains-pick-more-attractive-guys&title=Female guppies with bigger brains pick more attractive guys" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/female-guppies-bigger-brains-pick-more-attractive-guys">Female guppies with bigger brains pick more attractive guys</a></h4>

A larger-brained female guppy may pick primo males, but all that mental machinery costs her in other ways.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 07:54 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://blog.scienceexchange.com/2017/03/meet-service-providers-with-adc-antibody-drug-conjugate-expertise/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://blog.scienceexchange.com/2017/03/meet-service-providers-with-adc-antibody-drug-conjugate-expertise/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://blog.scienceexchange.com/2017/03/meet-service-providers-with-adc-antibody-drug-conjugate-expertise/&title=Meet Service Providers with ADC (Antibody-Drug Conjugate) Expertise" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://blog.scienceexchange.com/2017/03/meet-service-providers-with-adc-antibody-drug-conjugate-expertise/">Meet Service Providers with ADC (Antibody-Drug Conjugate) Expertise</a></h4>

<div class="wp-caption alignright" id="attachment_1040" style="width: 379px;"><img alt="" class="wp-image-1040" height="181" src="http://thebenchapp.s3.amazonaws.com/wp-content/uploads/2017/03/Emtansine_mab_structure_coloured-300x147.png" width="369" /><p class="wp-caption-text">Chemical structure of “emtansine” (mertansine plus linker) linked to a monoclonal antibody (maytansine black, mertansine modification red, linker blue.</p></div>
<p><span style="font-weight: 400;">The excitement around ADCs for treating cancer stems from the realization that traditional, small-molecule cytotoxic drugs and radiation are still some of the most potent anticancer agents, and that targeting them by tethering them to antibodies might bypass some of the side effects. </span></p>
<p>The recent founding <a href="http://www.businesswire.com/news/home/20170223005390/en/Fusion-Pharmaceuticals-Launches-25M-USD-Series-Financing">investment by Johnson &amp; Johnson of the ADC-focused startup, Fusion Pharmaceuticals</a>, may provide some momentum to the development of targeted radiotherapeutics, a specific type of ADC. Johnson &amp; Johnson’s investment followed on the heels of recent ADC investments by other large pharmas, including <a href="http://www.b3cnewswire.com/201611031480/nbe-therapeutics-raises-chf-20-million-in-series-b-funding.html">Boehringer Ingelheim</a> and <a href="http://www.fiercebiotech.com/biotech/astrazeneca-backs-adc-105m-to-support-6-clinical-oncology-candidates">AstraZeneca</a>. <a href="https://globenewswire.com/news-release/2017/01/18/906673/0/en/Immunomedics-Announces-New-Data-for-IMMU-132-at-Investor-R-D-Day.html">IMMU-132</a>, the ADC being developed by Immunomedics, and <a href="https://clinicaltrials.gov/ct2/show/NCT01969643">SGN-LIV1A</a> from Seattle Genetics are ADCs that are reported to target triple-negative breast cancer.</p>
<p>The biopharmaceutical industry currently has over two dozen <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3929452/">ADC candidates estimated to be in its pipeline</a>, indicating that there is a current demand for scientists with specific expertise in the techniques required for designing, synthesizing, and studying these molecules. In addition to developing expertise in-house, companies are frequently partnering with smaller companies or outsourcing projects to service providers to get the work done.</p>
<p>At Science Exchange, we have a unique bird’s-eye view of ADC-focused research and the service providers that are facilitating progress in this exciting field. Researchers who order services using the Science Exchange marketplace gain rapid access to an innovative network of 3,000+ service providers, including a number that supports ADC studies, through a single contract with Science Exchange. In this blog post, we’ll go through some of the key techniques and show how featured service providers in our network are meeting needs of ADC researchers.</p>
<h2>Key techniques for studying ADCs: Science Exchange service providers step up.</h2>
<h3></h3>
<h3><span style="color: #338dc8;">Target discovery: the abundance challenge</span></h3>
<p>Many ADC research programs seek to target cell surface proteins that are unique to the cell type that is to be killed by the cytotoxin. However, cell surface proteins, and other ADC targets, are usually low in abundance and underrepresented in traditional proteomic measurements.</p>
<p>One of the newest, cutting-edge providers on the Science Exchange network is <a href="https://www.scienceexchange.com/labs/biognosys-zurich">Biognosys</a>, offering discovery proteomics solutions based on Hyper Reaction Monitoring (HRM-MS™), a Next Generation proteomics technology. Invented at Biognosys, HRM-MS delivers quantification of up to 9’000 proteins per sample across treatments or conditions and identifies significantly regulated proteins. This platform is ideal for ADC target discovery studies, with one proof-of-concept study showing the quantification of over 500 cell surface proteins from matched biopsy samples.</p>
<h3><span style="color: #338dc8;">The conjugation challenge</span></h3>
<p><span style="color: #333333;">The ideal linker between the antibody and the </span>cytotoxin<span style="color: #333333;"> drug is stable in the bloodstream, and if needed, can be cleaved in the specific environment of the target. Some ADC linkers are designed to dissolve the </span>the reducing<span style="color: #333333;"> environment of the cytosol, while others require specific enzymes of certain subcellular compartments. Other linkers are non-cleavable. The linker also has to have minimal toxicity.</span></p>
<p>In addition to some wizardry in chemical synthesis, ADC development therefore requires experience in cell-based assays and drug metabolism studies. <a href="https://www.scienceexchange.com/labs/wuxi-apptec">WuXi Apptec</a>, <a href="https://www.scienceexchange.com/labs/mabplex">MabPlex</a>, and <a href="https://www.scienceexchange.com/labs/chempartner">ChemPartner</a> are service providers listed on the Science Exchange marketplace that have worked side by side with ADC developers on all aspects of linker synthesis and characterization.</p>
<p>Conjugation-related services that <a href="https://www.scienceexchange.com/labs/wuxi-apptec">WuXi Apptec</a>, <a href="https://www.scienceexchange.com/labs/chempartner">ChemPartner</a> and <a href="https://www.scienceexchange.com/labs/mabplex">MabPlex</a> provide include but are not limited to:</p>
<ul>
<li>Cytotoxin development</li>
<li>Linker development</li>
<li>Linker and cytotoxin conjugations</li>
<li>Drug linking site determination</li>
<li>Stability studies for ADC products</li>
</ul>
<p> </p>
<h3><span style="color: #338dc8;">Bioanalysis in ADC development: the heterogeneity challenge</span></h3>
<p>Unlike other categories of drug molecules, ADCs can be structurally heterogeneous, because of dynamic drug:antibody ratios (DAR) and variations in linker attachment chemistry. In a recent survey, <a href="https://www.bioanalysis-zone.com/2015/08/19/commentary-on-bioanalysis-of-adcs-survey-results/">69% of researchers cited this structural heterogeneity as the #1 challenge</a> facing bioanalysis in ADC development.</p>
<p>85% of the surveyed researchers reported using LC-MS for ADC bioanalysis. However, over 24% respondents had to adapt traditional LC-MS methods, using affinity capture LC-MS or accelerator MS. 42% of respondents reported using ligand-binding assays, illustrating that most researchers use more than one technique in analyzing ADCs. The complexity of analytes, in combination with the lack of regulatory guidance around ADC analysis, have resulted in the need to use multiple, individually developed, methods.</p>
<p>Fortunately, the Science Exchange marketplace features the services of <a href="https://www.scienceexchange.com/labs/biognosys-zurich">Biognosys</a>, <a href="https://www.scienceexchange.com/labs/anaquant">Anaquant</a>, and <a href="https://www.scienceexchange.com/labs/chempartner">ChemPartner</a>, all of which provide experience in developing analytical methods for ADCs.</p>
<p>To address the challenge of quantifying multiple species per sample, <a href="https://www.scienceexchange.com/labs/biognosys-zurich">Biognosys</a> provides targeted proteomics services using Multiple and Parallel Reaction Monitoring (MRM and PRM), which are techniques that offer highly specific and sensitive multiplexed quantification of selected proteins from complex biological samples. These techniques deliver absolute or relative quantification of up to 150 target proteins per run with a dynamic range of 6 orders of magnitude.</p>
<p>Countless other service providers, such as <a href="https://www.scienceexchange.com/labs/bio-synthesis">Bio-Synthesis</a>, <a href="https://www.scienceexchange.com/labs/bionova-scientific-inc">Bionova</a>, and <a href="https://www.scienceexchange.com/labs/maine-biotechnology-services">Maine Biotechnology Services</a>, are experts in analyzing ADCs using ligand-binding assays. In addition, Science Exchange’s in-house regulatory compliance team has expertise in working with representatives from regulatory agencies, to ensure that the analytical services carried out by our service providers meet necessary requirements.<br />
ADC bioanalysis services on the Science Exchange marketplace include:</p>
<ul>
<li>DAR (drug:antibody ratio) determination</li>
<li>Residual free drug analysis</li>
<li>Pharmacokinetics (PK) determination</li>
</ul>
<p> </p>
<h3><span style="color: #338dc8;">Process development for ADC</span></h3>
<p>Again, the heterogeneity of a batch of ADC can make it challenging to develop a scalable, reproducible, and robust manufacturing process. Manufacturing the antibody component of the ADC faces all the same challenges as does traditional therapeutic mAb production.</p>
<p>Given the demands of manufacturing, engineering quality by design is important in the nonclinical, preclinical and early clinical phases of ADC research. Expertise in antibody optimization, protein purification, and chemical synthesis are required to create less heterogeneous batches of antibodies, linkers, cytotoxins and conjugates.</p>
<p>Science Exchange service providers <a href="https://www.scienceexchange.com/labs/wuxi-apptec">WuXi Apptec</a>, <a href="https://www.scienceexchange.com/labs/mabplex">MabPlex</a> and <a href="https://www.scienceexchange.com/labs/chempartner">ChemPartner</a> all support ADC process development, with MabPlex’s services extending to GMP and scale-up (to kilogram scale).</p>
<p><a href="https://www.scienceexchange.com/products/marketplace">Browse our marketplace for ADC-related services</a> or contact our <a href="https://www.scienceexchange.com/products/concierge">Concierge Service</a>, who can match your project needs with the right service provider or a combination of service providers to move your ADC research forward.</p>
                <p class="text-muted">in <a href="https://blog.scienceexchange.com">The Science Exchange Blog</a> on March 22, 2017 07:35 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/anatomy-analysis-suggests-new-dinosaur-family-tree" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/anatomy-analysis-suggests-new-dinosaur-family-tree" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/anatomy-analysis-suggests-new-dinosaur-family-tree&title=Anatomy analysis suggests new dinosaur family tree" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/anatomy-analysis-suggests-new-dinosaur-family-tree">Anatomy analysis suggests new dinosaur family tree</a></h4>

A new analysis rewrites the dinosaur family tree, splitting up long-recognized groups.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 06:06 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/22/latest-sting-will-predatory-journals-hire-fake-editor-dr-fraud-answer-yes/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/22/latest-sting-will-predatory-journals-hire-fake-editor-dr-fraud-answer-yes/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/22/latest-sting-will-predatory-journals-hire-fake-editor-dr-fraud-answer-yes/&title=The latest sting: Will predatory journals hire “Dr. Fraud”?" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/22/latest-sting-will-predatory-journals-hire-fake-editor-dr-fraud-answer-yes/">The latest sting: Will predatory journals hire “Dr. Fraud”?</a></h4>

<p>From time to time, academics will devise a “sting” operation, designed to expose journals’ weaknesses. We’ve seen scientists submit a duplicated paper, a deeply flawed weight loss paper designed to generate splashy headlines (it worked), and an entirely fake paper – where even the author calls it a “pile of dung.” So it wasn’t a […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/22/latest-sting-will-predatory-journals-hire-fake-editor-dr-fraud-answer-yes/" rel="nofollow">The latest sting: Will predatory journals hire “Dr. Fraud”?</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 22, 2017 06:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.plos.org/~r/plos/Blog/~3/vPmk03UgwyE/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.plos.org/~r/plos/Blog/~3/vPmk03UgwyE/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.plos.org/~r/plos/Blog/~3/vPmk03UgwyE/&title=NYT: Why Trump’s N.I.H. Cuts Should Worry Us, by PLOS Co-Founder Harold Varmus" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.plos.org/~r/plos/Blog/~3/vPmk03UgwyE/">NYT: Why Trump’s N.I.H. Cuts Should Worry Us, by PLOS Co-Founder Harold Varmus</a></h4>

<img alt="22varmus-master768" class="attachment-thumbnail size-thumbnail wp-post-image" height="150" src="http://blogs.plos.org/plos/files/2017/03/22varmus-master768-150x150.jpg" style="display: block; margin-bottom: 5px; clear: both;" width="150" /><article class="category-listing post-8770 post type-post status-draft format-standard entry" id="post-8770">
<div class="entry-content">
<h6>PLOS is pleased to highlight here an important Op-Ed written by PLOS Co-Founder Harold Varmus, published in The New York Times on March 22, 2017.</h6>
<p>Source: <a href="https://www.nytimes.com/2017/03/22/opinion/why-trumps-nih-cuts-should-worry-us.html?_r=2" target="_blank">Why Trump’s N.I.H. Cuts Should Worry Us – The New York Times</a> [Featured image: Thoka Maer]</p>
<p class="story-body-text story-content">Last week I was in London to participate in a scientific symposium. During coffee breaks, many British colleagues asked me and other American visitors to explain the bewildering news that President Trump had announced his intention<a href="https://www.nytimes.com/2017/03/16/climate/trump-budget-science-research.html"> to cut the budget for the National Institutes of Health</a> by 18.3 percent, about $5.8 billion.</p>
<p class="story-body-text story-content">My answer to “What is going on?” did include some consoling reminders. A presidential budget request is a proposal, not a done deal. The actual fiscal year 2018 appropriation for the N.I.H. will be determined by Congress, which has historically provided enthusiastic bipartisan support for biomedical research. Although the N.I.H. has lost a substantial amount of its spending power gradually over the past decade, it has only rarely experienced a sharp decline in actual dollars and never of this magnitude. Furthermore, strong nonpartisan opposition to Mr. Trump’s proposal will come from many quarters, including advocates of research on specific diseases.</p>
<p class="story-body-text story-content">But it would be a mistake to be complacent about the president’s proposal, because it is likely to have real consequences. Yes, some have said that the proposed cut to the N.I.H. will be dead on arrival in Congress. But the president’s budget proposal is still important: The administration’s representatives will need to defend it at hearings, and it could be the starting point for negotiations among appropriators. It is not difficult to imagine a compromise in which the N.I.H. suffers a steep reduction.</p>
<figure class="wp-caption aligncenter" id="attachment_7420" style="width: 492px;"><img alt="DK comment" class="wp-image-7420 size-full" height="236" src="http://blogs.plos.org/plos/files/2017/03/DK-comment.jpg" width="492" />David Knutson, speaking for PLOS, offered this comment on the New York Times Op-Ed by Harold Varmus.</figure>
<p class="story-body-text story-content">To understand just how devastating a cut of less than 20 percent of an agency’s budget would be requires some understanding of how the N.I.H. operates. Very little of its typical annual budget is spent on the agency’s administration: The industrious, underpaid government scientists who manage the funding of the N.I.H.’s research programs consume less than 5 percent of its budget. Only a bit more, about 10 percent, supports the work of government scientists. In sharp contrast, over 80 percent of its resources are devoted to competitively reviewed biomedical research projects, training programs and science centers, affecting nearly every district in the country.</p>
<p class="story-body-text story-content">The N.I.H. awards multiyear grants and contracts, but receives annual appropriations that must be spent that year. This means that at the start of each year most of its dollars are already committed to recipients of awards from prior years. A budget cut of the size that is proposed would effectively prevent the awarding of new grants or the renewal of any that have reached the end of a multiyear commitment. Junior scientists, already struggling in a highly competitive atmosphere, may not get a chance to have an academic career. Senior investigators might need to lay off staff, disrupting research teams and leaving projects unfinished.</p>
<p class="story-body-text story-content" id="story-continues-1">A substantial N.I.H. budget cut would undermine the fiscal stability of universities and medical schools, many of which depend on N.I.H. funding; it would erode America’s leadership in medical research; and it would diminish opportunities to discover new ways to prevent and treat diseases.</p>
<p class="story-body-text story-content"><em>Read the complete <a href="https://www.nytimes.com/2017/03/22/opinion/why-trumps-nih-cuts-should-worry-us.html?_r=2">Op-Ed by Harold Varmus</a> in the New York Times, March 22, 2017</em></p>
<p class="story-body-text story-content"><strong>Harold Varmus, a professor at Weill Cornell Medicine and a co-recipient of the 1989 Nobel Prize in Physiology or Medicine, was the director of the National Institutes of Health from 1993 to 1999 and of the National Cancer Institute from 2010 to 2015. He <a href="https://www.plos.org/history" target="_blank">co-founded the Public Library of Science</a> (PLOS) in 2000 with Patrick O. Brown and Michael B. Eisen.</strong></p>
</div>
</article>
<nav class="navigation post-navigation"></nav>
<img alt="" height="1" src="http://feeds.feedburner.com/~r/plos/Blog/~4/vPmk03UgwyE" width="1" />
                <p class="text-muted">in <a href="http://blogs.plos.org/plos">The Official PLOS Blog</a> on March 22, 2017 05:34 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/blog/growth-curve/touches-early-life-may-make-big-impact-newborn-babies-brains" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/blog/growth-curve/touches-early-life-may-make-big-impact-newborn-babies-brains" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/blog/growth-curve/touches-early-life-may-make-big-impact-newborn-babies-brains&title=Touches early in life may make a big impact on newborn babies’ brains" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/blog/growth-curve/touches-early-life-may-make-big-impact-newborn-babies-brains">Touches early in life may make a big impact on newborn babies’ brains</a></h4>

The type and amount of touches a newborn baby gets in the first days of life may shape later responses to touch perception, a study suggests.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 04:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/lab-tests-aren%E2%80%99t-answer-every-science-question" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/lab-tests-aren%E2%80%99t-answer-every-science-question" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/lab-tests-aren%E2%80%99t-answer-every-science-question&title=Lab tests aren’t the answer for every science question" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/lab-tests-aren%E2%80%99t-answer-every-science-question">Lab tests aren’t the answer for every science question</a></h4>

Acting Editor in Chief Elizabeth Quill discusses the value of observational science.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 04:15 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/readers-question-supernova-physics" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/readers-question-supernova-physics" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/readers-question-supernova-physics&title=Readers question supernova physics" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/readers-question-supernova-physics">Readers question supernova physics</a></h4>

Star-destroying neutrinos, heart-hugging robots and more in reader feedback.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 04:10 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/22/president-taiwan-university-step-amidst-investigation/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/22/president-taiwan-university-step-amidst-investigation/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/22/president-taiwan-university-step-amidst-investigation/&title=President of Taiwan university to step down amidst investigation" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/22/president-taiwan-university-step-amidst-investigation/">President of Taiwan university to step down amidst investigation</a></h4>

<p>The president of a top university in Taiwan has announced he will resign from his post at the end of his first term in June. President Pan-chyr Yang of National Taiwan University (NTU) has opted not to seek a second term as president given two recent investigations — one which concluded last month and one which […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/22/president-taiwan-university-step-amidst-investigation/" rel="nofollow">President of Taiwan university to step down amidst investigation</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 22, 2017 03:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/how-pluto%E2%80%99s-haze-could-explain-its-red-spots" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/how-pluto%E2%80%99s-haze-could-explain-its-red-spots" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/how-pluto%E2%80%99s-haze-could-explain-its-red-spots&title=How Pluto’s haze could explain its red spots" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/how-pluto%E2%80%99s-haze-could-explain-its-red-spots">How Pluto’s haze could explain its red spots</a></h4>

Pluto’s collapsing atmosphere may explain the dwarf planet’s seemingly random ruddy spots.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 01:41 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/22/nightmare-scenario-text-stolen-manuscript-review/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/22/nightmare-scenario-text-stolen-manuscript-review/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/22/nightmare-scenario-text-stolen-manuscript-review/&title=Nightmare scenario: Text stolen from manuscript during review" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/22/nightmare-scenario-text-stolen-manuscript-review/">Nightmare scenario: Text stolen from manuscript during review</a></h4>

<p>A food science journal has retracted a paper over “a breach of reviewer confidentiality,” after editors learned it contained text from an unpublished manuscript — which one of the authors appears to have reviewed for another journal. The publisher and editors-in-chief of the Journal of Food Process Engineering became aware of the breach when the author of […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/22/nightmare-scenario-text-stolen-manuscript-review/" rel="nofollow">Nightmare scenario: Text stolen from manuscript during review</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 22, 2017 01:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://digest.bps.org.uk/2017/03/22/people-with-higher-working-memory-ability-suffer-more-from-brain-freeze/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://digest.bps.org.uk/2017/03/22/people-with-higher-working-memory-ability-suffer-more-from-brain-freeze/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://digest.bps.org.uk/2017/03/22/people-with-higher-working-memory-ability-suffer-more-from-brain-freeze/&title=People with higher working memory ability suffer more from brain freeze" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://digest.bps.org.uk/2017/03/22/people-with-higher-working-memory-ability-suffer-more-from-brain-freeze/">People with higher working memory ability suffer more from brain freeze</a></h4>

<p><em><img data-attachment-id="21364" data-comments-opened="1" data-image-meta="{" data-orig-file="https://researchdigest.files.wordpress.com/2017/03/gettyimages-72323592.jpg?w=845" data-orig-size="5601,4338" data-permalink="https://digest.bps.org.uk/2017/03/22/people-with-higher-working-memory-ability-suffer-more-from-brain-freeze/puzzled-businessman-scratching-his-head/" />By</em> <a href="http://www.psychologywriter.org.uk/">Christian Jarrett</a></p>
<p>Some fortunate people have more “working memory” than others. It’s as if they have an extra pair of hands available for mental juggling; extremely useful for doing arithmetic and similar tasks in your head. These folk with abundant working memory capacity also tend to fare well academically and in their careers. Little surprise that “brain training” games like Lumosity and Cogmed target working memory in pursuit of these knock-on benefits (though <a href="https://digest.bps.org.uk/2016/10/07/brain-training-exercises-just-make-you-better-at-brain-training-exercises/">the evidence</a> that the training brings such benefits is weak).</p>
<p>What is surprising is the discovery a number of years ago that mentally dextrous people with greater working memory capacity seem to be particularly susceptible to “brain freeze” or choking under pressure.</p>
<p>For a <a href="http://www.sciencedirect.com/science/article/pii/S2211368116302091">new study</a> in the<em> Journal of Applied Research in Memory and Cognition</em>, researchers at the University of Chicago and Michigan State University attempted to find out more about why this happens. Their results suggest that actually it’s only a subgroup of high working memory people who have this problem and it’s because of their high distractibility. These high ability chokers or brain freeze victims are “typically reliant on their higher working memory resources for advanced problem solving” but their poor attentional control renders them easily distracted by anxiety, causing their usual mental deftness to break down when the pressure is on.</p>
<p><span id="more-21226"></span></p>
<p>Jason Sattizahn and his colleagues recruited 83 participants aged 18-35, including 35 men, from areas near universities. First they tested the participants’ attention control abilities using the well-established Flanker Task: participants had to identify as quickly as possible the direction of central arrows that were either surrounded by congruent flankers (&lt;&lt;&lt;&lt;&lt;) or incongruent flankers (&lt;&lt;&gt;&lt;&lt;). It’s a test of attention control because it takes mental concentration to ignore the flankers, especially on incongruent trials.</p>
<p>Next, the researchers tested the participants’ on some tricky mental arithmetic questions, both without any pressure and then under high pressure conditions in which there was a monetary incentive, peer pressure (poor performance would adversely affect another participant) and risk of public shaming (they were told their performance would be shared with professors and others).</p>
<p>Finally, the participants completed two tests of their working memory capacity: they had to solve a sequence of basic mathematical operations or sentence comprehension questions, with each one interspersed with presentation of a single letter on screen. At the end of each run of between three and seven trials, the participants had to try to recall the letters in the correct order.</p>
<p>The results were clear: high stakes pressure adversely affected the maths performance of participants with high working memory capacity, but did not affect the performance of participants with lower working memory capacity (replicating a phenomenon first identified in a paper published in 2005; <a href="https://hpl.uchicago.edu/sites/hpl.uchicago.edu/files/uploads/Psychological%20Science2005.pdf">pdf</a>). What’s new here is the additional measure of attentional control. This showed that only high working memory participants with poor attentional control showed evidence of this mental performance choking under pressure.</p>
<p>What seems to be happening is that, under low pressure conditions, people with high working memory capacity make use of their extra pair of mental hands when performing arithmetic and similar tasks, deploying more sophisticated and demanding strategies that usually allow them to excel. When they are distracted by pressure – which is especially likely if they have poor attentional control – people with high working memory still try to rely on their usual cognitively demanding approach, but now their extra mental hands are tied, so they drop the balls, so to speak.</p>
<p>If this sounds like you – perhaps your brain works like lighting on your own but goes into meltdown when someone puts you on the spot at work – the good news is that the researchers have a few suggestions for how to protect yourself from choking under pressure. Sattizahn and his colleagues suggest two strategies: either reduce the distracting anxiety you experience under pressure, for example by using expressive writing to reflect on a challenge that you know is coming up; or boost your attention control skills, for instance by practising mindfulness or simply remembering to take time out to walk in nature, which is known to refresh our powers of concentration.</p>
<p>I propose a third, lazier approach: next time you suffer a potentially embarrassing bout of brain freeze in front of friends or colleagues, simply point them to this article and explain: thanks to your impressive working memory capacity you can be a little sensitive at times, would they please back off and allow your shy super brain the space that it needs.</p>
<p>—<a href="http://www.sciencedirect.com/science/article/pii/S2211368116302091">A closer look at who chokes under pressure</a></p>
<p><span class="s1"><a href="http://www.psychologywriter.org.uk/"><b>Christian Jarrett</b></a></span><span class="s2"><b> (</b><a href="https://twitter.com/psych_Writer"><span class="s3"><b>@Psych_Writer</b></span></a><b>) is Editor of </b><span class="s3"><b><a href="https://digest.bps.org.uk/">BPS Research Digest</a></b></span></span></p><br />  <a href="http://feeds.wordpress.com/1.0/gocomments/researchdigest.wordpress.com/21226/" rel="nofollow"><img alt="" border="0" src="http://feeds.wordpress.com/1.0/comments/researchdigest.wordpress.com/21226/" /></a> <img alt="" border="0" height="1" src="https://pixel.wp.com/b.gif?host=digest.bps.org.uk&amp;blog=114693443&amp;post=21226&amp;subd=researchdigest&amp;ref=&amp;feed=1" width="1" />
                <p class="text-muted">in <a href="https://digest.bps.org.uk">The British Psychological Society - Research Digest</a> on March 22, 2017 11:06 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/colorful-pinwheel-puts-new-spin-mouse-pregnancy" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/colorful-pinwheel-puts-new-spin-mouse-pregnancy" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/colorful-pinwheel-puts-new-spin-mouse-pregnancy&title=Colorful pinwheel puts a new spin on mouse pregnancy" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/colorful-pinwheel-puts-new-spin-mouse-pregnancy">Colorful pinwheel puts a new spin on mouse pregnancy</a></h4>

Among the winners of the 2017 Wellcome Image Awards is a rainbow of mouse placentas that shows how a mother’s immune system affects placental development.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 11:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/blog/science-ticker/sea-otters-use-rocks-tools-arent-related" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/blog/science-ticker/sea-otters-use-rocks-tools-arent-related" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/blog/science-ticker/sea-otters-use-rocks-tools-arent-related&title=Tool use in sea otters doesn't run in the family" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/blog/science-ticker/sea-otters-use-rocks-tools-arent-related">Tool use in sea otters doesn't run in the family</a></h4>

A genetic study suggests that tool-use behavior isn’t hereditary in sea otters, and that only some animals need to use tools due to the type of food available in their ecosystem.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 22, 2017 12:44 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/JkRaEDSrIJM/nature.2017.21681" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/JkRaEDSrIJM/nature.2017.21681" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/JkRaEDSrIJM/nature.2017.21681&title=Dinosaur family tree poised for colossal shake-up" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/JkRaEDSrIJM/nature.2017.21681">Dinosaur family tree poised for colossal shake-up</a></h4>

<p>'Textbook-changing' analysis of dinosaur bones upends long-accepted relationships among major groups.</p><p>Nature News   <a href="http://www.nature.com/doifinder/10.1038/nature.2017.21681">doi: 10.1038/nature.2017.21681</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/JkRaEDSrIJM" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/cL7M9XwWWbY/543481a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/cL7M9XwWWbY/543481a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/cL7M9XwWWbY/543481a&title=Predatory journals recruit fake editor" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/cL7M9XwWWbY/543481a">Predatory journals recruit fake editor</a></h4>

<p>An investigation finds that dozens of academic titles offered 'Dr Fraud' — a sham, unqualified scientist — a place on their editorial board. Katarzyna Pisanski and colleagues report.</p><p>Nature 543 481 <a href="http://www.nature.com/doifinder/10.1038/543481a">doi: 10.1038/543481a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/cL7M9XwWWbY" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/LYbUJ0fZA9U/543478a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/LYbUJ0fZA9U/543478a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/LYbUJ0fZA9U/543478a&title=How to hunt for a black hole with a telescope the size of Earth" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/LYbUJ0fZA9U/543478a">How to hunt for a black hole with a telescope the size of Earth</a></h4>

<p>Astronomers hope to grab the first images of an event horizon — the point of no return.</p><p>Nature 543 478 <a href="http://www.nature.com/doifinder/10.1038/543478a">doi: 10.1038/543478a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/LYbUJ0fZA9U" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/EXb5MR8QIZY/543474a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/EXb5MR8QIZY/543474a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/EXb5MR8QIZY/543474a&title=South Korea’s scientists seek change amid political chaos" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/EXb5MR8QIZY/543474a">South Korea’s scientists seek change amid political chaos</a></h4>

<p>President’s impeachment creates opportunity to shift how nation supports basic research.</p><p>Nature 543 474 <a href="http://www.nature.com/doifinder/10.1038/543474a">doi: 10.1038/543474a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/EXb5MR8QIZY" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/hhQqdoALFBE/543468a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/hhQqdoALFBE/543468a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/hhQqdoALFBE/543468a&title=Nuclear-test films, smoking declines and five new particles" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/hhQqdoALFBE/543468a">Nuclear-test films, smoking declines and five new particles</a></h4>

<p>The week in science: 17–23 March 2017.</p><p>Nature 543 468 <a href="http://www.nature.com/doifinder/10.1038/543468a">doi: 10.1038/543468a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/hhQqdoALFBE" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/EgyxbU7tDXE/543464a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/EgyxbU7tDXE/543464a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/EgyxbU7tDXE/543464a&title=Birds of play demonstrate the infectious power of emotion" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/EgyxbU7tDXE/543464a">Birds of play demonstrate the infectious power of emotion</a></h4>

<p>New Zealand parrots provide the latest support for a popular theory of crowd behaviour.</p><p>Nature 543 464 <a href="http://www.nature.com/doifinder/10.1038/543464a">doi: 10.1038/543464a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/EgyxbU7tDXE" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/yqWF1KxQmaM/nature21901" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/yqWF1KxQmaM/nature21901" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/yqWF1KxQmaM/nature21901&title=Biochemistry: A wine-induced breakdown" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/yqWF1KxQmaM/nature21901">Biochemistry: A wine-induced breakdown</a></h4>

<p>A polysaccharide called rhamnogalacturonan II is a major component of some fruits, but humans rely on their gut microbiota to digest it. The microbes and processes responsible for this digestion have now been revealed.</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21901">doi: 10.1038/nature21901</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/yqWF1KxQmaM" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/GS9y01bI1Ec/nature21900" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/GS9y01bI1Ec/nature21900" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/GS9y01bI1Ec/nature21900&title=Structural biology: Receptors grease the metabolic wheels" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/GS9y01bI1Ec/nature21900">Structural biology: Receptors grease the metabolic wheels</a></h4>

<p>Structural insights into adiponectin receptors provide evidence that these proteins have an inherent enzymatic activity, which gives them the ability to propagate signalling by their ligand, the hormone adiponectin.</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21900">doi: 10.1038/nature21900</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/GS9y01bI1Ec" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/jF1aDYWmFwc/nature21895" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/jF1aDYWmFwc/nature21895" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/jF1aDYWmFwc/nature21895&title=Marine conservation: How to heal an ocean" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/jF1aDYWmFwc/nature21895">Marine conservation: How to heal an ocean</a></h4>

<p>Marine protected areas are being implemented at an accelerating pace, and hold promise for restoring damaged ecosystems. But glaring shortfalls in staffing and funding often lead to suboptimal outcomes.</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21895">doi: 10.1038/nature21895</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/jF1aDYWmFwc" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/iJ7vOeD8jbo/nature21725" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/iJ7vOeD8jbo/nature21725" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/iJ7vOeD8jbo/nature21725&title=Complex pectin metabolism by gut bacteria reveals novel catalytic functions" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/iJ7vOeD8jbo/nature21725">Complex pectin metabolism by gut bacteria reveals novel catalytic functions</a></h4>

<p>
 The metabolism of carbohydrate polymers drives microbial diversity in the human gut microbiota. It is unclear, however, whether bacterial consortia or single organisms are required to depolymerize highly complex glycans. Here we show that the gut bacterium Bacteroides thetaiotaomicron uses the most structurally complex glycan known: the plant pectic polysaccharide rhamnogalacturonan-II, cleaving all but 1 of its 21 distinct glycosidic linkages. The deconstruction of rhamnogalacturonan-II side chains and backbone are coordinated to overcome steric constraints, and the degradation involves previously undiscovered enzyme families and catalytic activities. The degradation system informs revision of the current structural model of rhamnogalacturonan-II and highlights how individual gut bacteria orchestrate manifold enzymes to metabolize the most challenging glycan in the human diet. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21725">doi: 10.1038/nature21725</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/iJ7vOeD8jbo" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ZJui0D6hdxg/nature21714" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ZJui0D6hdxg/nature21714" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ZJui0D6hdxg/nature21714&title=Structural insights into adiponectin receptors suggest ceramidase activity" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ZJui0D6hdxg/nature21714">Structural insights into adiponectin receptors suggest ceramidase activity</a></h4>

<p>Adiponectin receptors (ADIPORs) are integral membrane proteins that control glucose and lipid metabolism by mediating, at least in part, a cellular ceramidase activity that catalyses the hydrolysis of ceramide to produce sphingosine and a free fatty acid (FFA). The crystal structures of the two receptor subtypes, ADIPOR1 and ADIPOR2, show a similar overall seven-transmembrane-domain architecture with large unoccupied cavities and a zinc binding site within the seven transmembrane domain. However, the molecular mechanisms by which ADIPORs function are not known. Here we describe the crystal structure of ADIPOR2 bound to a FFA molecule and show that ADIPOR2 possesses intrinsic basal ceramidase activity that is enhanced by adiponectin. We also identify a ceramide binding pose and propose a possible mechanism for the hydrolytic activity of ADIPOR2 using computational approaches. In molecular dynamics simulations, the side chains of residues coordinating the zinc rearrange quickly to promote the nucleophilic attack of a zinc-bound hydroxide ion onto the ceramide amide carbonyl. Furthermore, we present a revised ADIPOR1 crystal structure exhibiting a seven-transmembrane-domain architecture that is clearly distinct from that of ADIPOR2. In this structure, no FFA is observed and the ceramide binding pocket and putative zinc catalytic site are exposed to the inner membrane leaflet. ADIPOR1 also possesses intrinsic ceramidase activity, so we suspect that the two distinct structures may represent key steps in the enzymatic activity of ADIPORs. The ceramidase activity is low, however, and further studies will be required to characterize fully the enzymatic parameters and substrate specificity of ADIPORs. These insights into ADIPOR function will enable the structure-based design of potent modulators of these clinically relevant enzymes.</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21714">doi: 10.1038/nature21714</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/ZJui0D6hdxg" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ptOQ8VkEocA/nature21706" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ptOQ8VkEocA/nature21706" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ptOQ8VkEocA/nature21706&title=The lung is a site of platelet biogenesis and a reservoir for haematopoietic progenitors" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/ptOQ8VkEocA/nature21706">The lung is a site of platelet biogenesis and a reservoir for haematopoietic progenitors</a></h4>

<p>
 Platelets are critical for haemostasis, thrombosis, and inflammatory responses, but the events that lead to mature platelet production remain incompletely understood. The bone marrow has been proposed to be a major site of platelet production, although there is indirect evidence that the lungs might also contribute to platelet biogenesis. Here, by directly imaging the lung microcirculation in mice, we show that a large number of megakaryocytes circulate through the lungs, where they dynamically release platelets. Megakaryocytes that release platelets in the lungs originate from extrapulmonary sites such as the bone marrow; we observed large megakaryocytes migrating out of the bone marrow space. The contribution of the lungs to platelet biogenesis is substantial, accounting for approximately 50% of total platelet production or 10 million platelets per hour. Furthermore, we identified populations of mature and immature megakaryocytes along with haematopoietic progenitors in the extravascular spaces of the lungs. Under conditions of thrombocytopenia and relative stem cell deficiency in the bone marrow, these progenitors can migrate out of the lungs, repopulate the bone marrow, completely reconstitute blood platelet counts, and contribute to multiple haematopoietic lineages. These results identify the lungs as a primary site of terminal platelet production and an organ with considerable haematopoietic potential. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21706">doi: 10.1038/nature21706</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/ptOQ8VkEocA" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/zId4E5Uyy8s/nature21703" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/zId4E5Uyy8s/nature21703" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/zId4E5Uyy8s/nature21703&title=Somatic mutations reveal asymmetric cellular dynamics in the early human embryo" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/zId4E5Uyy8s/nature21703">Somatic mutations reveal asymmetric cellular dynamics in the early human embryo</a></h4>

<p>
 Somatic cells acquire mutations throughout the course of an individual’s life. Mutations occurring early in embryogenesis are often present in a substantial proportion of, but not all, cells in postnatal humans and thus have particular characteristics and effects. Depending on their location in the genome and the proportion of cells they are present in, these mosaic mutations can cause a wide range of genetic disease syndromes and predispose carriers to cancer. They have a high chance of being transmitted to offspring as de novo germline mutations and, in principle, can provide insights into early human embryonic cell lineages and their contributions to adult tissues. Although it is known that gross chromosomal abnormalities are remarkably common in early human embryos, our understanding of early embryonic somatic mutations is very limited. Here we use whole-genome sequences of normal blood from 241 adults to identify 163 early embryonic mutations. We estimate that approximately three base substitution mutations occur per cell per cell-doubling event in early human embryogenesis and these are mainly attributable to two known mutational signatures. We used the mutations to reconstruct developmental lineages of adult cells and demonstrate that the two daughter cells of many early embryonic cell-doubling events contribute asymmetrically to adult blood at an approximately 2:1 ratio. This study therefore provides insights into the mutation rates, mutational processes and developmental outcomes of cell dynamics that operate during early human embryogenesis. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21703">doi: 10.1038/nature21703</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/zId4E5Uyy8s" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/p3eT3NGRwTE/nature21702" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/p3eT3NGRwTE/nature21702" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/p3eT3NGRwTE/nature21702&title=The allosteric inhibitor ABL001 enables dual targeting of BCR–ABL1" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/p3eT3NGRwTE/nature21702">The allosteric inhibitor ABL001 enables dual targeting of BCR–ABL1</a></h4>

<p>
 Chronic myeloid leukaemia (CML) is driven by the activity of the BCR–ABL1 fusion oncoprotein. ABL1 kinase inhibitors have improved the clinical outcomes for patients with CML, with over 80% of patients treated with imatinib surviving for more than 10 years. Second-generation ABL1 kinase inhibitors induce more potent molecular responses in both previously untreated and imatinib-resistant patients with CML. Studies in patients with chronic-phase CML have shown that around 50% of patients who achieve and maintain undetectable BCR–ABL1 transcript levels for at least 2 years remain disease-free after the withdrawal of treatment. Here we characterize ABL001 (asciminib), a potent and selective allosteric ABL1 inhibitor that is undergoing clinical development testing in patients with CML and Philadelphia chromosome-positive (Ph+) acute lymphoblastic leukaemia. In contrast to catalytic-site ABL1 kinase inhibitors, ABL001 binds to the myristoyl pocket of ABL1 and induces the formation of an inactive kinase conformation. ABL001 and second-generation catalytic inhibitors have similar cellular potencies but distinct patterns of resistance mutations, with genetic barcoding studies revealing pre-existing clonal populations with no shared resistance between ABL001 and the catalytic inhibitor nilotinib. Consistent with this profile, acquired resistance was observed with single-agent therapy in mice; however, the combination of ABL001 and nilotinib led to complete disease control and eradicated CML xenograft tumours without recurrence after the cessation of treatment. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21702">doi: 10.1038/nature21702</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/p3eT3NGRwTE" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/f9FpdpaMRKo/nature21700" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/f9FpdpaMRKo/nature21700" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/f9FpdpaMRKo/nature21700&title=A new hypothesis of dinosaur relationships and early dinosaur evolution" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/f9FpdpaMRKo/nature21700">A new hypothesis of dinosaur relationships and early dinosaur evolution</a></h4>

<p>
 For 130 years, dinosaurs have been divided into two distinct clades—Ornithischia and Saurischia. Here we present a hypothesis for the phylogenetic relationships of the major dinosaurian groups that challenges the current consensus concerning early dinosaur evolution and highlights problematic aspects of current cladistic definitions. Our study has found a sister-group relationship between Ornithischia and Theropoda (united in the new clade Ornithoscelida), with Sauropodomorpha and Herrerasauridae (as the redefined Saurischia) forming its monophyletic outgroup. This new tree topology requires redefinition and rediagnosis of Dinosauria and the subsidiary dinosaurian clades. In addition, it forces re-evaluations of early dinosaur cladogenesis and character evolution, suggests that hypercarnivory was acquired independently in herrerasaurids and theropods, and offers an explanation for many of the anatomical features previously regarded as notable convergences between theropods and early ornithischians. 
</p><p>Nature 543 501 <a href="http://dx.doi.org/10.1038/nature21700">doi: 10.1038/nature21700</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/f9FpdpaMRKo" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/6zL5dLHyVqI/nature21682" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/6zL5dLHyVqI/nature21682" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/6zL5dLHyVqI/nature21682&title=Neural ensemble dynamics underlying a long-term associative memory" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/6zL5dLHyVqI/nature21682">Neural ensemble dynamics underlying a long-term associative memory</a></h4>

<p>
 The brain’s ability to associate different stimuli is vital for long-term memory, but how neural ensembles encode associative memories is unknown. Here we studied how cell ensembles in the basal and lateral amygdala encode associations between conditioned and unconditioned stimuli (CS and US, respectively). Using a miniature fluorescence microscope, we tracked the Ca2+ dynamics of ensembles of amygdalar neurons during fear learning and extinction over 6 days in behaving mice. Fear conditioning induced both up- and down-regulation of individual cells’ CS-evoked responses. This bi-directional plasticity mainly occurred after conditioning, and reshaped the neural ensemble representation of the CS to become more similar to the US representation. During extinction training with repetitive CS presentations, the CS representation became more distinctive without reverting to its original form. Throughout the experiments, the strength of the ensemble-encoded CS–US association predicted the level of behavioural conditioning in each mouse. These findings support a supervised learning model in which activation of the US representation guides the transformation of the CS representation. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21682">doi: 10.1038/nature21682</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/6zL5dLHyVqI" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/w9i3QwqMLiA/nature21433" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/w9i3QwqMLiA/nature21433" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/w9i3QwqMLiA/nature21433&title=Antigen presentation profiling reveals recognition of lymphoma immunoglobulin neoantigens" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/w9i3QwqMLiA/nature21433">Antigen presentation profiling reveals recognition of lymphoma immunoglobulin neoantigens</a></h4>

<p>
 Cancer somatic mutations can generate neoantigens that distinguish malignant from normal cells. However, the personalized identification and validation of neoantigens remains a major challenge. Here we discover neoantigens in human mantle-cell lymphomas by using an integrated genomic and proteomic strategy that interrogates tumour antigen peptides presented by major histocompatibility complex (MHC) class I and class II molecules. We applied this approach to systematically characterize MHC ligands from 17 patients. Remarkably, all discovered neoantigenic peptides were exclusively derived from the lymphoma immunoglobulin heavy- or light-chain variable regions. Although we identified MHC presentation of private polymorphic germline alleles, no mutated peptides were recovered from non-immunoglobulin somatically mutated genes. Somatic mutations within the immunoglobulin variable region were almost exclusively presented by MHC class II. We isolated circulating CD4+ T cells specific for immunoglobulin-derived neoantigens and found these cells could mediate killing of autologous lymphoma cells. These results demonstrate that an integrative approach combining MHC isolation, peptide identification, and exome sequencing is an effective platform to uncover tumour neoantigens. Application of this strategy to human lymphoma implicates immunoglobulin neoantigens as targets for lymphoma immunotherapy. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21433">doi: 10.1038/nature21433</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/w9i3QwqMLiA" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/-A6dedDXzkE/nature21408" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/-A6dedDXzkE/nature21408" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/-A6dedDXzkE/nature21408&title=LACTB is a tumour suppressor that modulates lipid metabolism and cell state" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/-A6dedDXzkE/nature21408">LACTB is a tumour suppressor that modulates lipid metabolism and cell state</a></h4>

<p>
 Post-mitotic, differentiated cells exhibit a variety of characteristics that contrast with those of actively growing neoplastic cells, such as the expression of cell-cycle inhibitors and differentiation factors. We hypothesized that the gene expression profiles of these differentiated cells could reveal the identities of genes that may function as tumour suppressors. Here we show, using in vitro and in vivo studies in mice and humans, that the mitochondrial protein LACTB potently inhibits the proliferation of breast cancer cells. Its mechanism of action involves alteration of mitochondrial lipid metabolism and differentiation of breast cancer cells. This is achieved, at least in part, through reduction of the levels of mitochondrial phosphatidylserine decarboxylase, which is involved in the synthesis of mitochondrial phosphatidylethanolamine. These observations uncover a novel mitochondrial tumour suppressor and demonstrate a connection between mitochondrial lipid metabolism and the differentiation program of breast cancer cells, thereby revealing a previously undescribed mechanism of tumour suppression. 
</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21408">doi: 10.1038/nature21408</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/-A6dedDXzkE" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/DaNiTr4_wkQ/543582a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/DaNiTr4_wkQ/543582a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/DaNiTr4_wkQ/543582a&title=Mr Singularity" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/DaNiTr4_wkQ/543582a">Mr Singularity</a></h4>

<p></p><p>Nature 543 582 <a href="http://dx.doi.org/10.1038/543582a">doi: 10.1038/543582a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/DaNiTr4_wkQ" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/I_qd-qbLcp8/543495a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/I_qd-qbLcp8/543495a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/I_qd-qbLcp8/543495a&title=Microbiology: Bacterial transmission tactics" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/I_qd-qbLcp8/543495a">Microbiology: Bacterial transmission tactics</a></h4>

<p>Genome sequencing of  Mycobacterium abscessus  strains that infect the lungs suggests a possible shift in the bacterium's mode of infection from environmental acquisition to human transmission. This finding has clinical implications.</p><p>Nature 543 495 <a href="http://dx.doi.org/10.1038/543495a">doi: 10.1038/543495a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/I_qd-qbLcp8" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/kftSrlF_0mg/543494a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/kftSrlF_0mg/543494a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/kftSrlF_0mg/543494a&title=Palaeontology: Dividing the dinosaurs" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/kftSrlF_0mg/543494a">Palaeontology: Dividing the dinosaurs</a></h4>

<p>The standard dinosaur evolutionary tree has two key branches: the 'bird-hipped' Ornithischia and the 'reptile-hipped' Saurischia. A revised tree challenges many ideas about the relationships between dinosaur groups.  See Article    p.501</p><p>Nature 543 494 <a href="http://dx.doi.org/10.1038/543494a">doi: 10.1038/543494a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/kftSrlF_0mg" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/Hu9QOuj3jPs/543492a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/Hu9QOuj3jPs/543492a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/Hu9QOuj3jPs/543492a&title=Eugene Garfield (1925–2017)" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/Hu9QOuj3jPs/543492a">Eugene Garfield (1925–2017)</a></h4>

<p>Inventor of the Science Citation Index.</p><p>Nature 543 492 <a href="http://dx.doi.org/10.1038/543492a">doi: 10.1038/543492a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/Hu9QOuj3jPs" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/8TUMTr0kF30/543491e" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/8TUMTr0kF30/543491e" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/8TUMTr0kF30/543491e&title=Chromatography: Shirts off to boost separation methods" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/8TUMTr0kF30/543491e">Chromatography: Shirts off to boost separation methods</a></h4>

<p></p><p>Nature 543 491 <a href="http://dx.doi.org/10.1038/543491e">doi: 10.1038/543491e</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/8TUMTr0kF30" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/k3C3WdNY6Gk/543491d" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/k3C3WdNY6Gk/543491d" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/k3C3WdNY6Gk/543491d&title=Environment: China's decadal pollution census" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/k3C3WdNY6Gk/543491d">Environment: China's decadal pollution census</a></h4>

<p></p><p>Nature 543 491 <a href="http://dx.doi.org/10.1038/543491d">doi: 10.1038/543491d</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/k3C3WdNY6Gk" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/UHSkAuh4Tz4/543491c" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/UHSkAuh4Tz4/543491c" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/UHSkAuh4Tz4/543491c&title=Politics: Listen to accused Turkish scientists" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/UHSkAuh4Tz4/543491c">Politics: Listen to accused Turkish scientists</a></h4>

<p></p><p>Nature 543 491 <a href="http://dx.doi.org/10.1038/543491c">doi: 10.1038/543491c</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/UHSkAuh4Tz4" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/M4e5a3ceEfo/543491b" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/M4e5a3ceEfo/543491b" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/M4e5a3ceEfo/543491b&title=Renewable energy: Commercial hurdles to wave power" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/M4e5a3ceEfo/543491b">Renewable energy: Commercial hurdles to wave power</a></h4>

<p></p><p>Nature 543 491 <a href="http://dx.doi.org/10.1038/543491b">doi: 10.1038/543491b</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/M4e5a3ceEfo" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/yuGuXMrgwyQ/543491a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/yuGuXMrgwyQ/543491a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/yuGuXMrgwyQ/543491a&title=Corporate culture: protect idea factories" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/yuGuXMrgwyQ/543491a">Corporate culture: protect idea factories</a></h4>

<p></p><p>Nature 543 491 <a href="http://dx.doi.org/10.1038/543491a">doi: 10.1038/543491a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/yuGuXMrgwyQ" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureNewsComment/~3/isgucXxAKTY/543490a" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureNewsComment/~3/isgucXxAKTY/543490a" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureNewsComment/~3/isgucXxAKTY/543490a&title=Evolution: Four takes on the evolution of art" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureNewsComment/~3/isgucXxAKTY/543490a">Evolution: Four takes on the evolution of art</a></h4>

<p>Nurin Veis tours a show on theories of creativity, curated by a quartet of scientists.</p><p>Nature 543 490 <a href="http://dx.doi.org/10.1038/543490a">doi: 10.1038/543490a</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureNewsComment/~4/isgucXxAKTY" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a> on March 22, 2017 12:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/blog/science-ticker/large-hadron-collider-experiment-nabs-five-new-particles" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/blog/science-ticker/large-hadron-collider-experiment-nabs-five-new-particles" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/blog/science-ticker/large-hadron-collider-experiment-nabs-five-new-particles&title=Large Hadron Collider experiment nabs five new particles" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/blog/science-ticker/large-hadron-collider-experiment-nabs-five-new-particles">Large Hadron Collider experiment nabs five new particles</a></h4>

LHCb experiment detects new particles composed of two strange quarks and one charm quark.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 21, 2017 07:25 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/21/editor-chief-caught-manipulating-citations-now-hes-asked-resign/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/21/editor-chief-caught-manipulating-citations-now-hes-asked-resign/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/21/editor-chief-caught-manipulating-citations-now-hes-asked-resign/&title=An editor in chief was caught manipulating citations. Now he’s been asked to resign." target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/21/editor-chief-caught-manipulating-citations-now-hes-asked-resign/">An editor in chief was caught manipulating citations. Now he’s been asked to resign.</a></h4>

<p>An earth science journal has asked an editor to resign after it was revealed he had been manipulating citations at multiple journals. Artemi Cerdà had already agreed to step down temporarily from Land Degradation &amp; Development after the publisher, Wiley, was alerted that Cerdà had resigned from other journals for citation manipulation. In a new statement, […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/21/editor-chief-caught-manipulating-citations-now-hes-asked-resign/" rel="nofollow">An editor in chief was caught manipulating citations. Now he’s been asked to resign.</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 21, 2017 05:35 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/21/macchiarini-blames-karolinska-losing-data-retracts-2014-paper/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/21/macchiarini-blames-karolinska-losing-data-retracts-2014-paper/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/21/macchiarini-blames-karolinska-losing-data-retracts-2014-paper/&title=Macchiarini blames Karolinska for losing data as he retracts 2014 paper" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/21/macchiarini-blames-karolinska-losing-data-retracts-2014-paper/">Macchiarini blames Karolinska for losing data as he retracts 2014 paper</a></h4>

<p>Ask and ye shall receive: A journal has retracted a 2014 paper by Paolo Macchiarini, upon request from the Karolinska Institutet (KI). The latest news is only one step in a long-running saga about former star surgeon Macchiarini, who was dismissed from KI last year. To read more, check out our timeline. KI announced it was […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/21/macchiarini-blames-karolinska-losing-data-retracts-2014-paper/" rel="nofollow">Macchiarini blames Karolinska for losing data as he retracts 2014 paper</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 21, 2017 04:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/cancer-cells-cast-sweet-spell-immune-system" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/cancer-cells-cast-sweet-spell-immune-system" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/cancer-cells-cast-sweet-spell-immune-system&title=Cancer cells cast a sweet spell on the immune system" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/cancer-cells-cast-sweet-spell-immune-system">Cancer cells cast a sweet spell on the immune system</a></h4>

Tumors have surface sugars that persuade the body’s defenses to look the other way. New therapies are being devised to break the trance.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 21, 2017 04:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/close-pass-sun-didnt-radically-alter-comet-67ps-landscape" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/close-pass-sun-didnt-radically-alter-comet-67ps-landscape" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/close-pass-sun-didnt-radically-alter-comet-67ps-landscape&title=Close pass by sun didn’t radically alter comet 67P’s landscape" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/close-pass-sun-didnt-radically-alter-comet-67ps-landscape">Close pass by sun didn’t radically alter comet 67P’s landscape</a></h4>

Landslides on comet 67P shot plumes of dust into space, but changes like these might not radically alter the landscape of the comet.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 21, 2017 02:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/21/nature-letter-adds-non-reproducibility-list-woes/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/21/nature-letter-adds-non-reproducibility-list-woes/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/21/nature-letter-adds-non-reproducibility-list-woes/&title=Nature paper adds non-reproducibility to its list of woes" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/21/nature-letter-adds-non-reproducibility-list-woes/">Nature paper adds non-reproducibility to its list of woes</a></h4>

<p>Despite taking some serious hits, a 2006 letter in Nature isn’t going anywhere. Years ago, a university committee determined that two figures in the letter had been falsified. The journal chose to correct the paper, rather than retract it — and then, the next year, published a correction of that correction due to “an error […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/21/nature-letter-adds-non-reproducibility-list-woes/" rel="nofollow">Nature paper adds non-reproducibility to its list of woes</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 21, 2017 01:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/21/cobweb-spiders-wrap-prey-diverse-silk-proteins-expanding-silk-applications/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/21/cobweb-spiders-wrap-prey-diverse-silk-proteins-expanding-silk-applications/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/21/cobweb-spiders-wrap-prey-diverse-silk-proteins-expanding-silk-applications/&title=Cobweb spiders wrap prey with diverse silk proteins, expanding silk applications" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/21/cobweb-spiders-wrap-prey-diverse-silk-proteins-expanding-silk-applications/">Cobweb spiders wrap prey with diverse silk proteins, expanding silk applications</a></h4>

<img alt="nature-1242617_960_720" class="attachment-thumbnail size-thumbnail wp-post-image" height="110" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/nature-1242617_960_720-140x110.jpg" style="display: block; margin-bottom: 5px; clear: both;" width="140" /><p>Walking through a spider web in the woods or sweeping out the silk from the corners of your house may seem like a pesky nuisance, but these tenacious webs are engineering feats. The majority of the cobweb, the frame of the orb-web, and the nearly invisible spider lifeline are all made of one type of spider silk that can have tensile strength rivaling steel. This is the major ampullate silk, named after the specialized abdominal gland in which it is synthesized. In terms of toughness, or the total amount of energy required to break, major ampullate silk outperforms steel and Kevlar. Consequently, it has been a prime target for human applications, with some textiles <a href="http://www.newyorker.com/tech/elements/in-the-future-well-all-wear-spider-silk?mbid=nl_170313_Daily&amp;CNDID=29472763&amp;spMailingID=10610432&amp;spUserID=MTMzMTgyOTk4NzI0S0&amp;spJobID=1121040554&amp;spReportId=MTEyMTA0MDU1NAS2">near production</a>.</p>
<p>Even more remarkable is that spiders do not produce just one silk type, unlike the silk moth caterpillars whose cocoons we use to make silk cloth. Instead, an orb-web weaving garden spider or cobweb weaver like the black widow make at least seven types of silk in as many morphologically distinct silk glands.</p>
<figure class="wp-caption alignleft" id="attachment_8730" style="width: 1290px;"><img alt="Silk glands were dissected from the false black widow, Steatoda grossa, and are still attached to their spinnerets (bottom center). The large ampule-shaped glands on other side are major ampullate. The smaller similar shaped glands in center top are minor ampullate." class="size-full wp-image-8730" height="1024" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Ayoub-1.jpg" width="1280" /><div class="attachment-image--caption">Silk glands were dissected from the false black widow, <em>Steatoda grossa</em>, and are still attached to their spinnerets (bottom center). The large ampule-shaped glands on other side are major ampullate. The smaller similar shaped glands in center top are minor ampullate.</div><div class="attachment-image--source">Nadia Ayoub</div></figure>
<p>The material properties of each of these silks appear to be <a href="http://jeb.biologists.org/content/209/13/2452">tailored to their ecological functions</a>. For instance, spiders cover their eggs with tubuliform silk, which in garden spiders is stiffer than any other silk type. Egg sacs are thus resistant to pressure that could crush the eggs. The aciniform silk used to wrap prey by garden spiders is the toughest of its silk types, withstanding the energy output of a struggling insect.</p>
<p>Orb-web weaving spiders use minor ampullate silks as temporary capture spirals, which are not as strong as draglines but are equally tough due to their higher stretchiness. Alternatively, black widow spiders use minor ampullate silk to wrap prey, but material properties from this species were previously unknown. My colleagues and I predicted that black widow and other cobweb weaver minor ampullate silks would have divergent material properties from orb-web weavers in part due to their varying functions. We further predicted that the DNA sequences of the genes encoding the minor ampullate silk proteins would differ between orb-web and cobweb weaving spiders.</p>
<p><strong>Sequencing spider silk genes in undergraduate courses</strong></p>
<p>Each type of spider silk is primarily made of one or more unique proteins that are collectively called spidroins (spider fibrous proteins). Spidroins and insect silk proteins share the property of being long (typically &gt;2000 amino acids) and repetitive (short sequences of amino acids could repeat 10-100s of times). Spidroins are thus notoriously difficult to completely sequence.</p>
<p>My colleagues and I have been using a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3563967/pdf/mss254.pdf">combination of approaches</a> to characterize spidroins. Undergraduate students started helping me collect these data when I was a post-doctoral researcher at the University of California, Riverside in Cheryl Hayashi’s lab. When I took a tenure-track job at Washington and Lee University I decided to scale up undergraduate involvement.</p>
<p>Each year in my Genetics Lab, 17-24 students tackle a new spider species. For our research on minor ampullate silks, <a href="http://bmcevolbiol.biomedcentral.com/articles/10.1186/s12862-017-0927-x">recently published in <em>BMC Evolutionary Biology</em></a>, encoding gene sequences were characterized by three different cohorts. Three students continued working with me in the summers and ultimately became the first three authors on our publication.</p>
<figure class="wp-caption alignleft" id="attachment_8731" style="width: 2394px;"><img alt="Undergraduates, Jeremy Miller (far left) and Jannelle Vienneau-Hathaway (second from left and first author), worked with Nadia Ayoub (second from right) on cobweb spiders’ minor ampullate silk genes. Bioinformatics specialist, Toby Clarke (far right), also co-authored the recent publication. Pictured at Arthropod Genomics Symposium, 2015" class="size-full wp-image-8731" height="1659" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Ayoub-2.jpg" width="2384" /><div class="attachment-image--caption">Undergraduates, Jeremy Miller (far left) and Jannelle Vienneau-Hathaway (second from left and first author), worked with Nadia Ayoub (second from right) on cobweb spiders’ minor ampullate silk genes. Bioinformatics specialist, Toby Clarke (far right), also co-authored the recent publication. Pictured at Arthropod Genomics Symposium, 2015.</div><div class="attachment-image--source">Jessica Garb</div></figure>
<p><strong>Variation in minor ampullate silk proteins correlates with variation in silk properties</strong></p>
<p>After years of sequencing spider silk genes, we discovered unexpected diversity of minor ampullate silk proteins. Not only did orb-web weaver sequences differ from cobweb weavers’, dramatic differences were also found among cobweb weavers. Even more surprising was that each species’ genome included multiple genes encoding minor ampullate silk proteins, and these sequences were often strikingly different.</p>
<p>As an example, the brown widow (invasive in the United States), has one minor ampullate gene that encodes a protein only 815 amino acids long and includes “motifs” typical for minor ampullate proteins: e.g. glycine-alanine repeats. Another brown widow gene encodes a minor ampullate protein more than double the size of the first (1791 amino acids) that is riddled with glycine-proline-glycine. Proline tends to make a kink in proteins. Thus many glycine-proline-glycine iterations form spring-like spirals, which should make the protein elastic.</p>
<p>The false black widow’s minor ampullate proteins were the most proline-rich. Their minor ampullate silks were correspondingly twice as extensible as the minor ampullate silks of the black and brown widow. Overall, cobweb weaver minor ampullate silks were more extensible, but not as strong as orb-web weaver silks.</p>
<p><strong>Leveraging cobweb prey wrapping proteins for human applications</strong></p>
<p>Our sequencing methods resulted in minor ampullate-encoding genes isolated in a format that would facilitate expression of pure protein. Our work could thus contribute to efforts to spin artificial fibers with custom-made properties. For instance, the proline-containing brown widow minor ampullate spidroin could be used to make extensible fibers incorporated into artificial tendons or <a href="http://www.livescience.com/1094-spider-silk-repair-human-ligaments.html">ligaments</a>. The longer alanine-rich black widow proteins may make stiffer fibers.</p>
<p> </p>
<p>Co-author websites:</p>
<p>Nadia Ayoub <a href="https://www.wlu.edu/directory/profile?ID=x8123">https://www.wlu.edu/directory/profile?ID=x8123</a></p>
<p>Jessica Garb <a href="http://blogs.uml.edu/garb-lab/">http://blogs.uml.edu/garb-lab/</a></p>
<p>Kelly Lane <a href="http://www.genetics.uga.edu/directory/kelly-lane">http://www.genetics.uga.edu/directory/kelly-lane</a></p>
<p>The post <a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/21/cobweb-spiders-wrap-prey-diverse-silk-proteins-expanding-silk-applications/" rel="nofollow">Cobweb spiders wrap prey with diverse silk proteins, expanding silk applications</a> appeared first on <a href="http://blogs.biomedcentral.com/bmcseriesblog" rel="nofollow">BMC Series blog</a>.</p>
                <p class="text-muted">in <a href="http://blogs.biomedcentral.com/bmcseriesblog">BMC Series blog</a> on March 21, 2017 11:16 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/life-earth-may-have-begun-dividing-droplets" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/life-earth-may-have-begun-dividing-droplets" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/life-earth-may-have-begun-dividing-droplets&title=Life on Earth may have begun as dividing droplets" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/life-earth-may-have-begun-dividing-droplets">Life on Earth may have begun as dividing droplets</a></h4>

Chemical droplets could split and reproduce in the presence of an energy source, new computer simulations suggest.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss">Science News</a> on March 21, 2017 11:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://digest.bps.org.uk/2017/03/21/more-human-than-thou-or-just-better-our-motivation-to-think-were-good-trumps-our-desire-to-feel-human/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://digest.bps.org.uk/2017/03/21/more-human-than-thou-or-just-better-our-motivation-to-think-were-good-trumps-our-desire-to-feel-human/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://digest.bps.org.uk/2017/03/21/more-human-than-thou-or-just-better-our-motivation-to-think-were-good-trumps-our-desire-to-feel-human/&title=More human than thou … or just better? Our motivation to think we’re good trumps our desire to feel human" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://digest.bps.org.uk/2017/03/21/more-human-than-thou-or-just-better-our-motivation-to-think-were-good-trumps-our-desire-to-feel-human/">More human than thou … or just better? Our motivation to think we’re good trumps our desire to feel human</a></h4>

<figure class="wp-caption alignnone" data-shortcode="caption" id="attachment_21204" style="width: 1732px;"><img data-attachment-id="21204" data-comments-opened="1" data-image-meta="{" data-orig-file="https://researchdigest.files.wordpress.com/2017/03/gettyimages-534032988.jpg?w=845" data-orig-size="1732,1732" data-permalink="https://digest.bps.org.uk/2017/03/21/more-human-than-thou-or-just-better-our-motivation-to-think-were-good-trumps-our-desire-to-feel-human/the-dark-side-of-human-the-mirror-concept-vector-illustration/" />A new critique challenges the claim that we’re so motivated to believe in our essential humanness that we happily endorse ugly parts of our human natures</figure>
<p><em>By</em> <a href="http://www.alexfradera.net/">Alex Fradera</a></p>
<p>Most of us have a sense of what it means to be human. Research shows that we agree with each other that traits like friendly, jealous or impatient are more “human” than others like unemotional or selfless. What’s more, we like to see ourselves as human: we care more about human traits and claim to possess them more than other people. In other words, we “self-humanise”, laying claim to the good <em>and</em> the bad as long as they emphasise our own humanity.</p>
<p>But this research on self-humanising presents a conundrum. A different, abundant line of evidence shows that humans bitterly protect a highly positive self-image, supported by cognitive biases that attribute our own failings to <a href="http://https://en.wikipedia.org/wiki/Self-serving_bias">circumstances</a> and other people’s to their <a href="http://https://en.wikipedia.org/wiki/Hostile_attribution_bias">deficiencies</a>. So, do we really overestimate the bad in ourselves, claiming to be more human, warts and all? According to <a href="http://www.tandfonline.com/doi/full/10.1080/00224545.2017.1282849">a critique</a> of the self-humanising field in <em>The Journal of Social Psychology</em>, this is an oversimplification: when it comes to undesirable human traits, we see ourselves as pretty similar to other people.</p>
<p><span id="more-21177"></span></p>
<p>The researchers, led by Marzena Cypryańska of SWPS University Warsaw, conducted their work with 250 students from universities in Poland, Korea, and Italy. The participants – mostly in their twenties and fairly gender-balanced – rated themselves on 40 different traits relative to other students. Some traits like Sympathetic and Jealous were strongly associated with human nature, others like Selfless and Unemotional were not.</p>
<p>Most previous self-humanising research has compared scores in two different categories, for example finding that people give themselves significantly higher self-ratings than others for negative human traits, but not for non-human negative traits. But Cypryańska’s team pointed out that this approach is problematic because of the kind of rating scales that have been used. Imagine a participant rates themselves on a “human” trait like Jealous using the scale below:</p>
<p><img data-attachment-id="21182" data-comments-opened="1" data-image-meta="{" data-orig-file="https://researchdigest.files.wordpress.com/2017/03/self-humanising-scale.png?w=480&amp;h=109" data-orig-size="361,82" data-permalink="https://digest.bps.org.uk/2017/03/21/more-human-than-thou-or-just-better-our-motivation-to-think-were-good-trumps-our-desire-to-feel-human/self-humanising-scale/" /></p>
<p>Strictly speaking, only a score above 4 would suggest that a participant truly believes they are “more Jealous than” others. However, the way that past research has worked, if a participant gave themselves a 3 on a negative human trait, this would be taken as a significantly higher self-rating than a 2 on a non-human negative trait. But the problem is that feeling “less” strongly that you have less of a trait than others is not the same as believing that you have “more” of it. Cypryańska and her colleagues point out that in their data, participants’ mean self-ratings of their negative traits – both human in nature and otherwise – were <em>never</em> significantly higher than 4.</p>
<p>What’s more, Cypryańska and her team revisited previous work that has reported statistically significant self-humanising for negative human traits using similar rating methods, and the same pattern emerged: that is, participants rarely claimed to have more of negative human traits, they were just less bold in their claims to have less of them. So the evidence doesn’t show people are so motivated to appear human that they happily endorse ugly parts of their human nature.</p>
<p>On the other hand, participants did often rate themselves above the scale midpoint for positive traits, both human nature traits and other positive ones. Rather than proving we associate ourselves with human vices and virtues, these findings can be explained with the well-established <a href="http://psycnet.apa.org/psycinfo/2005-14648-005">Better Than Average</a> effect, where we associate ourselves more with positive attributes, less with negative ones.</p>
<p>It’ll be interesting to see how the self-humanising research body responds to this critique of their methods. Cypryańska’s group are not opposed to the idea that human traits are treated a little differently; perhaps we distance ourselves from our weaknesses, but with less fervour when they stem from a common human heritage. More research would be needed to unpick this. But according to this study, the ways we see ourselves as special have a pecking order. Good first, human later.</p>
<p>—<a href="http://www.tandfonline.com/doi/full/10.1080/00224545.2017.1282849">Reconsidering research on self-humanizing: The importance of mean comparative judgments</a></p>
<p>Image via <a href="http://www.gettyimages.co.uk">gettyimages.co.uk</a></p>
<p><strong><a href="http://www.alexfradera.net/">Alex Fradera</a> (<a href="https://twitter.com/alexfradera">@alexfradera</a>) is Contributing Writer at <a href="https://digest.bps.org.uk/">BPS Research Digest</a></strong></p><br />  <a href="http://feeds.wordpress.com/1.0/gocomments/researchdigest.wordpress.com/21177/" rel="nofollow"><img alt="" border="0" src="http://feeds.wordpress.com/1.0/comments/researchdigest.wordpress.com/21177/" /></a> <img alt="" border="0" height="1" src="https://pixel.wp.com/b.gif?host=digest.bps.org.uk&amp;blog=114693443&amp;post=21177&amp;subd=researchdigest&amp;ref=&amp;feed=1" width="1" />
                <p class="text-muted">in <a href="https://digest.bps.org.uk">The British Psychological Society - Research Digest</a> on March 21, 2017 10:09 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/single-atom-magnets-store-bits-data" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/single-atom-magnets-store-bits-data" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/single-atom-magnets-store-bits-data&title=Single-atom magnets store bits of data" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/single-atom-magnets-store-bits-data">Single-atom magnets store bits of data</a></h4>

Scientists read and write data by harnessing the magnetic properties of holmium atoms.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss?0_8640516396055629=1">Science News</a> on March 20, 2017 08:25 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/genetic-switch-offers-clue-why-grasses-are-survival-masters" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/genetic-switch-offers-clue-why-grasses-are-survival-masters" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/genetic-switch-offers-clue-why-grasses-are-survival-masters&title=Genetic switch offers clue to why grasses are survival masters" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/genetic-switch-offers-clue-why-grasses-are-survival-masters">Genetic switch offers clue to why grasses are survival masters</a></h4>

Scientists have identified a genetic switch that helps grasses regulate their carbon dioxide intake.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss?0_8640516396055629=1">Science News</a> on March 20, 2017 07:35 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/20/leads-bias-scientific-literature-new-study-tries-answer/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/20/leads-bias-scientific-literature-new-study-tries-answer/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/20/leads-bias-scientific-literature-new-study-tries-answer/&title=What leads to bias in the scientific literature? New study tries to answer" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/20/leads-bias-scientific-literature-new-study-tries-answer/">What leads to bias in the scientific literature? New study tries to answer</a></h4>

<p>By now, most of our readers are aware that some fields of science have a reproducibility problem. Part of the problem, some argue, is the publishing community’s bias toward dramatic findings — namely, studies that show something has an effect on something else are more likely to be published than studies that don’t. Many have […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/20/leads-bias-scientific-literature-new-study-tries-answer/" rel="nofollow">What leads to bias in the scientific literature? New study tries to answer</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 20, 2017 07:00 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/under-lasers-feathered-dino-shows-some-skin" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/under-lasers-feathered-dino-shows-some-skin" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/under-lasers-feathered-dino-shows-some-skin&title=Under lasers, a feathered dino shows some skin" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/under-lasers-feathered-dino-shows-some-skin">Under lasers, a feathered dino shows some skin</a></h4>

Laser-stimulated fluorescence reveals detailed images of soft tissue in a feathered dinosaur from 160 million years ago.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss?_=261">Science News</a> on March 20, 2017 06:40 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/20/scientist-sued-university-earns-two-retractions-bringing-total-five/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/20/scientist-sued-university-earns-two-retractions-bringing-total-five/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/20/scientist-sued-university-earns-two-retractions-bringing-total-five/&title=Scientist who sued university earns two more retractions, bringing total to five" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/20/scientist-sued-university-earns-two-retractions-bringing-total-five/">Scientist who sued university earns two more retractions, bringing total to five</a></h4>

<p>A scientist who sued his employer for millions of dollars has earned two more retractions, for papers that had already been flagged by the journal. By our count, Rakesh Kumar now has five retractions and multiple corrections. Kumar sued his employer, George Washington University, for $8 million, alleging emotional distress when they put him on leave […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/20/scientist-sued-university-earns-two-retractions-bringing-total-five/" rel="nofollow">Scientist who sued university earns two more retractions, bringing total to five</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 20, 2017 03:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://digest.bps.org.uk/2017/03/20/have-we-overestimated-the-effectiveness-of-psychotherapy/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://digest.bps.org.uk/2017/03/20/have-we-overestimated-the-effectiveness-of-psychotherapy/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://digest.bps.org.uk/2017/03/20/have-we-overestimated-the-effectiveness-of-psychotherapy/&title=Have we overestimated the effectiveness of psychotherapy?" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://digest.bps.org.uk/2017/03/20/have-we-overestimated-the-effectiveness-of-psychotherapy/">Have we overestimated the effectiveness of psychotherapy?</a></h4>

<p><em><img data-attachment-id="21170" data-comments-opened="1" data-image-meta="{" data-orig-file="https://researchdigest.files.wordpress.com/2017/03/gettyimages-177731703.jpg?w=845" data-orig-size="3087,2822" data-permalink="https://digest.bps.org.uk/2017/03/20/have-we-overestimated-the-effectiveness-of-psychotherapy/depression-relief/" />By</em> <a href="http://www.psychologywriter.org.uk/">Christian Jarrett</a></p>
<p>Most people who undertake psychotherapy seem to benefit from it. How do we know? Arguably, the most important evidence comes from meta-analyses that combine the results from many – sometimes hundreds – of randomly controlled trials. Based on this, it’s been estimated that psychotherapy is effective for about 80 per cent of people (meanwhile, between five to 10 per cent of clients <a href="https://thepsychologist.bps.org.uk/volume-21/edition-1/when-therapy-causes-harm">may suffer adverse effects</a>).</p>
<p>But now the more concerning news: a team of researchers led by Evangelos Evangelou at the University of Ioannina, Greece has assessed the quality of 247 of these psychotherapy meta-analyses and <a href="http://onlinelibrary.wiley.com/doi/10.1111/acps.12713/full">they report</a> in <em>Acta Psychiatrica Scandinavica</em> that many of them have serious methodological short-comings.</p>
<p>Coincidentally, a separate research group led by Brent Roberts at the University of Illinois, Urbana-Champaign has <a href="http://www.sciencedirect.com/science/article/pii/S0092656616302410">just published</a> in <em>Journal of Personality</em> some of the first observational data on how people’s personalities change after undertaking psychotherapy. In contrast to what’s been found <a href="https://digest.bps.org.uk/2017/01/19/a-little-discussed-effect-of-therapy-it-changes-your-personality/">in the clinical literature</a>, they report that people who’ve been in therapy seem to show negative changes in personality and other psychological outcomes.</p>
<p><span id="more-21028"></span></p>
<p>In their “umbrella review” of the psychotherapy literature, Evangelou and his team chose to include published meta-analyses covering any form of psychotherapy and almost any kind of target mental health condition, but with the proviso that each meta-analysis had itself combined the results from a minimum of ten studies. Their comprehensive search resulted in 247 unique meta-analyses which collectively synthesised data from over five thousand randomly controlled trials.</p>
<p>Overall, 80 per cent of the published psychotherapy meta-analyses had reported a significant and positive benefit of whatever form of psychotherapy was their focus. This sounds impressive at first, but after applying “state-of-the-art” tests of their robustness, Evangelou and his colleagues report that just 16 of the 247 meta-analyses had provided “convincing evidence”.</p>
<p>The researchers identified a number of issues:</p>
<ul>
<li>Many meta-analyses showed a statistically significant amount of heterogeneity between the trials that they’d combined. The worry is that too many meta-analyses are comparing apples and oranges, although there is scholarly debate about what level of heterogeneity is unacceptable.</li>
<li>The researchers found many instances of the “small study bias”, which is the tendency for smaller, less robust studies to report larger effects.</li>
<li>They found evidence of “<a href="http://journals.sagepub.com/doi/abs/10.1177/1740774507079441?url_ver=Z39.88-2003&amp;rfr_id=ori:rid:crossref.org&amp;rfr_dat=cr_pub%3dpubmed">excess significance bias</a>“, which is when an over-abundance of trials seem to report positive findings given what we know so far about psychotherapy’s effectiveness. This suggests negative findings are remaining unpublished for whatever reason.</li>
</ul>
<p>Evangelou and his team conclude that the field of psychotherapy research needs to work harder to ensure that negative results are published as well as good news results, especially given <a href="https://digest.bps.org.uk/2017/02/20/concerning-study-says-psychotherapy-research-has-a-problem-with-undeclared-researcher-bias/">the findings of another recent paper</a> suggesting that the field has a problem with undeclared researcher allegiance to particular therapeutic approaches. One way round this is to ensure all trials and meta-analyses are preregistered before they are conducted, alongside information on the statistical tests that are planned.</p>
<p>For the reader undertaking psychotherapy or who knows someone who is, it is worth keeping some perspective: this is just one critique and the weight of evidence still suggests that psychotherapy is, more often than not, beneficial.</p>
<p>For scholars, the methodological concerns raised by this new paper feed into an already contentious field. Some experts believe that, high-quality or not, randomly controlled trials (and by extension, meta-analyses based on those trials) are not really an appropriate way to gauge the effectiveness of psychotherapy because of all complex, myriad factors involved in the dynamic between a client and his or her therapist.</p>
<p>An alternative approach is to look at observational data. Rather than signing people up to a controlled trial, with all the contrivances that that entails (such as standardising the delivery of therapy as much as possible), this approach is less hands-on and involves looking instead at the outcomes of people who happen to have been in therapy and those who haven’t.</p>
<p>Brent Roberts and his colleagues found two sources of this kind of data: hundreds of students in Tübingen who were enrolled in a long-term personality study and who’d completed measures twice across four years; and a group of thousands of older Americans who similarly had completed personality and other measures twice across four years.</p>
<p>Crucially, both these longitudinal surveys included a question about whether the participants had undertaken psychotherapy in the intervening period between the two data collection points. One hundred and twenty-eight of the Tübingen students had completed some therapy and, compared to the other students, they showed negative changes in their personality in terms of higher scores in neuroticism, less extraversion and conscientiousness, as well as reductions in self-esteem, increases in depression, and less life satisfaction. It was a similar story for the older American adults who’d been in therapy: they showed negative changes in personality and other psychological outcomes.</p>
<p>There are problems with how to interpret these findings – an obvious shortcoming of observational data of this kind is that it’s less controlled than an experimental trial. For example, perhaps undertaking therapy was a consequence of these unwelcome psychological changes rather than the cause (although this still wouldn’t explain why the therapy hadn’t been more helpful, more often). Whatever the explanation, however, the results <a href="https://digest.bps.org.uk/2017/01/19/a-little-discussed-effect-of-therapy-it-changes-your-personality/">stand in contrast to the findings from controlled psychotherapy trials</a> which have pointed overwhelming to positive personality changes arising from therapy. As Roberts and his team conclude, “the gravity of the issue necessitates that researchers investigate the apparent discrepancy between these findings and those from well-controlled trials”.</p>
<p>—<a href="http://onlinelibrary.wiley.com/doi/10.1111/acps.12713/full">Does psychotherapy work? An umbrella review of meta-analyses of randomised controlled trials</a></p>
<p>—<a href="http://www.sciencedirect.com/science/article/pii/S0092656616302410">Therapy experience in naturalistic observational studies is associated with negative changes in personality</a></p>
<p>Image via <a href="http://www.gettyimages.co.uk">gettyimages.co.uk</a></p>
<p><span class="s1"><a href="http://www.psychologywriter.org.uk/"><b>Christian Jarrett</b></a></span><span class="s2"><b> (</b><a href="https://twitter.com/psych_Writer"><span class="s3"><b>@Psych_Writer</b></span></a><b>) is Editor of </b><span class="s3"><b><a href="https://digest.bps.org.uk/">BPS Research Digest</a></b></span></span></p><br />  <a href="http://feeds.wordpress.com/1.0/gocomments/researchdigest.wordpress.com/21028/" rel="nofollow"><img alt="" border="0" src="http://feeds.wordpress.com/1.0/comments/researchdigest.wordpress.com/21028/" /></a> <img alt="" border="0" height="1" src="https://pixel.wp.com/b.gif?host=digest.bps.org.uk&amp;blog=114693443&amp;post=21028&amp;subd=researchdigest&amp;ref=&amp;feed=1" width="1" />
                <p class="text-muted">in <a href="https://digest.bps.org.uk">The British Psychological Society - Research Digest</a> on March 20, 2017 01:54 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://retractionwatch.com/2017/03/20/japan-taiwan-taking-closer-look-fraud-stop/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://retractionwatch.com/2017/03/20/japan-taiwan-taking-closer-look-fraud-stop/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://retractionwatch.com/2017/03/20/japan-taiwan-taking-closer-look-fraud-stop/&title=Japan, Taiwan taking closer look at fraud — and how to stop it" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://retractionwatch.com/2017/03/20/japan-taiwan-taking-closer-look-fraud-stop/">Japan, Taiwan taking closer look at fraud — and how to stop it</a></h4>

<p>Two countries have recently announced plans to learn more about research misconduct, with the goal of preventing it from happening in the first place. In Japan, the effort takes the form of a joint study group among six universities, which will interview researchers who have engaged in misconduct to discover patterns and common factors for their […]</p>
<p>The post <a href="http://retractionwatch.com/2017/03/20/japan-taiwan-taking-closer-look-fraud-stop/" rel="nofollow">Japan, Taiwan taking closer look at fraud — and how to stop it</a> appeared first on <a href="http://retractionwatch.com" rel="nofollow">Retraction Watch</a>.</p>
                <p class="text-muted">in <a href="http://retractionwatch.com">Retraction watch</a> on March 20, 2017 01:30 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.elsevier.com/connect/cashews-to-combat-mosquitoes-top-5-proposals-for-the-green-and-sustainable-chemistry-challenge" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.elsevier.com/connect/cashews-to-combat-mosquitoes-top-5-proposals-for-the-green-and-sustainable-chemistry-challenge" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.elsevier.com/connect/cashews-to-combat-mosquitoes-top-5-proposals-for-the-green-and-sustainable-chemistry-challenge&title=Cashews to combat mosquitoes? Top 5 proposals for the Green and Sustainable Chemistry Challenge" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.elsevier.com/connect/cashews-to-combat-mosquitoes-top-5-proposals-for-the-green-and-sustainable-chemistry-challenge">Cashews to combat mosquitoes? Top 5 proposals for the Green and Sustainable Chemistry Challenge</a></h4>

Finalists for this Elsevier Foundation challenge to present their proposals in Berlin, vying for a €50,000 first prize
                <p class="text-muted">in <a href="">Elsevier Connect</a> on March 20, 2017 01:28 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.elsevier.com/connect/game-based-engineering-contest-challenges-students-to-solve-real-world-problems" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.elsevier.com/connect/game-based-engineering-contest-challenges-students-to-solve-real-world-problems" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.elsevier.com/connect/game-based-engineering-contest-challenges-students-to-solve-real-world-problems&title=Game-based engineering contest challenges students to solve real-world problems" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.elsevier.com/connect/game-based-engineering-contest-challenges-students-to-solve-real-world-problems">Game-based engineering contest challenges students to solve real-world problems</a></h4>

2017 Engineering Academic Challenge is open for registration
                <p class="text-muted">in <a href="">Elsevier Connect</a> on March 20, 2017 01:06 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/20/late-date-differences-mating-time-may-drive-species-apart/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/20/late-date-differences-mating-time-may-drive-species-apart/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://blogs.biomedcentral.com/bmcseriesblog/2017/03/20/late-date-differences-mating-time-may-drive-species-apart/&title=Too late to date: how differences in mating time may drive a species apart" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/20/late-date-differences-mating-time-may-drive-species-apart/">Too late to date: how differences in mating time may drive a species apart</a></h4>

<img alt="Adult moths on a corn plant" class="attachment-thumbnail size-thumbnail wp-post-image" height="110" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Hanninger-1-1-140x110.jpg" style="display: block; margin-bottom: 5px; clear: both;" width="140" /><p>The notorious fall armyworm – you’ve probably heard of it. It’s a night active moth and it’s been in the news lately as an invasive species in Africa that causes severe damage to corn and other crops, <a href="http://www.bbc.com/news/science-environment-38859851">threatens the livelihoods of farmers</a> and may <a href="https://www.theguardian.com/world/2017/feb/14/southern-africas-food-crops-under-threat-from-fall-armyworm-invasion">strip southern Africa of key food sources</a>. The fall armyworm (<em>Spodoptera frugiperda</em>) originates from the Americas, where e.g. Brazil spends <a href="http://www.herald.co.zw/fao-to-spend-millions-of-dollars-in-fight-against-fall-armyworm/">over 600 million dollars every year</a> to control the pest.</p>
<p>While we need to figure out how to control the crop pest in Africa, there are some important things we can already learn from the moth. One of the central questions in biology is: where does all the biodiversity come from?</p>
<p>The fall armyworm can help us answer this question, as this species occurs as two different strains, called corn-strain and rice-strain. The strains look exactly alike, but differ in some features, like <a href="https://www.ncbi.nlm.nih.gov/pubmed/23456344">pheromone</a> <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0089255">composition</a> and <a href="http://onlinelibrary.wiley.com/doi/10.1111/eea.12215/abstract">host plant preference</a>, but most importantly <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1420-9101.2009.01759.x/abstract;jsessionid=85B84C3A8F56A14D4DCFF7F3FCB8EDD3.f04t04">in their timing of mating</a>: the corn-strain mates early in the night, while the rice strain mates late in the night. How did we find these differences?</p>
<p>We observed hundreds of couples in a dark room, lit by a red light and a red flashlight, as moths can’t see red light. After the lights went off and night set in, the corn-strain moths became active, soon followed by matings. Meanwhile, the rice-strain moths were sitting still, motionless. Only after several hours, the rice-strain moths woke up and started their mating activities.</p>
<figure class="wp-caption alignleft" id="attachment_8725" style="width: 2570px;"><img alt="Female adult moth extruding her pheromone gland, advertising her availability for mating" class="size-full wp-image-8725" height="1920" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Hanninger-2.jpg" width="2560" /><div class="attachment-image--caption">Female adult moth extruding her pheromone gland, advertising her availability for mating</div><div class="attachment-image--source">Sabine Hänniger</div></figure>
<p>When two populations (or strains) of the same species don’t mate with each other, genes are not exchanged between these populations. Mutations that occur spontaneously are thus not exchanged and stay within the population, so that different mutations can accumulate in each population. In this way, two populations will become more and more genetically different over time, which can result in the development into two species.</p>
<p>Looking at the strains in our dark room that were either mating early or late, we wanted to know what genetic differences had accumulated between the two strains that could explain their differences in mating times at night. Therefore, we hybridized the two strains and observed the mating activities of the parental strains together with the hybrids and their offspring throughout the night with our red flashlights.</p>
<figure class="wp-caption alignleft" id="attachment_8726" style="width: 310px;"><img alt="Hundreds of beakers for observation" class="size-medium wp-image-8726" height="225" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Hanninger-4-300x225.jpg" width="300" /><div class="attachment-image--caption">Hundreds of beakers for observation</div><div class="attachment-image--source">Sabine Hänniger</div></figure>
<p>We then analyzed all those individuals genetically, and by combining the observation and genetic data, we could identify one chromosome that explained most of the timing difference between the strains. We were excited when we also found out that this chromosome harbors a gene that is part of the circadian clock: <em>vrille</em>!</p>
<p>Thus, the circadian rhythm of the corn-strain and the rice-strain appear to be out of sync and we identified <em>vrille</em> as a candidate to cause the asynchrony. When comparing the DNA sequence of this gene between the strains, we found some interesting differences in the promoter region.</p>
<p>The circadian clock is an internal molecular clock present in most living beings on earth. Dependent on light input, genes and proteins interact with each other in a circadian rhythm that is roughly 24 hours. This rhythm helps mice, for example, to wake up and get active at night when they can’t easily be seen by predators, and it helps the fall armyworm to wake up at the right time to find a mating partner.</p>
<figure class="wp-caption alignleft" id="attachment_8727" style="width: 2570px;"><img alt="Adult moths mating." class="size-full wp-image-8727" height="1920" src="http://blogs.biomedcentral.com/bmcseriesblog/wp-content/uploads/sites/9/2017/03/Hanninger-3.jpg" width="2560" /><div class="attachment-image--caption">Adult moths mating.</div><div class="attachment-image--source">Sabine Hänniger</div></figure>
<p>We also looked at differences in the mRNA levels at every hour for 24 hours and found that <em>vrille</em> is always more expressed in the rice-strain than in the corn-strain. This made us curious about how this difference influences the other genes of the circadian clock. Excitingly, we did find some differences in expression levels of the genes that directly interact with vrille in the circadian clock.</p>
<p>In short, we may have identified a mechanism underlying evolution in progress between two strains that seem to become two different species. A change in the promoter region of the gene <em>vrille</em> may cause a change in the transcription of the gene, which can influence the circadian clock. An altered circadian clock alters the behavior of the fall armyworm, causing the different strains to mate at different time points at night. This timing difference constitutes a mating barrier that keeps the strains apart and may facilitate the development into different species.</p>
<p>Thus, in addition to being a severe pest that needs to be eradicated, the fall armyworm has given us insights in the evolutionary process of speciation, which has led to the great biodiversity on earth.</p>
<p>The post <a href="http://blogs.biomedcentral.com/bmcseriesblog/2017/03/20/late-date-differences-mating-time-may-drive-species-apart/" rel="nofollow">Too late to date: how differences in mating time may drive a species apart</a> appeared first on <a href="http://blogs.biomedcentral.com/bmcseriesblog" rel="nofollow">BMC Series blog</a>.</p>
                <p class="text-muted">in <a href="http://blogs.biomedcentral.com/bmcseriesblog">BMC Series blog</a> on March 20, 2017 12:28 PM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with.html" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with.html" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with.html&title=Nerd Food: Northwind, or Using Dogen with ODB - Part III" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://mcraveiro.blogspot.com/2017/03/nerd-food-northwind-or-using-dogen-with.html">Nerd Food: Northwind, or Using Dogen with ODB - Part III</a></h4>

Nerd Food: Northwind, or Using Dogen with ODB - Part III<div id="content"><table border="0"><tbody><tr><td width="50%"></td><td width="50%"><p class="verse" style="text-align: left;"><small>Optimism is an occupational hazard of programming; feedback is the treatment. <i>-- <a href="https://books.google.co.uk/books?id=G8EL4H4vf7UC&amp;lpg=PA31&amp;ots=j9AMurgRCr&amp;dq=Optimism%20is%20an%20occupational%20hazard%20of%20programming%3B%20feedback%20is%20the%20treatment.%20Extreme%20Programming%20Explained&amp;pg=PA31#v=onepage&amp;q&amp;f=false">Kent Beck</a></i></small></p></td></tr></tbody></table> <p>Welcome to the third part of a series of N blog posts on using Dogen with ODB against an Oracle database. If you want more than the TL;DR, please read <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with.html">Part I</a> and <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with_24.html">Part II</a>. Otherwise, the story so far can be quickly summarised as follows: we got our Oracle Express database installed and set up by adding the required users; we then built the ODB libraries and installed the ODB compiler. </p> <p>After this rather grand build up, we shall finally get to look at Dogen - just about. It now seems clear these series will have to be extended by at least one or two additional instalments in order to provide a vaguely sensible treatment of the material I had initially planned to cover. I wasn't expecting N to become so large, but - like every good software project - I'm now realising you can only estimate the size of the series properly once you've actually finished it. And to rub salt into the wounds, before we can proceed we must start by addressing some of the instructions in the previous posts which were <i>not quite right</i>. </p> <div class="outline-2" id="outline-container-sec-1"><h2 id="sec-1">Est Humanum Errare?</h2><div class="outline-text-2" id="text-1"><p>The first and foremost point in the <i>errata</i> agenda is concerned with the additional Oracle packages we downloaded in <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with.html">Part I</a>. When I had originally checked my Oracle XE install, I did not find an <code>include</code>directory, which led me to conclude that a separate download was required for driver libraries and header files. I did find this state of affairs somewhat unusual - but then again, it is Oracle we're talking about here, so "unusual" is the default behaviour. As it turns out, I was wrong; the header files are indeed part of the Oracle XE install, just placed under a rather… shall we say, <i>creative</i>, location: <code>/u01/app/oracle/product/11.2.0/xe/rdbms/public</code>. The libraries are there too, under the slightly more conventionally named <code>lib</code> directory. </p> <p>This is quite an important find because the downloaded OCI driver has moved on to v12 whereas XE is still on v11. There is backwards compatibility, of course - and everything should work fine connecting a v12 client against an v11 database - but it does introduce an extra layer of complexity: you now need to make sure you do not simultaneously have both v11 and v12 shared objects in the path when linking and running or else you will start to get some strange warnings. As usual, we try our best to confuse only one issue at a time, so we need to make sure we are making use of v11 and purge all references to v12; this entails recompiling ODB's oracle support. </p> <p>If you followed the instructions on <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with_24.html">Part II</a> and you have already installed the ODB Oracle library, you'll need to remove it first: </p> <pre class="example"><br />rm /full/path/to/local/lib/libodb-oracle* /full/path/to/local/include/odb/oracle<br /></pre> <p>Remember to replace <code>/full/path/to/local</code> with the path to your local directory. Then, you can build by following the instructions as per previous post, but with <b>one crucial difference</b> at configure time: point to the Oracle XE directories instead of the external OCI driver directories: </p> <pre class="example"><br />. /u01/app/oracle/product/11.2.0/xe/bin/oracle_env.sh<br />LD_LIBRARY_PATH=/u01/app/oracle/product/11.2.0/xe/lib CPPFLAGS="-I/full/path/to/local/include -I/u01/app/oracle/product/11.2.0/xe/rdbms/public" LDFLAGS="-L/full/path/to/local/lib -L/u01/app/oracle/product/11.2.0/xe/lib" ./configure --prefix=/full/path/to/local<br /></pre> <p>Again, replacing the paths accordingly. If all goes well, the end result should be an ODB Oracle library that uses the OCI driver from Oracle XE. You then just need to make sure you have executed <code>oracle_env.sh</code> before running your binary, but don't worry too much because I'll remind you later on. Whilst we're on the subject of Oracle packages, it's worth mentioning that I did a minor update to <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with.html">Part I</a>: you didn't need to download SQLPlus separately either, as it is also included in XE package. So, in conclusion, after a lot of faffing, it turns out you can get away with just downloading XE and nothing else. </p> <p>The other minor alteration to what was laid out on the original posts is that I removed the need for the <code>basic</code> database schema. In truth, the entities placed in that schema were not adding a lot of value; their use cases are already covered by the <code>northwind</code> schema, so I removed the need for two schemas and collapsed them into one. </p> <p>A final note - not quite an <i>errata per se</i> but still, something worthwhile mentioning. We didn't do a "proper" Oracle setup, so when you reboot your box you will find that the service is no longer running. You can easily restart it from the shell, logged in as root: </p> <pre class="example"><br /># cd /etc/init.d/<br /># ./oracle-xe start<br />Starting oracle-xe (via systemctl): oracle-xe.service.<br /></pre> <p>Notice that Debian is actually clever enough to integrate the Oracle scripts with systemd, so you can use the usual tools to find out more about this service: </p> <pre class="example"><br /># systemctl status oracle-xe<br />● oracle-xe.service - SYSV: This is a program that is responsible for taking care of<br />   Loaded: loaded (/etc/init.d/oracle-xe; generated; vendor preset: enabled)<br />   Active: active (exited) since Sun 2017-03-12 15:10:47 GMT; 6s ago<br />     Docs: man:systemd-sysv-generator(8)<br />  Process: 16761 ExecStart=/etc/init.d/oracle-xe start (code=exited, status=0/SUCCESS)<br /><br />Mar 12 15:10:37 lorenz systemd[1]: Starting SYSV: This is a program that is responsible for taking c…e of...<br />Mar 12 15:10:37 lorenz oracle-xe[16761]: Starting Oracle Net Listener.<br />Mar 12 15:10:37 lorenz su[16772]: Successful su for oracle by root<br />Mar 12 15:10:37 lorenz su[16772]: + ??? root:oracle<br />Mar 12 15:10:37 lorenz su[16772]: pam_unix(su:session): session opened for user oracle by (uid=0)<br />Mar 12 15:10:39 lorenz oracle-xe[16761]: Starting Oracle Database 11g Express Edition instance.<br />Mar 12 15:10:39 lorenz su[16800]: Successful su for oracle by root<br />Mar 12 15:10:39 lorenz su[16800]: + ??? root:oracle<br />Mar 12 15:10:39 lorenz su[16800]: pam_unix(su:session): session opened for user oracle by (uid=0)<br />Mar 12 15:10:47 lorenz systemd[1]: Started SYSV: This is a program that is responsible for taking care of.<br />Hint: Some lines were ellipsized, use -l to show in full.<br /></pre> <p>With all of this said, lets resume from where we left off. </p></div></div> <div class="outline-2" id="outline-container-sec-2"><h2 id="sec-2">Installing the Remaining Packages</h2><div class="outline-text-2" id="text-2"><p>We still have a number of packages to install, but fortunately the installation steps are easy enough so we'll cover them quickly in this section. Let's start with Dogen. </p></div> <div class="outline-3" id="outline-container-sec-2-1"><h3 id="sec-2-1">Dogen</h3><div class="outline-text-3" id="text-2-1"><p>Installing Dogen is fairly straightforward: we can just grab the latest release from BinTray: </p> <ul class="org-ul"><li><a href="https://dl.bintray.com/domaindrivenconsulting/Dogen/0.99.0/dogen_0.99.0_amd64-applications.deb">dogen 0.99.0 amd64-applications.deb</a></li></ul> <p>As it happens, we must install v99 or above because we did a number of fixes to Dogen as a result of this series of articles; previous releases had shortcomings with their ODB support. </p> <p>As expected, the setup is pretty standard-fare debian: </p> <pre class="example"><br />$ wget https://dl.bintray.com/domaindrivenconsulting/Dogen/0.99.0/dogen_0.99.0_amd64-applications.deb -O dogen_0.99.0_amd64-applications.deb<br />$ sudo dpkg -i dogen_0.99.0_amd64-applications.deb<br />[sudo] password for USER:<br />Selecting previously unselected package dogen-applications.<br />(Reading database ... 551550 files and directories currently installed.)<br />Preparing to unpack dogen_0.99.0_amd64-applications.deb ...<br />Unpacking dogen-applications (0.99.0) ...<br />Setting up dogen-applications (0.99.0) ...<br /></pre> <p>If all has gone according to plan, you should see something along the lines of: </p> <pre class="example"><br />$ dogen.knitter --version<br />Dogen Knitter v0.99.0<br />Copyright (C) 2015-2017 Domain Driven Consulting Plc.<br />Copyright (C) 2012-2015 Marco Craveiro.<br />License: GPLv3 - GNU GPL version 3 or later &lt;http://gnu.org/licenses/gpl.html&gt;.<br /></pre></div></div> <div class="outline-3" id="outline-container-sec-2-2"><h3 id="sec-2-2">Dia</h3><div class="outline-text-3" id="text-2-2"><p>Dogen has multiple frontends - at the time of writing, JSON and <a href="https://wiki.gnome.org/Apps/Dia/">Dia</a>. We'll stick with Dia because of its visual nature, but keep in mind that what you can do with Dia you can also do with JSON. </p> <p>A quick word on Dia for those not in the know, copied <i>verbatim</i> from its home page: </p> <blockquote><p>Dia is a GTK+ based diagram creation program for GNU/Linux, MacOS X, Unix, and Windows, and is released under the GPL license. </p> <p>Dia is roughly inspired by the commercial Windows program 'Visio,' though more geared towards informal diagrams for casual use. It can be used to draw many different kinds of diagrams. It currently has special objects to help draw entity relationship diagrams, UML diagrams, flowcharts, network diagrams, and many other diagrams. </p></blockquote> <p>Dia does not change very often, which means any old version will do. You should be able to install dia straight off of package manager: </p> <pre class="example"><br />apt-get install dia<br /></pre></div></div> <div class="outline-3" id="outline-container-sec-2-3"><h3 id="sec-2-3">Other Dependencies</h3><div class="outline-text-3" id="text-2-3"><p>I had previously assumed Boost to be installed on <a href="http://mcraveiro.blogspot.co.uk/2017/02/nerd-food-northwind-or-using-dogen-with_24.html">Part II</a> but - if nothing else, purely for the sake of completeness - here are the instructions to set it up, as well as CMake and Ninja. We will need these in order to build our application, but we won't dwell on them <i>too</i> much on them or else this series of posts would go on forever. Pretty much any recent version of Boost and CMake will do, so again we'll just stick to vanilla package manager: </p> <pre class="example"><br /># apt-get install cmake ninja-build libboost-all-dev<br /></pre> <p>Mind you, you don't <i>actually</i> need the entirety of Boost for this exercise, but it's just easier this way. </p></div></div> <div class="outline-3" id="outline-container-sec-2-4"><h3 id="sec-2-4">Emacs and SQL Plus</h3><div class="outline-text-3" id="text-2-4"><p>Finally, a couple of lose notes which I might as well add here. If you wish to use SQLPlus from within Emacs - and you should, since the <a href="http://dbspecialists.com/an-introduction-to-emacs-sqli-mode/">SQLi mode</a> is just simply glorious - you can configure it to use our Oracle Express database quite easily: </p> <pre class="example"><br />(add-to-list 'exec-path "/u01/app/oracle/product/11.2.0/xe/bin")<br />(setenv "PATH" (concat (getenv "PATH") ":/u01/app/oracle/product/11.2.0/xe/bin"))<br />(setenv "ORACLE_HOME" "/u01/app/oracle/product/11.2.0/xe")<br /></pre> <p>After this you will be able to start SQL Plus from Emacs with the usual <code>sql-oracle</code> command. I recommend you to do at least a minimal setup of SQL Plus too, to make it usable: </p> <pre class="example"><br />SQL&gt; set linesize 8192<br />SQL&gt; set pagesize 50000<br /></pre></div></div></div> <div class="outline-2" id="outline-container-sec-3"><h2 id="sec-3">Introducing Zango</h2><div class="outline-text-2" id="text-3"><p>After this excruciatingly long setup process, we can at <i>long last</i>start to create our very "simple" project. Simple in quotes because it ended up being a tad more complex than what was originally envisioned, so it was easier to create a <a href="https://github.com/DomainDrivenConsulting/zango">GitHub repository</a> for it. It would have been preferable to describe it from first principles, but then the commentary would literally go on for ever. A compromise had to be made. </p> <p>In order to follow the remainder of this post please clone <code>zango</code>from GitHub: </p> <pre class="example"><br />git clone git@github.com:DomainDrivenConsulting/zango.git<br /></pre> <p>Zango is a very small Dogen project that builds with CMake. Here are some notes on the folder structure to help you navigate: </p> <ul class="org-ul"><li><code>build/cmake</code>: additional CMake modules that are not part of the standard CMake distribution. We need this for ODB, Oracle and Dogen. </li><li><code>data</code>: some application data that we will use to populate our database. </li><li><code>projects</code>: where all the code lives. </li><li><code>projects/input_models</code>: location of the Dogen models - in this case, we just have one. You could, of course, place it anywhere you'd like, but traditionally this is where they live. </li><li><code>projects/northwind</code>: code output of the Dogen model. This is the key project of <code>zango</code>. </li><li><code>projects/application</code>: our little command line driver for the application. </li></ul> <p>Now, before we get to look at the code I'd like to first talk about Northwind and on the relationship between Dogen and ODB. </p></div> <div class="outline-3" id="outline-container-sec-3-1"><h3 id="sec-3-1">Northwind Schema</h3><div class="outline-text-3" id="text-3-1"><p>Microsoft makes the venerable Northwind database available in CodePlex, at <a href="https://northwinddatabase.codeplex.com/">this</a> location. I found a useful description of the Northwind database <a href="http://www.geeksengine.com/article/northwind.html">here</a>, which I quote: </p> <blockquote><p>Northwind Traders Access database is a sample database that shipped with Microsoft Office suite. The Northwind database contains the sales data for a fictitious company called Northwind Traders, which imports and exports specialty foods from around the world. You can use and experiment with Access with Northwind database while you're learning and develop ideas for Access. </p></blockquote> <p>If you <i>really</i> want a thorough introduction to Northwind, you could do worse than reading this paper: <a href="http://eduglopedia.org/resource/cea008e18dee42899aeff7a0c58dddaf8e9b5bc6/download">Adapting the Access Northwind Database to Support a Database Course</a>. Having said that, for the purposes of this series we don't really need to dig <i>that</i> deep. In fact, I'll just present CodePlex's diagram with the tables and their relationships to give you an idea of the schema - without any further commentary - and that's more or less all that needs to be said about it: </p> <img height="80%" src="https://raw.githubusercontent.com/DomainDrivenConsulting/dogen/master/doc/blog/images/Northwind_A4_size_for_Print.png" width="80%" /><div>Northwind Schema (C) Microsoft.</div> <p>Now, <i>in theory</i>, we could use this image to manually extract all the required information to create a Dia diagram that follows Dogen's conventions, code-generate that and Bob's your Uncle. However, <i>in practice</i> we have a problem: the CodePlex project only contains <a href="https://northwinddatabase.codeplex.com/releases/view/71634">the SQL statements for Microsoft SQL Server</a>. Part of the point of this exercise is to show that we can load real data from Oracle, rather than just generate random data, so it would be nice to load up the "real" Northwind data from their own tables. This would be more of an "end-to-end" test, as opposed to using ODB to generate the tables, and Dogen to generate random data which we can push to the database. </p> <p>However, its not entirely trivial to convert T-SQL into Oracle SQL, and since this is supposed to be a "quick" project on the side - focusing on ODB and Dogen - I was keen on not spending time on unrelated activities such as SQL conversions. Fortunately, I found exactly what I was looking for: a series of posts from <a href="http://www.geeksengine.com/">GeeksEngine</a>entitled "Convert MS Access Northwind database to Oracle". For reference, these are as follows: </p> <ul class="org-ul"><li><a href="http://www.geeksengine.com/article/northwind-oracle.html">How the data types in Access Northwind are converted to Oracle</a></li><li><a href="http://www.geeksengine.com/article/northwind-oracle-2.html">Building Oracle Northwind database objects</a></li><li><a href="http://www.geeksengine.com/article/northwind-oracle-3.html">Queries to generate aggregated data for Oracle Northwind database</a></li></ul> <p>If you don't care too much about the details, you can just look at the Oracle SQL statements, available <a href="http://www.geeksengine.com/lg.php?i=oracle-northwind-sql">here</a> and copied across into the Zango project. I guess it's still worthwhile mentioning that GeeksEngine has reduced considerably the number of entities in the schema - for which they provide a rationale. Before we start an in-depth discussions into the merits of normalisation and de-normalisation and other DBA level topics, I have to stop you in your tracks. Please do not get too hung-up on the "quality" of the database schema of Northwind - either the Microsoft or the GeeksEngine one. The purpose of this exercise is merely to demonstrate how Dogen and ODB work together to provide an ORM solution. From this perspective, any vaguely realistic database schema is adequate - provided it allows us to test-drive all the features we're interested in. Whether you agree or not with the decisions the original creators of this schema made is a completely different matter, which is well beyond the scope of this series of posts. </p> <p>Right, so now we need to setup our Northwind schema and populate it with data. For this you can open a SQL Plus session with user Northwind as explained previously and then run in the SQL script: </p> <pre class="example"><br />@/path/to/zango/data/Oracle-Northwind.sql<br /></pre> <p>Replacing <code>/path/to</code> with the full path to your Zango checkout. This executes the GeeksEngine script against your local Oracle XE database. If all has gone well, you should now have a whole load of tables and data. You can sanity-check the setup by running the following SQL: </p> <pre class="example"><br />SQL&gt; select table_name from all_tables where owner = 'NORTHWIND';<br /><br />TABLE_NAME<br />------------------------------<br />ORDER_DETAILS<br />CATEGORIES<br />CUSTOMERS<br />EMPLOYEES<br />SUPPLIERS<br />SHIPPERS<br />PRODUCTS<br />ORDERS<br /><br />8 rows selected.<br /><br />SQL&gt; select employee_id, firstname, lastname from employees where rownum &lt;3;<br /><br />EMPLOYEE_ID FIRSTNAME  LASTNAME<br />----------- ---------- --------------------<br />      1 Nancy      Davolio<br />      2 Andrew     Fuller<br /></pre> <p>Now then, let's model these entities in Dogen. </p></div></div> <div class="outline-3" id="outline-container-sec-3-2"><h3 id="sec-3-2">The Dogen Model for Northwind</h3><div class="outline-text-3" id="text-3-2"><p>Before we proceed, I'm afraid I must make yet another disclaimer: a proper explanation on how to use Dia (and UML in general) is outside the scope of these articles, so you'll see me hand-waving quite a lot. Hopefully the diagrams are sufficiently self-explanatory for you to get the idea. </p> <p>The process of modeling is simply to take the entities of the GeeksEngine SQL schema and to model them in Dia, following Dogen's conventions: each SQL type is converted to what we deemed to be the closest C++ type. You can open the diagram from the folder <code>projects/input_models/northwind.dia</code>, but if you haven't got it handy, here's a screenshot of most of the UML model: </p> <img height="90%" src="https://raw.githubusercontent.com/DomainDrivenConsulting/dogen/master/doc/blog/images/dia_with_northwind.png" width="90%" /><div>Dogen Northwind model.</div> <p>The first point of note in that diagram is - if you pardon the pun - the UML note. </p>  <div class="figure"><p><img alt="odb_uml_note.png" src="https://raw.githubusercontent.com/DomainDrivenConsulting/dogen/master/doc/blog/images/odb_uml_note.png" /></p><p><span class="figure-number">Figure 1:</span> UML Note from northwind model.</p></div> <p>This configuration is quite important so we'll discuss it a bit more detail. All lines starting with <code>#DOGEN</code> are an extension mechanism used to supply meta-data into Dogen. First, lets have a very quick look at the model's more "general settings": </p> <ul class="org-ul"><li><code>yarn.dia.comment</code>: this is a special command that tells Dogen to use this UML note as the source code comments for the namespace of the model (i.e. <code>northwind</code>). Thus the text "The Northwind model is a…" will become part of a doxygen comment for the namespace. </li><li><code>yarn.dia.external_modules</code>: this places all types into the top-level namespace <code>northwind</code>. </li><li><code>yarn.input_language</code>: the notation for types used in this model is C++. We won't delve on this too much, but just keep in mind that Dogen supports both C++ and C#. </li><li><code>quilt.cpp.enabled</code>: as we are using C++, we must enable it. </li><li><code>quilt.cpp.hash.enabled</code>: we not require this feature for the purposes of this exercise. </li><li><code>quilt.csharp.enabled</code>: As this is a C++-only model, we will disable C#. </li><li><code>annotations.profile</code>: Do not worry too much about this knob, it just sets a lot of default options for this project such as copyright notices and so forth. </li></ul> <p>As promised, you won't fail to notice we hand-waved quite a lot on the description of these settings. It is very difficult to explain them properly them without giving the reader an immense amount of context about Dogen. This, of course, needs to be done - particularly since we haven't really spent the required time updating the manual. However, in the interest of keeping this series of posts somewhat focused on ODB and ORM, we'll just leave it at that, with a promise to create Dogen-specific posts on them. </p> <p>Talking about ORM, the next batch of settings is exactly related to that. </p> <ul class="org-ul"><li><code>yarn.orm.database_system</code>: here, we're stating that we're interested in both <code>oracle</code> and <code>postgresql</code> databases. </li><li><code>yarn.orm.letter_case</code>: this sets the "case" to use for all identifiers; either <code>upper_case</code> or <code>lower_case</code>. So if you choose <code>upper_case</code>, all your table names will be in upper case and vice-versa. This applies to all columns and object names on the entirety of this model (e.g. <code>customers</code> becomes <code>CUSTOMERS</code> and so forth). </li><li><code>yarn.orm.schema_name</code>: finally we set the schema name to <code>northwind</code>. Remember that we are in upper case, so the name becomes <code>NORTHWIND</code>. </li></ul> <p>In addition to the meta-data, the second point worth noticing is that there is a large overlap between C++ classes and the entities in the original diagram. For example, we have <code>customers</code>, <code>suppliers</code>, <code>employees</code> and so forth - the Object-Relational Mapping is very "linear". This is a characteristic of the Dogen approach to ORM, but you do not necessarily need to use ODB in this manner; we discuss this in the next section. </p> <p>If one is to look at a properties of a few attributes in more detail, one can see additional Dogen meta-data. Take <code>customer_id</code> in the <code>customers</code> class: </p>  <div class="figure"><p><img alt="odb_customer_id_attributes.png" src="https://raw.githubusercontent.com/DomainDrivenConsulting/dogen/master/doc/blog/images/odb_customer_id_attributes.png" /></p><p><span class="figure-number">Figure 2:</span> Properties of customer<sub>id</sub> in the customer class.</p></div> <p>The setting <code>yarn.orm.is_primary_key</code> tells Dogen that this attribute is the primary key of the table. Note that we did not use an <code>int</code> as the type of <code>customer_id</code> but instead made use of a Dogen feature called "primitives". Primitives are simple wrappers around builtins and "core" types such as <code>std::string</code>, intended to have little or no overhead after the compiler is done with them. They are useful when you want to use domain concepts to clarify intent. For example, primitives help making it obvious when you try to use a <code>customer_id</code>when a <code>supplier_id</code> was called for. It's also worth noticing that <code>customer_id</code> makes use of <code>yarn.orm.is_nullable</code> - settable to <code>true</code>or <code>false</code>. It results in Dogen telling ODB if a column can be <code>NULL</code>or not. </p> <p>As we stated, each of the attributes of these classes has the closest C++ type we could find that maps to the SQL type used in the database schema. Of course, different developers can make different choices for these types. For example, were we to store the picture data rather than a path to the picture as GeeksEngine decided to do, we would use a <code>std::vector&lt;char&gt;</code> instead of a <code>std::string</code>. In that case, we'd have to perform some additional mapping too: </p> <pre class="example"><br />#DOGEN yarn.orm.type_override=postgresql,BYTEA<br />#DOGEN yarn.orm.type_override=oracle,BLOB<br /></pre> <p>This tells Dogen about the mapping of the attribute's type to the SQL type. Dogen then conveys this information to ODB. </p> <p>Dogen's ORM support is still quite young - literally a couple of sprints old - so there will be cases where you may need to perform some customisation which is not yet available in its meta-model. In these cases, you can bypass Dogen and make use of ODB pragmas directly. As an example, GeeksEngine Oracle schema named a few columns in <code>Employees</code> without underscores such as <code>FIRSTNAME</code> and <code>LASTNAME</code>. We want the C++ classes to have the correct names (e.g. <code>first_name</code>, <code>last_name</code>, etc) so we simply tell ODB that these columns have different names in the database. Take last name for example: </p>  <div class="figure"><p><img alt="odb_last_name_attributes.png" src="https://raw.githubusercontent.com/DomainDrivenConsulting/dogen/master/doc/blog/images/odb_last_name_attributes.png" /></p><p><span class="figure-number">Figure 3:</span> Properties of last name in the employee class.</p></div> <p>A final note on composite keys. Predictably, Dogen follows the ODB approach - in that primary keys that have more than one column must be expressed as a class on its own right. In <code>northwind</code>, we use the postfix <code>_key</code> for these class names in order to make them easier to identify - e.g. <code>order_details_key</code>. You won't fail to notice that this class has the flag <code>yarn.orm.is_value</code> set. It tells Dogen - and, by extension, ODB - that it is not really a full-blown type, which would map it to a table, but instead should be treated like other value types such as <code>std::string</code>. </p></div></div> <div class="outline-3" id="outline-container-sec-3-3"><h3 id="sec-3-3">Interlude: Dogen with ODB vs Plain ODB</h3><div class="outline-text-3" id="text-3-3"><p>"The technical minutiae is all well and good", the inquisitive reader will say, "but why Dogen <i>and</i> ODB? Why add yet another layer of indirection when one can just use ODB?" Indeed, it may be puzzling for there to be a need for a code-generator which generates code for another code-generator. "Turtles all the way down" and "We over-engineered it yet again", the crowd chants from the terraces. </p> <p>Let me attempt to address some of these concerns. </p> <p>First, it is important to understand the argument we're trying to make here: Dogen models benefit greatly from ODB, but its not necessarily the case that <i>all</i> ODB users benefit from Dogen. Let's start with a classic ODB use case, which is to take an existing code base and add ORM support to it. In this scenario it makes no sense to introduce Dogen; after all, ODB requires only small changes to the original source code and has the ability to parse very complex C++. And, of course, using ODB in this manner also allows one to deal with impedance mismatches between the relational model and the object model of your domain. </p> <p>Dogen, on the other hand, exists mainly to support <a href="https://en.wikipedia.org/wiki/Model-driven_software_development">Model Driven Software Development</a> (MDSD), so the modeling process is the driver. This means that one is expected to start with a Dogen model, and to use the traditional MDSD techniques for the management of the life-cycle of your model - and eventually for the generation of <i>entire software product lines</i>. Of course, you need not buy into the whole MDSD sales pitch in order to make use of Dogen, but you should at least understand it in this context. At a bare minimum, it requires you to think in terms of Domain Models - as <a href="https://en.wikipedia.org/wiki/Domain-driven_design">Domain Driven Development</a>defines them - and then in terms of "classes of features" required by the elements of your domain. These we call "facets" in Dogen parlance. There are many such facets like <code>io</code>, which is the ability to dump an object's state into a C++ stream - at present using JSON notation - or <code>serialization</code> which is the ability to serialise an object using Boost serialisation. It is in this context that ODB enters the Dogen world. We could, of course, generate ORM mappings (and SQL) directly from Dogen. But given what we've seen from ODB, it seems this would be a very large project - or, conversely, we'd have very poor support, not dealing with a great number of corner cases. By generating the very minimal and very non-intrusive code that ODB needs, we benefit from the years of experience accumulated in this tool whilst at the same time making life easier for Dogen users. </p> <p>Of course, as with all engineering trade-offs, this one is not without its disadvantages. When things do go wrong you now have more moving parts to go through when root-causing: was it an error in the diagram, or was it Dogen, or was it the mapping between Dogen and ODB or was it ODB?  Fortunately, I found that this situation is minimised by the way in which you end up using Dogen. For instance, all generated code can be version-controlled, so you can look at the ODB input files generated by Dogen and observe how they change with changes in the Dogen model. The Dogen ODB files should also look very much like regular hand-crafted ODB files - making use of <code>pragmas</code> and so forth - and you are also required to run ODB manually against them. Thus, in practice, I have found troubleshooting straightforward enough that the layers of indirection end up not constituting a real problem. </p> <p>Finally, its worth pointing out that the Domain Models Dogen generates have a fairly straightforward shape to them, making the ODB mapping a lot more trivial that "general" C++ code would have. It is because of this that we have <code>orm</code> parameters in Dogen, which can expand to multiple ODB pragmas - the user should not need to worry about that expansion. </p></div></div></div> <div class="outline-2" id="outline-container-sec-4"><h2 id="sec-4">Conclusion</h2><div class="outline-text-2" id="text-4"><p>This part is already becoming quite large, so I'm afraid we need to stop it here and continue on Part IV. However, we have managed to address a few of the mistakes of the Oracle setup of previous parts, introduced the remaining applications that we need to install and then discussed Northwind - both in terms of its original intent and also in terms of the Dogen objectives. Finally we provided an explanation of how Dogen and ODB fit together in a tooling ecosystem. </p></div></div></div><div class="status" id="postamble"><p class="date">Created: 2017-03-19 Sun 23:07</p><p class="creator"><a href="http://www.gnu.org/software/emacs/">Emacs</a> 25.1.1 (<a href="http://orgmode.org">Org</a> mode 8.2.10)</p><p class="validation"><a href="http://validator.w3.org/check?uri=referer">Validate</a></p></div>
                <p class="text-muted">in <a href="http://mcraveiro.blogspot.com/">Marco Craveiro</a> on March 20, 2017 11:50 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.plos.org/~r/plos/Blog/~3/lJI9NLVUis4/" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.plos.org/~r/plos/Blog/~3/lJI9NLVUis4/" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.plos.org/~r/plos/Blog/~3/lJI9NLVUis4/&title=A Sincere Thank You from PLOS" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.plos.org/~r/plos/Blog/~3/lJI9NLVUis4/">A Sincere Thank You from PLOS</a></h4>

<img alt="Thank-you-640x320" class="attachment-thumbnail size-thumbnail wp-post-image" height="150" src="http://blogs.plos.org/plos/files/2017/03/Thank-you-640x320-150x150.png" style="display: block; margin-bottom: 5px; clear: both;" width="150" /><p>2017 marks the third year that PLOS formally acknowledges our community of reviewers and editors with a public thank you published as a citable journal article. We do this to honor the dedicated service and substantial contributions made by working scientists – scientists with day jobs – who graciously provide their time and expertise to our organization. Thank you journal articles provide reviewers and editors the academic citation and recognition for their valuable service to colleagues, institutions, funders and the public.</p>
<p>It takes a global network of more than 78,000 reviewers and 7,000 editors to ensure that articles, perspectives, editorials and more achieve the highest quality possible. The more than 12 million article views per month (on average) this past year shows that PLOS reviewers and editors are up to the task of vetting the science, the ethics, the reporting guidelines and journal publication criteria presented in the more than 27,000 research articles published in 2016.</p>
<p><strong>For the first time we are including in each of the seven journals’ thank you articles the number of newly submitted and published research articles brought to the public in 2016. We are confident that this data, in real terms, supports our ongoing commitment to increased transparency surrounding the publishing process.</strong></p>
<p>We are also confident that readers will appreciate the workload required of reviewers and editors to support these publication numbers. In addition, contributors to <em>PLOS Biology</em> were tasked with fine-tuning Aperta™, our new submission system, as they reviewed, discussed and moved submissions through the publication process.</p>
<p>As they shepherd papers and provide feedback so authors may improve manuscripts and refine experimental work, it is more than quality that our contributor community champions. Reviewers, Editors and Editorial Board members who choose to do so also communicate the mission of Open Access, the value of Open Data and the relevance of Open Science to society at large.</p>
<h4>These individuals serve as ambassadors not just for PLOS but for all of science. Public access to rigorous peer-reviewed research, the role of journals in communicating the work of our communities and the enthusiasm of our own staff are all sustained by this extensive pool of inspired and dedicated scientists.</h4>
<p>To all of our Reviewers, Academic Editors, Guest Editors and Editorial Board members, thank you! Those wanting to explore deeper will find every reviewer’s and editor’s name in the Supporting Information of each journal’s published article; links to these articles are below.</p>
<ul>
<li><a href="http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2002409&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=bio-1703-contribty" target="_blank">PLOS Biology</a></li>
<li><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005442&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=cbio-1703-contribty" target="_blank">PLOS Computational Biology</a></li>
<li><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006671&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=gen-1703-contribty" target="_blank">PLOS Genetics</a></li>
<li><a href="http://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1002281&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=med-1703-contribty" target="_blank">PLOS Medicine</a></li>
<li><a href="http://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0005469&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=ntd-1703-contribty" target="_blank">PLOS Neglected Tropical Diseases</a></li>
<li><a href="http://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1006278&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=path-1703-contribty" target="_blank">PLOS Pathogens</a></li>
<li><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0174259&amp;utm_source=plos&amp;utm_medium=blog&amp;utm_campaign=one-1703-contribty" target="_blank">PLOS ONE</a></li>
</ul>
<img alt="" height="1" src="http://feeds.feedburner.com/~r/plos/Blog/~4/lJI9NLVUis4" width="1" />
                <p class="text-muted">in <a href="http://blogs.plos.org/plos">The Official PLOS Blog</a> on March 20, 2017 11:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=https://www.sciencenews.org/article/shocking-stories-tell-tale-london-zoos-founding" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=https://www.sciencenews.org/article/shocking-stories-tell-tale-london-zoos-founding" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=https://www.sciencenews.org/article/shocking-stories-tell-tale-london-zoos-founding&title=Shocking stories tell tale of London Zoo’s founding" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="https://www.sciencenews.org/article/shocking-stories-tell-tale-london-zoos-founding">Shocking stories tell tale of London Zoo’s founding</a></h4>

In The Zoo, Isobel Charman pens a gripping narrative of the London Zoo’s early days, when workers had a hard time keeping animals alive.
                <p class="text-muted">in <a href="https://www.sciencenews.org/feeds/headlines.rss?_=261">Science News</a> on March 20, 2017 11:00 AM.</p>
              </li>
              <li class="list-group-item">
                <p class="text-right">-&nbsp;<a href="https://app.wallabag.it/bookmarklet?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/_o_IrLjJIGU/nature21726" target="wallabagit"><small>Wallabag.it!</small></a>&nbsp;-&nbsp;<a href="http://instapaper.com/hello2?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/_o_IrLjJIGU/nature21726" target="instapaper"><small>Save to Instapaper</small></a>&nbsp;-&nbsp;<a href="http://getpocket.com/save?url=http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/_o_IrLjJIGU/nature21726&title=Cerebellar granule cells encode the expectation of reward" target="pocket"><small>Save to Pocket</small></a>&nbsp;-</p>
                <h4><a href="http://feeds.nature.com/~r/NatureBiologicalSciencesResearch/~3/_o_IrLjJIGU/nature21726">Cerebellar granule cells encode the expectation of reward</a></h4>

<p>The human brain contains approximately 60 billion cerebellar granule cells, which outnumber all other brain neurons combined. Classical theories posit that a large, diverse population of granule cells allows for highly detailed representations of sensorimotor context, enabling downstream Purkinje cells to sense fine contextual changes. Although evidence suggests a role for the cerebellum in cognition, granule cells are known to encode only sensory and motor context. Here, using two-photon calcium imaging in behaving mice, we show that granule cells convey information about the expectation of reward. Mice initiated voluntary forelimb movements for delayed sugar-water reward. Some granule cells responded preferentially to reward or reward omission, whereas others selectively encoded reward anticipation. Reward responses were not restricted to forelimb movement, as a Pavlovian task evoked similar responses. Compared to predictable rewards, unexpected rewards elicited markedly different granule cell activity despite identical stimuli and licking responses. In both tasks, reward signals were widespread throughout multiple cerebellar lobules. Tracking the same granule cells over several days of learning revealed that cells with reward-anticipating responses emerged from those that responded at the start of learning to reward delivery, whereas reward-omission responses grew stronger as learning progressed. The discovery of predictive, non-sensorimotor encoding in granule cells is a major departure from the current understanding of these neurons and markedly enriches the contextual information available to postsynaptic Purkinje cells, with important implications for cognitive processing in the cerebellum.</p><p>Nature   <a href="http://dx.doi.org/10.1038/nature21726">doi: 10.1038/nature21726</a></p><img alt="" height="1" src="http://feeds.feedburner.com/~r/NatureBiologicalSciencesResearch/~4/_o_IrLjJIGU" width="1" />
                <p class="text-muted">in <a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a> on March 20, 2017 12:00 AM.</p>
              </li>
          </ul>
        </div>
        <div class="col-md-3">
          <div class="panel panel-default">
            <div class="panel-heading">
              <h3 class="panel-title">Feed list</h3>
            </div>
            <ul class="list-group">
                <li class="list-group-item"><a href="http://ankursinha.in/blog/">Ankur Sinha</a></li>
                <li class="list-group-item"><a href="http://blogs.biomedcentral.com/bmcseriesblog">BMC Series blog</a></li>
                <li class="list-group-item"><a href="http://brainblogger.com">Brain Blogger</a></li>
                <li class="list-group-item"><a href="https://medium.com/brain-byte-blog?source=rss----8bf7660fc7aa---4">Brain Byte - The HBP blog</a></li>
                <li class="list-group-item"><a href="http://discovermagazine.com/rss/blog-feeds/neuroskeptic">Discovery magazine - Neuroskeptic</a></li>
                <li class="list-group-item"><a href="">Elsevier Connect</a></li>
                <li class="list-group-item"><a href="https://emckiernan.wordpress.com">Erin C. McKiernan</a></li>
                <li class="list-group-item"><a href="http://blog.heliyon.com/">Heliyon blog</a></li>
                <li class="list-group-item"><a href="http://mcraveiro.blogspot.com/">Marco Craveiro</a></li>
                <li class="list-group-item"><a href="http://blogs.nature.com/actionpotential">Nature - Action potential</a></li>
                <li class="list-group-item"><a href="http://www.nature.com/nature/research/biological-sciences.rdf">Nature Biological Sciences Research</a></li>
                <li class="list-group-item"><a href="http://www.nature.com/nature/rdf/newsandcomment/newsandcomment">Nature News & Comment</a></li>
                <li class="list-group-item"><a href="https://neurorexia.wordpress.com">Neurorexia</a></li>
                <li class="list-group-item"><a href="http://blog.ted.com">Neuroscience - TED Blog</a></li>
                <li class="list-group-item"><a href="https://www.oist.jp/groups/computational-neuroscience-unit-erik-de-schutter">OIST Japan - CNU - Eric De Schutter</a></li>
                <li class="list-group-item"><a href="http://blog.ketyov.com/">Oscillatory Thoughts</a></li>
                <li class="list-group-item"><a href="http://blogs.plos.org/neuro">PLOS Neuroscience Community</a></li>
                <li class="list-group-item"><a href="http://blog.pubpeer.com">PubPeer blog</a></li>
                <li class="list-group-item"><a href="http://retractionwatch.com">Retraction watch</a></li>
                <li class="list-group-item"><a href="http://romainbrette.fr">Romain Brette</a></li>
                <li class="list-group-item"><a href="https://www.sciencenews.org/">Science News</a></li>
                <li class="list-group-item"><a href="http://www.nature.com/scientificamerican/current_issue/">Scientific American</a></li>
                <li class="list-group-item"><a href="http://www.nature.com/scientificamericanmind/current_issue/">Scientific American Mind</a></li>
                <li class="list-group-item"><a href="https://digest.bps.org.uk">The British Psychological Society - Research Digest</a></li>
                <li class="list-group-item"><a href="https://www.theguardian.com/science/neurophilosophy">The Guardian - Neurophilosophy by Mo Constandi</a></li>
                <li class="list-group-item"><a href="http://neurocritic.blogspot.com/">The Neurocritic</a></li>
                <li class="list-group-item"><a href="http://blogs.plos.org/plos">The Official PLOS Blog</a></li>
                <li class="list-group-item"><a href="https://blog.scienceexchange.com">The Science Exchange Blog</a></li>
                <li class="list-group-item"><a href="http://silverlab.org">The Silver Lab</a></li>
                <li class="list-group-item"><a href="http://biocomputation.herts.ac.uk/">UH Biocomputation group</a></li>
                <li class="list-group-item"><a href="https://xcorr.net">xcorr.net</a></li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer class="footer">
    <div class="container">
      <p class="text-center text-muted">All content on this page is owned by their respective owners.</p>
    </div>
  </footer>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
  <script type="text/javascript">
    $( document ).ready(function(){
    $("img").addClass("img-responsive");
    $("figure").addClass("img-responsive");
    });
  </script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script type="text/javascript" src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
</body>
</html>
